---
title: "Other useful functions"
author:
- name: Rohit Satyam
  affiliation: King Abdullah University of Science & Technology, Saudi Arabia
- name: Alberto Maillo
  affiliation: King Abdullah University of Science & Technology, Saudi Arabia
- name: David Gomez-Cabrero
  affiliation: King Abdullah University of Science & Technology, Saudi Arabia
- name: Arnab Pain
  affiliation: King Abdullah University of Science & Technology, Saudi Arabia
abstract: >
  This section covers the capability of plasmoRUtils to carry out other mundane bioinformatics tasks. 
clean: false
date: "`r format(Sys.Date(), '%d %B, %Y')`"
package: plasmoRUtils
output: 
  BiocStyle::html_document:
    toc_float: true
    toc: true
    number_sections: true
vignette: >
  %\VignetteIndexEntry{Miscellaneous functions}
  %\VignetteEncoding{UTF-8}
  %\VignetteDepends{BiocStyle} 
  %\VignettePackage{plasmoRUtils}
  %\VignetteEngine{knitr::rmarkdown}
bibliography: references.bib
link-citations: yes
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,cache = TRUE, warning = FALSE,message = FALSE,
  comment = "#>"
)
```

# Introduction

Other than search functions, the `r Rpackage("plasmoRUtils")` package offers various other functions that are routinely needed when performing bioinformatics analysis. Their usage is discussed below.

```{r}
# Load package and some other useful packages by using
suppressPackageStartupMessages(
  suppressWarnings({
    library(plasmoRUtils)
    library(dplyr)
    library(plyr)
    library(tibble)}))
```

## Making OrgDb and TxDb objects

For non-model organism, using R packages such as `r Biocpkg("clusterProfiler")` for enrichment analysis requires org.db packages. Unfortunately, this requires ability to understand and use `r Biocpkg("AnnotationDbi")` package. To make development of such packages quickly, we wrote a wrapper function that instantly creates a TAR file that can be used and shared by the user in no time.

```{r, eval=FALSE}
## List organism you are interested in making org.db package from VEuPathDB.
## Use the links from PlasmoDB directly or provide locally saved GFF and GAF files.

## Get taxonomy ID using listVeupathdb(c("primary_key","ncbi_tax_id"))

easymakeOrgdb(
  gff =
    "https://plasmodb.org/common/downloads/release-68/Pfalciparum3D7/gff/data/PlasmoDB-68_Pfalciparum3D7.gff",
  gaf =
    "https://plasmodb.org/common/downloads/release-68/Pfalciparum3D7/gaf/PlasmoDB-68_Pfalciparum3D7_Curated_GO.gaf.gz",
  out.dir = ".",
  taxid = 36329,
  genus = "Plasmodium",
  sp = "falciparum3D7",
  version = 0.1,
  verbose = FALSE,
  maintainer = "John doe <johndoe@gmail.com>" ## Your name if you are maintaining it.
)

## Once the package is ready, one can use clusterProfiler as follows

library(clusterProfiler)
library(org.Pfalciparum3D7.eg.db)


ego <- enrichGO(gene          = genes,
                universe      = bkg_genes,
                OrgDb         = org.Pfalciparum3D7.eg.db,
                keyType = "GID",
                ont           = "BP",
                minGSSize=3,
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
        readable      = FALSE)
```

The function above will create an org.db tar file which can be installed and loaded to be used with `r Biocpkg("clusterProfiler")`.

Similarly, if users want to make TxDb object in R to use other R packages such as `r Biocpkg("gDNAx")` used to access the genomic DNA contamination, they can do so easily using `easyTxDbmaker()` function.

```{r, eval=FALSE}
txdb<-easyTxDbmaker(
 gff="https://toxodb.org/common/downloads/release-68/TgondiiME49/gff/data/ToxoDB-68_TgondiiME49.gff",
 fasta="https://toxodb.org/common/downloads/release-68/TgondiiME49/fasta/data/ToxoDB-68_TgondiiME49_Genome.fasta",
 abbr="TgondiiME49",
 taxid=508771,org = "Toxoplasma gondii ME49",
 db = "ToxoDB release 68")

```

## Other easy functions

Users are also provided with easy functions which are wrapper functions around routinely used to analyze bioinformatics data.

### TPM normalization and visualization

In this section we will see how we can calculate effective gene lengths from a GFF file followed by TPM normalization and visualize the resulting normalized matrix. These functions are provided since most of the RNASeq data present on VEuPathDb is TPM normalized and therefore make it easy to plot the database procured values alongside your own samples.

```{r}

# To calculate the effective length of the genes you can use the following function
baseurl <- "https://plasmodb.org/common/downloads/release-68/"
getEffLen(paste0(baseurl,
"Pfalciparum3D7/gff/data/PlasmoDB-68_Pfalciparum3D7.gff")) %>% head()
```

The above function returns a dataframe which can be fed to the `easyTPM()` function alongside the count matrix. The function also adds the effective length as a column at the end so that people can find it easy to convert the values to the raw counts. This is also helpful since the exon-intron boundaries are always evolving as the new data-sets become available.

```{r}
library(ggplot2)
## Generating dummy effective length for quick demonstration
gene_info <- data.frame(GeneID = c("PF3D7_0100100", "PF3D7_0100200", "PF3D7_0100300"), Length = c(6492, 996, 3984))

## Making a dummy count matrix
count_matrix <- matrix(c(10, 20, 30, 40, 50, 60),
                       nrow = 3, ncol = 2,
                       dimnames = list(c("PF3D7_0100100", "PF3D7_0100200", "PF3D7_0100300"), c("sample1", "sample2"))
                       )

## TPM normalization
test <- easyTPM(count_matrix, gene_info) %>% rownames_to_column(var = "GeneID")

## TPM visualization as a bubble plot
df <- reshape2::melt(test[,-ncol(test)],na.rm = T)
easyExpPlot(df,x="variable",y="GeneID",value="value", type = "bubble")+labs(x="Samples",y="Gene IDs")
```

You can also plot normalized expression data as line plot if you have cluster of genes with similar pattern of expression. To demonstrate this we will use a subset of genes from [@Subudhi2020] time series dataset. We will perform TPM normalization, followed by log transformation and eventually use the Z-scores for visualization.

```{r}

data("subudhi2020")
count_matrix <- subudhi2020@assays@data$counts
baseurl <- "https://plasmodb.org/common/downloads/release-68/"
gene_info <- getEffLen(paste0(baseurl,"Pfalciparum3D7/gff/data/PlasmoDB-68_Pfalciparum3D7.gff"))

normalised <- easyTPM(count_matrix, gene_info) 

## Since normalized matrix have effective gene length column at the end
logTransformed = log2(normalised[,-ncol(normalised)] + 1)

## Get z-scores
z.mat <- t(scale(t(logTransformed), scale=TRUE, center=TRUE)) %>% as.data.frame() %>% rownames_to_column(var = "GeneID")

## Subsetting few genes of interest
gois <- c("PF3D7_1476300","PF3D7_0220800","PF3D7_0936200","PF3D7_0402200","PF3D7_1401600","PF3D7_0831600","PF3D7_0204100","PF3D7_1458300","PF3D7_0935800","PF3D7_0929400","PF3D7_1439000","PF3D7_0905400","PF3D7_1334600","PF3D7_1121300","PF3D7_0302500","PF3D7_1232800","PF3D7_0310400","PF3D7_1001600")

## Transforming data frame for visualisation
df <- z.mat %>% 
  subset(.,GeneID %in% gois) %>% 
  reshape2::melt(.,na.rm = T)

easyExpPlot(df,x="variable",y="value",value="GeneID")+
  labs(y="Z-scores",x="Samples")
```

`easyExpPlot()` can be used to plot expression values obtained from any kind of applied normalization and transformation.

### Support for NOISeq `readData` function for non-model organism

When using `r Biocpkg("NOISeq")` for QC, users are often required to provide various information alongside count matrices. These information includes biotype, chromosome, gc content, length of the features etc. These arguments are listed as optional but are required by `r Biocpkg("NOISeq")` package for making QC plots. To enable users to get these information from the same reference FASTA file and GFF file used by user for alignment, we wrote `easyNOISeqAnnot()` function that enable users to quickly get this information from GTF/GFF file and FASTA file and return a data frame. These can be provided as a URL or as locally stored files.

```{r}
gene_info <- easyNOISeqAnnot(
 gff="https://toxodb.org/common/downloads/release-68/EpraecoxHoughton/gff/data/ToxoDB-68_EpraecoxHoughton.gff",
 fasta = "https://toxodb.org/common/downloads/release-68/EpraecoxHoughton/fasta/data/ToxoDB-68_EpraecoxHoughton_Genome.fasta")

gene_info %>% head()

```

You can now pass the data frame contents to `NOISeq::readData()` function as follows.

```{r, eval=FALSE}
data_NOISEQ <- NOISeq::readData(data = counts,
                        length=setNames(gene_info$length, gene_info$gene_id),
                        gc=setNames(gene_info$gc, gene_info$gene_id),
                        biotype=setNames(gene_info$gene_biotype, gene_info$gene_id),
                        chromosome = gene_info[,c("chr","starts","ends")],
                        factors = meta)

```

### Check synchronization of Bulk RNASeq samples using `easyLabelTransfer()`

`easyLabelTransfer` is a wrapper function written around `r Biocpkg("SingleR")` `SingleR()` function to quickly use count matrices from Bulk/ Single cell Reference data sets to check synchronization (time point and stage) of the Parasite Bulk RNASeq samples and can be used for QC purposes.

Here, for demonstration purpose, we will use single cell RNASeq data from Malaria Cell Atlas and use [@Subudhi2020] Bulk time-series dataset to transfer labels and see the distribution of the cells across different time points.

Other data sets provided with the package includes:

-   Time course Microarray dataset from [@painter2018genome]: `data("painter2018")`
-   Time course Bulk RNASeq (Single-end) from [@toenhake2018chromatin]: `data("toenhake2018")`
-   Reanalyzed scRNAseq of \*T. gondii\* data from [@lou2024single]: `data("gubbels2024")`
-   Reanalyzed scRNAseq of \*T. gondii\* data from [@xue2020single]: `data("boothroyd2020")`

Next, Malaria cell atlas (MCA) is a database comprising of single cell data sets and can be accessed using the following functions.

1.  `listMCA()` function enables you to filter and list scRNASeq data sets for which data download links are available. User can use this function to view a table to metadata about the dataset or download these datasets recursively rather than manually downloading single dataset at a time.
2.  `easyMCA()` enables users to import the count matrix and metadata directly in the R environment directly. Written mainly for pipeline development.

Now, lets use the function enlisted above alongside `easyLabelTransfer()` as shown below.

> Note: The output obtained from `easyLabelTransfer()` function is a DFrame object and can be fed to other SingleR functions for visual purposes.

```{r}
library(SingleR)
set.seed(12458)
## Fetching the URL
mcalist <- listMCA()
data("subudhi2020") ## reference dataset

## Using this reference set
url <- "https://www.malariacellatlas.org/downloads/pf-ch10x-set4-biorxiv.zip"

raw_counts <- easyMCA(url,type = "raw")
rownames(raw_counts) <- gsub("-","_",rownames(raw_counts))
meta <- easyMCA(url,type="data")

## Retaining only Asexual stage cells and Lab isolates.
meta <- subset(meta, meta$STAGE_LR %in% c("ring","trophozoite","schizont") & DAY != "Field")
raw_counts <- raw_counts[,rownames(meta)]

## Filtering away field isolates and asexual stage cells

labels <- easyLabelTransfer(queryCounts = raw_counts,
                            refCounts = subudhi2020@assays@data$counts, referenceMeta = subudhi2020@colData, labelCol = "timetag", isrefBulk = TRUE)

table(labels$pruned.labels)

plotScoreHeatmap(labels)
```

You can see that cells from some time points are underrepresented in our Malaria Cell Atlas.

### GO Enrichment using `easytopGO()`

VEuPathDB constituent database allow users to perform GO Enrichment analysis. However, there is no option to set the background genes and by default all the genes in the are used. This is not an ideal approach. Moreover, since Gene Ontologies are curated using different pipelines by different enrichment providers, the results might differ just based on source of Ontologies. Similarly the Ontology data might not be available in other databases such as BioMart and you would like to use custom GAF file

Users of parasite domain might wish to use GAF files provided by VEuPathDB constituent database and of specific genome assembly as per their hypothesis. To enable users to do that, GO enrichment analysis can be performed easily and quickly using our `easytopGO()` wrapper function. This function requires users to provide a named numeric vector, where names are Ensembl Gene IDs and numeric vector could be adjusted p-values. Users can also supply genes to be used as background using `bkggset` argument. Finally users can use `gaf` argument to supply .gaf file obtained from VEuPathDBs constituent database of interest.

For well known organisms, Biomart Ontologies can also be used. In such case user has to provide `mart` argument and figure out key for emsembl gene IDs . However, many times the parasite assemblies used as reference by Ensembl might not match with parasite assembly used by VEuPathDB. In such cases there might be mismatches between gene IDs (see one such issue [here](https://github.com/Huber-group-EMBL/biomaRt/issues/110)).

You can also change `category` argument to specify which GO category you want to test enrichment for. Since for most of the parasites, sub-cellular localization and Molecular function information is sparingly available, "BP" has been set as the default.

Finally, you can also use ORA results obtained from VEuPathDBs as input to `easytopGO()`.

```{r}
# Making a numeric vector of padjusted values
gois <- c("PF3D7_0102200","PF3D7_0207400","PF3D7_0207500","PF3D7_0207600","PF3D7_0207700","PF3D7_0207800","PF3D7_0207900","PF3D7_0208000","PF3D7_0404700","PF3D7_0501500","PF3D7_0502400","PF3D7_0618000","PF3D7_0731800","PF3D7_0930300","PF3D7_1116000","PF3D7_1247800","PF3D7_1334700")


p_values <- c(1.767929e-49,3.886063e-148,6.459285e-269,0.000000e+00,6.842121e-132,
5.282318e-178,1.140221e-224,1.625665e-100,6.177129e-08,1.484155e-03,2.789648e-10,2.300720e-12,7.450697e-30,3.417972e-89,4.846689e-73,1.207966e-11,4.173777e-51)
names(p_values) <- gois

## Using all genes captured as a background

background.gset <- rownames(subudhi2020@assays@data$counts)
baseurl <- "https://plasmodb.org/common/downloads/Current_Release/"
url<-paste0(baseurl,"Pfalciparum3D7/gaf/PlasmoDB-68_Pfalciparum3D7_GO.gaf.gz")

## Performing ORA
gores<-easytopGO(geneID = p_values,useGAF = TRUE,useBiomart = FALSE,gaf=url,
bkggset = background.gset, category = "BP", stats = "ks")

## Plotting the results. 
easyGOPlot(gores, title = "GO Biological processes", limit = 20, sortby = "ks")

```

The `easyGOPlot()` is adapted and modified version of what has been provided by [Kevin Blighe](https://www.biostars.org/p/471549/) on Biostars. You can use `limit` argument to limit number of terms to be plotted.

### Screening the Signal Peptides from PDBs

We also offer a convenient function, `easyAF2Signal()`, to diagnose potential false-positive signal peptides using SignalP predictions from the VEuPathDB database and AlphaFold2 structures, based on an observational study conducted by [@Sanaboyana2024]. The authors found that true N-terminal signal peptides (\~24-24 amino acids) are typically disengaged from the protein body and lack atomic contacts, and AlphaFold2 attempts to model them in a similar way.

`easyAF2Signal()` is an R equivalent with slight modifications to the Fortran code provided by the authors, which reports additional information, such as the number of residues remaining after pLDDT filtering.

We did this modification because, if all the signal peptide residues have low pLDDT scores and are filtered out, no residues will remain to calculate contacts with the protein body, resulting in zero contacts. This could give a false impression that the first 25 amino acids are signal peptides, but it cannot be confirmed, as the zero-contact observed is due to no residues remaining after filtering, not because the signal peptide was disengaged from the protein body.

```{r}

easyAF2Signal("https://alphafold.ebi.ac.uk/files/AF-Q9TY95-F1-model_v4.pdb")

```

In the example above, we see that signal peptide has low mean pLDDT score and therefore there are no residues left post filtering of bad quality residues leading to zero residue-residue counts `res_res_conts`. Thus we can't say for sure if the PDB in hand have signal peptide or not.

# Session Info {.unnumbered}

```{r}
sessionInfo()
```

# References {.unnumbered}
