<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Accessing component databases of VEuPathDB • plasmoRUtils</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Accessing component databases of VEuPathDB">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">plasmoRUtils</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Introduction_to_plasmoRUtils.html">Introduction to plasmoRUtils</a></li>
    <li><a class="dropdown-item" href="../articles/Gene_ID_Conversion.html">Accessing component databases of VEuPathDB</a></li>
    <li><a class="dropdown-item" href="../articles/Miscellaneous_function.html">Other useful functions</a></li>
    <li><a class="dropdown-item" href="../articles/Need_for_reanalysis.html">RNASeq: Need for reanalysis?</a></li>
<<<<<<< HEAD
    <li><a class="dropdown-item" href="../articles/retrieving_orthologs.html">Retrieving Orthologs</a></li>
=======
>>>>>>> main
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Accessing component databases of VEuPathDB</h1>
                        <h4 data-toc-skip class="author">Rohit
Satyam</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Alberto
Maillo</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">David
Gomez-Cabrero</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Arnab
Pain</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
<<<<<<< HEAD
Arabia<br><h4 data-toc-skip class="date">21 August, 2025</h4>
=======
Arabia<br><h4 data-toc-skip class="date">20 July, 2025</h4>
>>>>>>> main
      

      <div class="d-none name"><code>Gene_ID_Conversion.Rmd</code></div>
    </address>
</address>
</address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This section covers the capability of <code>plasmoRUtils</code>
      to access various VEuPathDb database and carry out ID conversion
      tasks, fetch preconfigured data tables and few other tasks.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <em>plasmoRUtils</em> package streamlines access to VEuPathDB’s
family of 12 specialized databases such as ToxoDB, PlasmoDB, and
PiroplasmaDB <span class="citation">(<a href="#ref-amos2022veupathdb">Amos et al. 2022</a>)</span>. It provides
direct data retrieval capabilities through VEuPathDB’s RESTful API,
enabling seamless integration of biological data into R workflows. The
package supports downloading both standard and customized data tables,
making it particularly valuable for researchers needing to combine data
from multiple sources for downstream analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package and some other useful packages by using</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rohit-satyam.github.io/plasmoRUtils/">plasmoRUtils</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://had.co.nz/plyr" class="external-link">plyr</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="gene-id-conversion">Gene ID Conversion<a class="anchor" aria-label="anchor" href="#gene-id-conversion"></a>
</h2>
<p>A common challenge in bioinformatics involves mapping between
different identifier systems across databases. For apicomplexan
research, this might include converting between UniProt IDs, legacy gene
identifiers, and current Ensembl gene IDs. The <code><a href="../reference/toGeneid.html">toGeneid()</a></code>
function addresses this need by retrieving up-to-date annotations from
VEuPathDB databases, supporting bidirectional conversion between various
ID types through flexible parameter specification.</p>
<div class="section level3">
<h3 id="retrieving-gene-annotations-and-alternative-ids">Retrieving Gene Annotations and Alternative IDs<a class="anchor" aria-label="anchor" href="#retrieving-gene-annotations-and-alternative-ids"></a>
</h3>
<p>The <code><a href="../reference/toGeneid.html">toGeneid()</a></code> function enables annotation retrieval
when provided with Ensembl gene IDs. By default, it returns essential
information including gene names, symbols, outdated gene IDs, and
protein Uniprot IDs. The function’s versatility extends to supporting
custom field requests through its <code>customFields</code> parameter,
with available options documented in the <code><a href="../reference/getTable.html">getTable()</a></code> help
section.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get annotations for list of geneIDs for PF3D7</span></span>
<span><span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0420300"</span>, <span class="st">"PF3D7_0621000"</span><span class="op">)</span>, from<span class="op">=</span><span class="st">"ensembl"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 10</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     `Product Description`        `Gene Strand` `Gene Name or Symbol`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_0420300 AP2 domain transcription fa… forward       ApiAP2               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_0621000 RNA polymerase subunit sigm… forward       ApSigma              </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6 more variables: `Previous ID(s)` &lt;chr&gt;, `Entrez Gene ID` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `UniProt ID(s)` &lt;chr&gt;, `Protein Length` &lt;chr&gt;, `# TM Domains` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `SignalP Peptide` &lt;chr&gt;</span></span></span>
<span><span class="co">## Get annotations for list of geneIDs for organisms other than PF3D7</span></span>
<span><span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span>inputid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGME49_304740"</span>,<span class="st">"TGME49_208030"</span><span class="op">)</span>,from<span class="op">=</span><span class="st">"ensembl"</span>,org<span class="op">=</span><span class="st">"Toxoplasma gondii ME49"</span>, db<span class="op">=</span><span class="st">"toxodb"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 10</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     `Product Description`        `Gene Strand` `Gene Name or Symbol`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TGME49_208030 microneme protein MIC4       forward       MIC4                 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> TGME49_304740 rhoptry kinase family prote… reverse       ROP35                </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6 more variables: `Previous ID(s)` &lt;chr&gt;, `Entrez Gene ID` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `UniProt ID(s)` &lt;chr&gt;, `Protein Length` &lt;chr&gt;, `# TM Domains` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `SignalP Peptide` &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="co">## Convert uniprot IDs back to gene IDs. It will also provide Product description and Gene Symbol</span></span>
<span></span>
<span><span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span>inputid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q8I1N6"</span>,<span class="st">"C6KT48"</span><span class="op">)</span>,from<span class="op">=</span><span class="st">"uniprot"</span>,to<span class="op">=</span><span class="st">"ensembl"</span> <span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     `UniProt ID(s)`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_0420300 Q8I1N6         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_0621000 C6KT48</span></span>
<span></span>
<span></span>
<span><span class="co">## Using customFields to get only columns of interest</span></span>
<span><span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span>inputid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGME49_304740"</span>,<span class="st">"TGME49_208030"</span><span class="op">)</span>,</span>
<span>         from<span class="op">=</span><span class="st">"ensembl"</span>,org<span class="op">=</span><span class="st">"Toxoplasma gondii ME49"</span>,</span>
<span>         db<span class="op">=</span><span class="st">"toxodb"</span>,</span>
<span>         customFields<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"primary_key"</span>,<span class="st">"predicted_go_component"</span>,<span class="st">"annotated_go_function"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 3</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     `Computed GO Components` `Curated GO Functions`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TGME49_208030 extracellular region     N/A                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> TGME49_304740 N/A                      N/A</span></span></code></pre></div>
<blockquote>
<p><strong>Note:</strong> Successful ID conversion requires precise
organism nomenclature matching VEuPathDB’s conventions. For example,
<em><code>Toxoplasma gondii ME49</code></em> must include proper spacing
and special characters. Invalid query would be
<em><code>Toxoplasma gondiiME49</code></em> or short forms
<code>TgME49</code>.</p>
</blockquote>
<p>This functionality proves particularly valuable for enhancing
differential expression analysis results with comprehensive annotations,
enabling complete workflow automation without leaving command-line
interfaces on HPC systems.</p>
</div>
<div class="section level3">
<h3 id="accessing-preconfigured-data-tables-from-veupathdbs-component-sites">Accessing Preconfigured Data Tables from VEuPathDB’s component
sites<a class="anchor" aria-label="anchor" href="#accessing-preconfigured-data-tables-from-veupathdbs-component-sites"></a>
</h3>
<p>Since VEuPathDB API documentation specifically encourages to use
specific organism database as quoted below, we developed
<code><a href="../reference/getTable.html">getTable()</a></code> function to fetch fields of interests from each
database separately.</p>
<blockquote>
<p>There are 12 component sites and one portal: VEuPathDB.org. The
component sites are: AmoebaDB, CryptoDB, FungiDB, GiardiaDB, HostDB,
MicrosporidiaDB, PiroplasmaDB, PlasmoDB, ToxoDB, TrichDB, TriTrypDB and
VectorBase. For most record types (all but dataset and organism), when
running a search, the portal reaches out to component sites to get the
search results. That means it will be faster to use a component site
directly when you can.</p>
</blockquote>
<p>Some frequently required fields have been provided in the help
section of <code><a href="../reference/getTable.html">getTable()</a></code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## To fetch table for all the genes present in an organism</span></span>
<span></span>
<span><span class="fu"><a href="../reference/getTable.html">getTable</a></span><span class="op">(</span>org<span class="op">=</span><span class="st">"Plasmodium falciparum 3D7"</span>, db<span class="op">=</span><span class="st">"plasmodb"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 10</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     `Product Description`        `Gene Strand` `Gene Name or Symbol`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_0100100 erythrocyte membrane protei… forward       VAR                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_0100200 rifin                        reverse       RIF                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PF3D7_0100300 erythrocyte membrane protei… reverse       VAR                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PF3D7_0100400 rifin                        forward       RIF                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PF3D7_0100500 erythrocyte membrane protei… reverse       N/A                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PF3D7_0100600 rifin                        reverse       RIF                  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6 more variables: `Previous ID(s)` &lt;chr&gt;, `Entrez Gene ID` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `UniProt ID(s)` &lt;chr&gt;, `Protein Length` &lt;chr&gt;, `# TM Domains` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `SignalP Peptide` &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="co">## User can also provide custom fields. For example we wish to download the P. falciparum 3D7 Proteome and phosphoproteome data during intraerythrocytic development (Quantitative) (Pease et al.)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/getTable.html">getTable</a></span><span class="op">(</span>org<span class="op">=</span><span class="st">"Plasmodium falciparum 3D7"</span>, db<span class="op">=</span><span class="st">"plasmodb"</span>, customFields <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"primary_key"</span>,<span class="st">"pan_6365"</span>,<span class="st">"pan_6366"</span>,<span class="st">"pan_6367"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   `Gene ID` Ring Ave (Global pro…¹ Troph Ave (Global pr…² Schizont Ave (Global…³</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_01… N/A                    N/A                    N/A                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_01… N/A                    N/A                    N/A                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PF3D7_01… N/A                    N/A                    N/A                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PF3D7_01… N/A                    N/A                    N/A                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PF3D7_01… N/A                    N/A                    N/A                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PF3D7_01… N/A                    N/A                    N/A                   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated names: ¹​`Ring Ave (Global proteome and phosphoproteome)`,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​`Troph Ave (Global proteome and phosphoproteome)`,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ³​`Schizont Ave (Global proteome and phosphoproteome)`</span></span></span></code></pre></div>
<p>For more information on fields that can be supplied to
<code><a href="../reference/getTable.html">getTable()</a></code>, use the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Go to database of your interest (Say “PlasmoDB”)</p></li>
<li><p>Click on <code>Annotation, curation and identifiers</code> tab on
your left and select <code>List of IDs</code></p></li>
<li><p>Scroll down and click on
<code>Build a Web Services URL from this Search &gt;&gt;</code>
hyperlink.</p></li>
<li><p>Under the section <code>Choose Columns:</code> choose fields of
your interest. Most of these fields are included in help section of
<code><a href="../reference/getTable.html">getTable()</a></code>. However, fields specific to a particular
database such as dataset related fields (starts with “pan_”) are
excluded.</p></li>
<li><p>Once you select fields, they are updated in <strong>POST</strong>
section of the webpage query builder.</p></li>
</ol>
<p>The constituent databases of VEuPathDB also provide some
preconfigured tables which can not be fetched via
<code><a href="../reference/getTable.html">getTable()</a></code> function. To enable users to fetch such tables,
we wrote another function called <code><a href="../reference/getPreconfiguredTable.html">getPreconfiguredTable()</a></code>
the usage of which has been shown below.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Fetch pathway table for all the genes from MPMP database</span></span>
<span></span>
<span><span class="fu"><a href="../reference/getPreconfiguredTable.html">getPreconfiguredTable</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"Plasmodium falciparum 3D7"</span>,db <span class="op">=</span> <span class="st">"plasmodb"</span>,customField <span class="op">=</span> <span class="st">"MetabolicPathwaysMPMP"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     pathway_id          Pathway                             Activity</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_0100100 struc_telom         Structure of telomere and sub-telo… erythro…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_0100100 PfEMP1              PfEMP1 domain architectures         erythro…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PF3D7_0100100 PQS                 P. falciparum genes harboring G-qu… erythro…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PF3D7_0100100 cytoadherencescheme Interactions between modified host… erythro…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PF3D7_0100100 gene_lumef          Gene expression affected by lumefa… erythro…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PF3D7_0100100 virulence           Candidate genes related to virulen… erythro…</span></span></code></pre></div>
<p>Please note that the MPMP pathway version provided by PlasmoDB is <a href="https://plasmodb.org/plasmo/app/record/dataset/DS_1e177b728b" class="external-link">outdated</a>
(<code>03-2019</code>). Some pathways have been revised or removed in
its entirety. If you wish to access the latest MPMP version, you can use
<code>data("mpmp.28Aug2024")</code> for you analysis which was scraped
by us. If you wish to use this geneset for MPMP pathway enrichment
analysis using <em><a href="https://CRAN.R-project.org/package=pathfindR" class="external-link">pathfindR</a></em>,
you can do so by using <code>data("pathfindrMPMP")</code>.</p>
<p>Similarly, predictions like <code>TMHMM</code> and
<code>SignalP</code> and <code>InterPro</code> have not been updated
given the recent funding crunch and should be therefore used with
caution. We will discuss more about it in a separate tutorial.</p>
<blockquote>
<p>Note: We urge the users to cite the original articles of the related
datasets alongside plasmoRUtils.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="fetching-genome-metadata-and-strain-names">Fetching genome metadata and strain names<a class="anchor" aria-label="anchor" href="#fetching-genome-metadata-and-strain-names"></a>
</h3>
<p>In above examples, we saw the importance of passing exact name to the
<code>org</code> argument for <code><a href="../reference/toGeneid.html">toGeneid()</a></code> to function
properly. A helper function is provided to achieve this called
<code><a href="../reference/listVeupathdb.html">listVeupathdb()</a></code>. By default, 11 columns are returned
including organism name, the respective database present in “VEuPathDB
Project” column and some more additional information. However, you can
limit the search to columns of your interests as shown below. As stated
by VEuPathDB:</p>
<blockquote>
<p>The best use of the VEuPathDB portal is to get a table with all
organisms in our sites, and for each organism: the component site, and
the urls to access their fasta and gff files.</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/listVeupathdb.html">listVeupathdb</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 11</span></span></span>
<span><span class="co">#&gt;   Organism                 Species Genome Fasta Downloa…¹ CDS Fasta Download L…²</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Edhazardia aedis USNM 4… Edhaza… http://MicrosporidiaD… http://MicrosporidiaD…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Kluyveromyces marxianus… Kluyve… http://FungiDB.org/co… http://FungiDB.org/co…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Aspergillus luchuensis … Asperg… http://FungiDB.org/co… http://FungiDB.org/co…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Epichloe glyceriae E277  Epichl… http://FungiDB.org/co… http://FungiDB.org/co…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Aspergillus versicolor … Asperg… http://FungiDB.org/co… http://FungiDB.org/co…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Aspergillus sydowii CBS… Asperg… http://FungiDB.org/co… http://FungiDB.org/co…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated names: ¹​`Genome Fasta Download Link`,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​`CDS Fasta Download Link`</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7 more variables: `Transcript Fasta Download Link` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Protein Fasta Download Link` &lt;chr&gt;, `VEuPathDB Project` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Genes &lt;chr&gt;, `GFF Download Link` &lt;chr&gt;, `Genome Source` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Structural Annotation Source` &lt;chr&gt;</span></span></span>
<span><span class="fu"><a href="../reference/listVeupathdb.html">listVeupathdb</a></span><span class="op">(</span>customFields<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"species"</span>, <span class="st">"project_id"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   Species                  `VEuPathDB Project`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Entamoeba nuttalli       AmoebaDB           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Acanthamoeba castellanii AmoebaDB           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Mastigamoeba balamuthi   AmoebaDB           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Acanthamoeba sp.         AmoebaDB           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Acanthamoeba sp.         AmoebaDB           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Acanthamoeba sp.         AmoebaDB</span></span></code></pre></div>
<p>Since this function also provide URLs of FASTA and GFF files, you can
use it to find the URLs of the files you are interested in and import
them in R directly without leaving the console.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/listVeupathdb.html">listVeupathdb</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">.</span>,<span class="va">Organism</span> <span class="op">==</span><span class="st">"Edhazardia aedis USNM 41457"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`GFF Download Link`</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html" class="external-link">import.gff3</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 12 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames    ranges strand |    source                type     score</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;factor&gt;            &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1] AFBI030000... 5287-6076      + | VEuPathDB protein_coding_gene        NA</span></span>
<span><span class="co">#&gt;   [2] AFBI030000... 5287-6076      + | VEuPathDB mRNA                       NA</span></span>
<span><span class="co">#&gt;   [3] AFBI030000... 5287-6076      + | VEuPathDB exon                       NA</span></span>
<span><span class="co">#&gt;   [4] AFBI030000... 5447-6043      + | VEuPathDB CDS                        NA</span></span>
<span><span class="co">#&gt;   [5] AFBI030000... 5287-5446      + | VEuPathDB five_prime_UTR             NA</span></span>
<span><span class="co">#&gt;   [6] AFBI030000... 6044-6076      + | VEuPathDB three_prime_UTR            NA</span></span>
<span><span class="co">#&gt;           phase            ID   description   ebi_biotype          Parent</span></span>
<span><span class="co">#&gt;       &lt;integer&gt;   &lt;character&gt;   &lt;character&gt;   &lt;character&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">#&gt;   [1]      &lt;NA&gt;    EDEG_00001 hypothetic... protein_co...                </span></span>
<span><span class="co">#&gt;   [2]      &lt;NA&gt; EDEG_00001... hypothetic...          &lt;NA&gt;      EDEG_00001</span></span>
<span><span class="co">#&gt;   [3]      &lt;NA&gt; exon_EDEG_...          &lt;NA&gt;          &lt;NA&gt;   EDEG_00001...</span></span>
<span><span class="co">#&gt;   [4]         0 EDEG_00001...          &lt;NA&gt;          &lt;NA&gt;   EDEG_00001...</span></span>
<span><span class="co">#&gt;   [5]      &lt;NA&gt; utr_EDEG_0...          &lt;NA&gt;          &lt;NA&gt;   EDEG_00001...</span></span>
<span><span class="co">#&gt;   [6]      &lt;NA&gt; utr_EDEG_0...          &lt;NA&gt;          &lt;NA&gt;   EDEG_00001...</span></span>
<span><span class="co">#&gt;       gene_ebi_biotype     gene_id protein_source_id            Note</span></span>
<span><span class="co">#&gt;            &lt;character&gt; &lt;character&gt;       &lt;character&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">#&gt;   [1]             &lt;NA&gt;        &lt;NA&gt;              &lt;NA&gt;                </span></span>
<span><span class="co">#&gt;   [2]    protein_co...        &lt;NA&gt;              &lt;NA&gt;                </span></span>
<span><span class="co">#&gt;   [3]             &lt;NA&gt;  EDEG_00001              &lt;NA&gt;                </span></span>
<span><span class="co">#&gt;   [4]             &lt;NA&gt;  EDEG_00001     EDEG_00001...                </span></span>
<span><span class="co">#&gt;   [5]             &lt;NA&gt;        &lt;NA&gt;              &lt;NA&gt;                </span></span>
<span><span class="co">#&gt;   [6]             &lt;NA&gt;        &lt;NA&gt;              &lt;NA&gt;                </span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 342 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="mapping-pdb-ids-to-gene-ids">Mapping PDB IDs to Gene IDs<a class="anchor" aria-label="anchor" href="#mapping-pdb-ids-to-gene-ids"></a>
</h3>
<p>There is currently no facility in VEuPathDB to convert the PDB IDs to
respective gene IDs. If PDB ID corresponds to a multimer complex and if
you have multiple such PDB ids, it becomes arduous to map them to Gene
IDs manually. To provide solution to this issue, we can first convert
the PDB chains to Uniprot IDs using our own <code><a href="../reference/pdb2uniprot.html">pdb2uniprot()</a></code>
function and then can use <code><a href="../reference/toGeneid.html">toGeneid()</a></code> function to obtain
gene IDs.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pdbids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"7D2W"</span>,<span class="st">"4U5A"</span>,<span class="st">"6E10"</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">pdbids</span>, <span class="va">pdb2uniprot</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/ldply.html" class="external-link">ldply</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">geneids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span>inputid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">attribute</span><span class="op">)</span>,from <span class="op">=</span> <span class="st">"uniprot"</span>,to <span class="op">=</span> <span class="st">"ensembl"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Combining the geneIDs with df</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">df</span>,<span class="va">geneids</span>,all<span class="op">=</span><span class="cn">TRUE</span>, by.x<span class="op">=</span><span class="st">"attribute"</span>, by.y<span class="op">=</span><span class="st">"UniProt ID(s)"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;    attribute entity_id chain_id struct_asym_id unp_start unp_end</span></span>
<span><span class="co">#&gt; 1 A0A143ZZR8         1        A              A        27     206</span></span>
<span><span class="co">#&gt; 2 A0A143ZZR8         1        B              B        27     206</span></span>
<span><span class="co">#&gt; 3     Q75UY1         1        C              C        42     241</span></span>
<span><span class="co">#&gt; 4     Q75UY1         1        D              D        42     241</span></span>
<span><span class="co">#&gt; 5     Q75UY1         1        A              A        42     241</span></span>
<span><span class="co">#&gt; 6     Q75UY1         1        B              B        42     241</span></span>
<span><span class="co">#&gt;   start.residue_number start.author_residue_number start.author_insertion_code</span></span>
<span><span class="co">#&gt; 1                    3                          NA                            </span></span>
<span><span class="co">#&gt; 2                    3                          NA                            </span></span>
<span><span class="co">#&gt; 3                    2                          NA                            </span></span>
<span><span class="co">#&gt; 4                    2                          NA                            </span></span>
<span><span class="co">#&gt; 5                    2                          NA                            </span></span>
<span><span class="co">#&gt; 6                    2                          NA                            </span></span>
<span><span class="co">#&gt;   end.residue_number end.author_residue_number end.author_insertion_code query</span></span>
<span><span class="co">#&gt; 1                182                       181                            7D2W</span></span>
<span><span class="co">#&gt; 2                182                       181                            7D2W</span></span>
<span><span class="co">#&gt; 3                201                        NA                            4U5A</span></span>
<span><span class="co">#&gt; 4                201                        NA                            4U5A</span></span>
<span><span class="co">#&gt; 5                201                        NA                            4U5A</span></span>
<span><span class="co">#&gt; 6                201                        NA                            4U5A</span></span>
<span><span class="co">#&gt;         Gene ID</span></span>
<span><span class="co">#&gt; 1 PF3D7_1372300</span></span>
<span><span class="co">#&gt; 2 PF3D7_1372300</span></span>
<span><span class="co">#&gt; 3          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6          &lt;NA&gt;</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session">Session<a class="anchor" aria-label="anchor" href="#session"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 26100)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_India.utf8  LC_CTYPE=English_India.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_India.utf8 LC_NUMERIC=C                  </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_India.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Riyadh</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<<<<<<< HEAD
<span><span class="co">#&gt; [1] plyr_1.8.9         dplyr_1.1.4        plasmoRUtils_1.1.0 rlang_1.1.6       </span></span>
=======
<span><span class="co">#&gt; [1] plyr_1.8.9         dplyr_1.1.4        plasmoRUtils_1.0.0 rlang_1.1.6       </span></span>
>>>>>>> main
<span><span class="co">#&gt; [5] readr_2.1.5        janitor_2.2.1      BiocStyle_2.32.1  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] segmented_2.1-4             fs_1.6.6                   </span></span>
<span><span class="co">#&gt;   [3] ProtGenerics_1.36.0         matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt;   [5] bitops_1.0-9                lubridate_1.9.4            </span></span>
<span><span class="co">#&gt;   [7] pRoloc_1.44.1               httr_1.4.7                 </span></span>
<span><span class="co">#&gt;   [9] RColorBrewer_1.1-3          doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [11] ggsci_3.2.0                 tools_4.4.1                </span></span>
<span><span class="co">#&gt;  [13] MSnbase_2.30.1              backports_1.5.0            </span></span>
<span><span class="co">#&gt;  [15] utf8_1.2.6                  R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [17] lazyeval_0.2.2              withr_3.0.2                </span></span>
<span><span class="co">#&gt;  [19] prettyunits_1.2.0           gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [21] preprocessCore_1.66.0       cli_3.6.5                  </span></span>
<span><span class="co">#&gt;  [23] Biobase_2.64.0              textshaping_1.0.1          </span></span>
<span><span class="co">#&gt;  [25] gt_1.0.0                    sass_0.4.10                </span></span>
<span><span class="co">#&gt;  [27] topGO_2.56.0                mvtnorm_1.3-3              </span></span>
<span><span class="co">#&gt;  [29] randomForest_4.7-1.2        proxy_0.4-27               </span></span>
<span><span class="co">#&gt;  [31] pkgdown_2.1.3               Rsamtools_2.20.0           </span></span>
<span><span class="co">#&gt;  [33] systemfonts_1.2.3           txdbmaker_1.0.1            </span></span>
<span><span class="co">#&gt;  [35] AnnotationForge_1.46.0      dichromat_2.0-0.1          </span></span>
<<<<<<< HEAD
<span><span class="co">#&gt;  [37] parallelly_1.45.1           limma_3.60.6               </span></span>
=======
<span><span class="co">#&gt;  [37] parallelly_1.45.0           limma_3.60.6               </span></span>
>>>>>>> main
<span><span class="co">#&gt;  [39] rstudioapi_0.17.1           impute_1.78.0              </span></span>
<span><span class="co">#&gt;  [41] RSQLite_2.4.1               FNN_1.1.4.1                </span></span>
<span><span class="co">#&gt;  [43] generics_0.1.4              BiocIO_1.14.0              </span></span>
<span><span class="co">#&gt;  [45] vroom_1.6.5                 gtools_3.9.5               </span></span>
<span><span class="co">#&gt;  [47] car_3.1-3                   dendextend_1.19.1          </span></span>
<span><span class="co">#&gt;  [49] GO.db_3.19.1                Matrix_1.7-1               </span></span>
<span><span class="co">#&gt;  [51] MALDIquant_1.22.3           drawProteins_1.24.0        </span></span>
<span><span class="co">#&gt;  [53] S4Vectors_0.42.1            abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [55] lifecycle_1.0.4             yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [57] snakecase_0.11.1            carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [59] SummarizedExperiment_1.34.0 recipes_1.3.1              </span></span>
<span><span class="co">#&gt;  [61] SparseArray_1.4.8           BiocFileCache_2.12.0       </span></span>
<span><span class="co">#&gt;  [63] grid_4.4.1                  blob_1.2.4                 </span></span>
<span><span class="co">#&gt;  [65] promises_1.3.3              crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [67] PSMatch_1.8.0               lattice_0.22-6             </span></span>
<<<<<<< HEAD
<span><span class="co">#&gt;  [69] beachmat_2.20.0             annotate_1.82.0            </span></span>
<span><span class="co">#&gt;  [71] GenomicFeatures_1.56.0      chromote_0.5.1             </span></span>
<span><span class="co">#&gt;  [73] mzR_2.38.0                  KEGGREST_1.44.1            </span></span>
<span><span class="co">#&gt;  [75] pillar_1.11.0               knitr_1.50                 </span></span>
<span><span class="co">#&gt;  [77] GenomicRanges_1.56.2        rjson_0.2.23               </span></span>
<span><span class="co">#&gt;  [79] lpSolve_5.6.23              future.apply_1.20.0        </span></span>
<span><span class="co">#&gt;  [81] codetools_0.2-20            mgsub_1.7.3                </span></span>
<span><span class="co">#&gt;  [83] glue_1.8.0                  pcaMethods_1.96.0          </span></span>
<span><span class="co">#&gt;  [85] data.table_1.17.8           MultiAssayExperiment_1.30.3</span></span>
<span><span class="co">#&gt;  [87] vctrs_0.6.5                 png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [89] gtable_0.3.6                kernlab_0.9-33             </span></span>
<span><span class="co">#&gt;  [91] cachem_1.1.0                gower_1.0.2                </span></span>
<span><span class="co">#&gt;  [93] xfun_0.52                   prodlim_2025.04.28         </span></span>
<span><span class="co">#&gt;  [95] S4Arrays_1.4.1              polyglotr_1.7.0            </span></span>
<span><span class="co">#&gt;  [97] coda_0.19-4.1               survival_3.8-3             </span></span>
<span><span class="co">#&gt;  [99] ncdf4_1.24                  timeDate_4041.110          </span></span>
<span><span class="co">#&gt; [101] SingleCellExperiment_1.26.0 iterators_1.0.14           </span></span>
<span><span class="co">#&gt; [103] hardhat_1.4.1               lava_1.8.1                 </span></span>
<span><span class="co">#&gt; [105] statmod_1.5.0               MLInterfaces_1.84.0        </span></span>
<span><span class="co">#&gt; [107] ipred_0.9-15                nlme_3.1-166               </span></span>
<span><span class="co">#&gt; [109] bit64_4.6.0-1               progress_1.2.3             </span></span>
<span><span class="co">#&gt; [111] filelock_1.0.3              LaplacesDemon_16.1.6       </span></span>
<span><span class="co">#&gt; [113] GenomeInfoDb_1.40.1         bslib_0.9.0                </span></span>
<span><span class="co">#&gt; [115] affyio_1.74.0               irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt; [117] rpart_4.1.23                colorspace_2.1-1           </span></span>
<span><span class="co">#&gt; [119] BiocGenerics_0.50.0         DBI_1.2.3                  </span></span>
<span><span class="co">#&gt; [121] nnet_7.3-19                 tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt; [123] processx_3.8.6              bit_4.6.0                  </span></span>
<span><span class="co">#&gt; [125] compiler_4.4.1              curl_6.4.0                 </span></span>
<span><span class="co">#&gt; [127] rvest_1.0.4                 httr2_1.2.0                </span></span>
<span><span class="co">#&gt; [129] graph_1.82.0                SparseM_1.84-2             </span></span>
<span><span class="co">#&gt; [131] xml2_1.3.8                  desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [133] DelayedArray_0.30.1         plotly_4.11.0              </span></span>
<span><span class="co">#&gt; [135] bookdown_0.43               rtracklayer_1.64.0         </span></span>
<span><span class="co">#&gt; [137] scales_1.4.0                hexbin_1.28.5              </span></span>
<span><span class="co">#&gt; [139] affy_1.82.0                 rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt; [141] stringr_1.5.1               digest_0.6.37              </span></span>
<span><span class="co">#&gt; [143] mixtools_2.0.0.1            rmarkdown_2.29             </span></span>
<span><span class="co">#&gt; [145] XVector_0.44.0              htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [147] pkgconfig_2.0.3             SingleR_2.6.0              </span></span>
<span><span class="co">#&gt; [149] sparseMatrixStats_1.16.0    MatrixGenerics_1.16.0      </span></span>
<span><span class="co">#&gt; [151] dbplyr_2.5.0                fastmap_1.2.0              </span></span>
<span><span class="co">#&gt; [153] htmlwidgets_1.6.4           UCSC.utils_1.0.0           </span></span>
<span><span class="co">#&gt; [155] DelayedMatrixStats_1.26.0   farver_2.1.2               </span></span>
<span><span class="co">#&gt; [157] jquerylib_0.1.4             jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt; [159] mclust_6.1.1                BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt; [161] mzID_1.42.0                 ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">#&gt; [163] BiocSingular_1.20.0         RCurl_1.98-1.17            </span></span>
<span><span class="co">#&gt; [165] magrittr_2.0.3              scuttle_1.14.0             </span></span>
<span><span class="co">#&gt; [167] Formula_1.2-5               GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt; [169] Rcpp_1.1.0                  viridis_0.6.5              </span></span>
<span><span class="co">#&gt; [171] MsCoreUtils_1.16.1          vsn_3.72.0                 </span></span>
<span><span class="co">#&gt; [173] pROC_1.18.5                 stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [175] zlibbioc_1.50.0             MASS_7.3-61                </span></span>
<span><span class="co">#&gt; [177] listenv_0.9.1               parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [179] Biostrings_2.72.1           splines_4.4.1              </span></span>
<span><span class="co">#&gt; [181] hms_1.1.3                   ps_1.9.1                   </span></span>
<span><span class="co">#&gt; [183] igraph_2.1.4                ggpubr_0.6.1               </span></span>
<span><span class="co">#&gt; [185] QFeatures_1.14.2            ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt; [187] reshape2_1.4.4              biomaRt_2.60.1             </span></span>
<span><span class="co">#&gt; [189] stats4_4.4.1                ScaledMatrix_1.12.0        </span></span>
<span><span class="co">#&gt; [191] XML_3.99-0.18               evaluate_1.0.4             </span></span>
<span><span class="co">#&gt; [193] BiocManager_1.30.26         tzdb_0.5.0                 </span></span>
<span><span class="co">#&gt; [195] foreach_1.5.2               tidyr_1.3.1                </span></span>
<span><span class="co">#&gt; [197] purrr_1.1.0                 future_1.67.0              </span></span>
<span><span class="co">#&gt; [199] clue_0.3-66                 bio3d_2.4-5                </span></span>
<span><span class="co">#&gt; [201] ggplot2_3.5.2               rsvd_1.0.5                 </span></span>
<span><span class="co">#&gt; [203] xtable_1.8-4                broom_1.0.8                </span></span>
<span><span class="co">#&gt; [205] restfulr_0.0.16             AnnotationFilter_1.28.0    </span></span>
<span><span class="co">#&gt; [207] easyPubMed_2.13             e1071_1.7-16               </span></span>
<span><span class="co">#&gt; [209] rstatix_0.7.2               later_1.4.2                </span></span>
<span><span class="co">#&gt; [211] class_7.3-22                viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [213] ragg_1.4.0                  tibble_3.3.0               </span></span>
<span><span class="co">#&gt; [215] websocket_1.4.4             memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [217] AnnotationDbi_1.66.0        GenomicAlignments_1.40.0   </span></span>
<span><span class="co">#&gt; [219] IRanges_2.38.1              cluster_2.1.8              </span></span>
<span><span class="co">#&gt; [221] globals_0.18.0              timechange_0.3.0           </span></span>
<span><span class="co">#&gt; [223] caret_7.0-1                 sampling_2.11</span></span></code></pre></div>
=======
<span><span class="co">#&gt;  [69] beachmat_2.20.0             echarts4r_0.4.5.9000       </span></span>
<span><span class="co">#&gt;  [71] annotate_1.82.0             GenomicFeatures_1.56.0     </span></span>
<span><span class="co">#&gt;  [73] chromote_0.5.1              mzR_2.38.0                 </span></span>
<span><span class="co">#&gt;  [75] KEGGREST_1.44.1             pillar_1.11.0              </span></span>
<span><span class="co">#&gt;  [77] knitr_1.50                  GenomicRanges_1.56.2       </span></span>
<span><span class="co">#&gt;  [79] rjson_0.2.23                lpSolve_5.6.23             </span></span>
<span><span class="co">#&gt;  [81] future.apply_1.20.0         codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [83] mgsub_1.7.3                 glue_1.8.0                 </span></span>
<span><span class="co">#&gt;  [85] pcaMethods_1.96.0           data.table_1.17.6          </span></span>
<span><span class="co">#&gt;  [87] MultiAssayExperiment_1.30.3 vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [89] png_0.1-8                   gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [91] kernlab_0.9-33              cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [93] gower_1.0.2                 xfun_0.52                  </span></span>
<span><span class="co">#&gt;  [95] prodlim_2025.04.28          S4Arrays_1.4.1             </span></span>
<span><span class="co">#&gt;  [97] mime_0.13                   coda_0.19-4.1              </span></span>
<span><span class="co">#&gt;  [99] survival_3.8-3              ncdf4_1.24                 </span></span>
<span><span class="co">#&gt; [101] timeDate_4041.110           SingleCellExperiment_1.26.0</span></span>
<span><span class="co">#&gt; [103] iterators_1.0.14            hardhat_1.4.1              </span></span>
<span><span class="co">#&gt; [105] lava_1.8.1                  statmod_1.5.0              </span></span>
<span><span class="co">#&gt; [107] MLInterfaces_1.84.0         ipred_0.9-15               </span></span>
<span><span class="co">#&gt; [109] nlme_3.1-166                bit64_4.6.0-1              </span></span>
<span><span class="co">#&gt; [111] progress_1.2.3              filelock_1.0.3             </span></span>
<span><span class="co">#&gt; [113] LaplacesDemon_16.1.6        GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt; [115] bslib_0.9.0                 affyio_1.74.0              </span></span>
<span><span class="co">#&gt; [117] irlba_2.3.5.1               rpart_4.1.23               </span></span>
<span><span class="co">#&gt; [119] colorspace_2.1-1            BiocGenerics_0.50.0        </span></span>
<span><span class="co">#&gt; [121] DBI_1.2.3                   nnet_7.3-19                </span></span>
<span><span class="co">#&gt; [123] tidyselect_1.2.1            processx_3.8.6             </span></span>
<span><span class="co">#&gt; [125] bit_4.6.0                   compiler_4.4.1             </span></span>
<span><span class="co">#&gt; [127] curl_6.4.0                  rvest_1.0.4                </span></span>
<span><span class="co">#&gt; [129] httr2_1.2.0                 graph_1.82.0               </span></span>
<span><span class="co">#&gt; [131] SparseM_1.84-2              xml2_1.3.8                 </span></span>
<span><span class="co">#&gt; [133] plotly_4.11.0               desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [135] DelayedArray_0.30.1         bookdown_0.43              </span></span>
<span><span class="co">#&gt; [137] rtracklayer_1.64.0          scales_1.4.0               </span></span>
<span><span class="co">#&gt; [139] hexbin_1.28.5               affy_1.82.0                </span></span>
<span><span class="co">#&gt; [141] rappdirs_0.3.3              stringr_1.5.1              </span></span>
<span><span class="co">#&gt; [143] digest_0.6.37               mixtools_2.0.0.1           </span></span>
<span><span class="co">#&gt; [145] rmarkdown_2.29              XVector_0.44.0             </span></span>
<span><span class="co">#&gt; [147] htmltools_0.5.8.1           pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [149] SingleR_2.6.0               sparseMatrixStats_1.16.0   </span></span>
<span><span class="co">#&gt; [151] MatrixGenerics_1.16.0       dbplyr_2.5.0               </span></span>
<span><span class="co">#&gt; [153] fastmap_1.2.0               htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [155] UCSC.utils_1.0.0            shiny_1.11.1               </span></span>
<span><span class="co">#&gt; [157] DelayedMatrixStats_1.26.0   farver_2.1.2               </span></span>
<span><span class="co">#&gt; [159] jquerylib_0.1.4             jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt; [161] mclust_6.1.1                BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt; [163] mzID_1.42.0                 ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">#&gt; [165] BiocSingular_1.20.0         RCurl_1.98-1.17            </span></span>
<span><span class="co">#&gt; [167] magrittr_2.0.3              scuttle_1.14.0             </span></span>
<span><span class="co">#&gt; [169] Formula_1.2-5               GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt; [171] Rcpp_1.0.14                 viridis_0.6.5              </span></span>
<span><span class="co">#&gt; [173] MsCoreUtils_1.16.1          vsn_3.72.0                 </span></span>
<span><span class="co">#&gt; [175] pROC_1.18.5                 stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [177] zlibbioc_1.50.0             MASS_7.3-61                </span></span>
<span><span class="co">#&gt; [179] listenv_0.9.1               parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [181] splines_4.4.1               Biostrings_2.72.1          </span></span>
<span><span class="co">#&gt; [183] hms_1.1.3                   ps_1.9.1                   </span></span>
<span><span class="co">#&gt; [185] igraph_2.1.4                ggpubr_0.6.1               </span></span>
<span><span class="co">#&gt; [187] QFeatures_1.14.2            ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt; [189] reshape2_1.4.4              biomaRt_2.60.1             </span></span>
<span><span class="co">#&gt; [191] stats4_4.4.1                ScaledMatrix_1.12.0        </span></span>
<span><span class="co">#&gt; [193] XML_3.99-0.18               evaluate_1.0.4             </span></span>
<span><span class="co">#&gt; [195] BiocManager_1.30.26         tzdb_0.5.0                 </span></span>
<span><span class="co">#&gt; [197] foreach_1.5.2               httpuv_1.6.16              </span></span>
<span><span class="co">#&gt; [199] tidyr_1.3.1                 purrr_1.0.4                </span></span>
<span><span class="co">#&gt; [201] future_1.58.0               clue_0.3-66                </span></span>
<span><span class="co">#&gt; [203] bio3d_2.4-5                 ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt; [205] rsvd_1.0.5                  broom_1.0.8                </span></span>
<span><span class="co">#&gt; [207] xtable_1.8-4                restfulr_0.0.16            </span></span>
<span><span class="co">#&gt; [209] AnnotationFilter_1.28.0     easyPubMed_2.13            </span></span>
<span><span class="co">#&gt; [211] e1071_1.7-16                rstatix_0.7.2              </span></span>
<span><span class="co">#&gt; [213] later_1.4.2                 class_7.3-22               </span></span>
<span><span class="co">#&gt; [215] viridisLite_0.4.2           ragg_1.4.0                 </span></span>
<span><span class="co">#&gt; [217] tibble_3.3.0                websocket_1.4.4            </span></span>
<span><span class="co">#&gt; [219] memoise_2.0.1               AnnotationDbi_1.66.0       </span></span>
<span><span class="co">#&gt; [221] GenomicAlignments_1.40.0    IRanges_2.38.1             </span></span>
<span><span class="co">#&gt; [223] cluster_2.1.8               globals_0.18.0             </span></span>
<span><span class="co">#&gt; [225] timechange_0.3.0            caret_7.0-1                </span></span>
<span><span class="co">#&gt; [227] sampling_2.11</span></span></code></pre></div>
>>>>>>> main
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-amos2022veupathdb" class="csl-entry">
Amos, Beatrice, Cristina Aurrecoechea, Matthieu Barba, Ana Barreto,
Evelina Y Basenko, Robert Belnap, Ann S Blevins, et al. 2022.
<span>“VEuPathDB: The Eukaryotic Pathogen, Vector and Host
Bioinformatics Resource Center.”</span> <em>Nucleic Acids Research</em>
50 (D1): D898–911.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Rohit Satyam, Alberto Maillo, David Gomez-Cabrero, Arnab Pain.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
