<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to plasmoRUtils • plasmoRUtils</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to plasmoRUtils">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">plasmoRUtils</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Introduction_to_plasmoRUtils.html">Introduction to plasmoRUtils</a></li>
    <li><a class="dropdown-item" href="../articles/Gene_ID_Conversion.html">Accessing component databases of VEuPathDB</a></li>
    <li><a class="dropdown-item" href="../articles/Miscellaneous_function.html">Other useful functions</a></li>
    <li><a class="dropdown-item" href="../articles/Need_for_reanalysis.html">RNASeq: Need for reanalysis?</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="Introduction_to_plasmoRUtils_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="Introduction_to_plasmoRUtils_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="Introduction_to_plasmoRUtils_files/plotly-binding-4.11.0/plotly.js"></script><script src="Introduction_to_plasmoRUtils_files/typedarray-0.1/typedarray.min.js"></script><link href="Introduction_to_plasmoRUtils_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="Introduction_to_plasmoRUtils_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><link href="Introduction_to_plasmoRUtils_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="Introduction_to_plasmoRUtils_files/plotly-main-2.11.1/plotly-latest.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to plasmoRUtils</h1>
                        <h4 data-toc-skip class="author">Rohit
Satyam</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Alberto
Maillo</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">David
Gomez-Cabrero</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Arnab
Pain</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="date">20 July, 2025</h4>
      

      <div class="d-none name"><code>Introduction_to_plasmoRUtils.Rmd</code></div>
    </address>
</address>
</address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The package <code>plasmoRUtils</code> is designed to enable
      users to access various <em>Plasmodium</em> and
      Apicomplexan-related databases through single-line R functions. It
      also provides convenience functions for rapid analysis.</p>
    </div>
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Before downloading the package, install the following
dependencies.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cranpkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'BiocManager'</span>,<span class="st">'randomcoloR'</span>, <span class="st">'janitor'</span>, <span class="st">'readr'</span>, <span class="st">'rlang'</span>, <span class="st">'dplyr'</span>, <span class="st">'ggsci'</span>, <span class="st">'rvest'</span>, <span class="st">'easyPubMed'</span>, <span class="st">'plyr'</span>, <span class="st">'scales'</span>, <span class="st">'ggplot2'</span>, <span class="st">'glue'</span>, <span class="st">'tidyr'</span>, <span class="st">'tibble'</span>, <span class="st">'data.table'</span>, <span class="st">'plotly'</span>, <span class="st">'purrr'</span>, <span class="st">'stringr'</span>, <span class="st">'S4Vectors'</span>, <span class="st">'echarts4r'</span>, <span class="st">'magrittr'</span>, <span class="st">'bio3d'</span>, <span class="st">'httr'</span>, <span class="st">'jsonlite'</span>, <span class="st">'ggpubr'</span>, <span class="st">'gt'</span>, <span class="st">'mgsub'</span>, <span class="st">'reshape2'</span>,<span class="st">'pathfindR'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">cranpkgs</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">biocpkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rmarkdown"</span>,<span class="st">"pRoloc"</span>,<span class="st">"knitr"</span>,<span class="st">"BiocStyle"</span>,<span class="st">"DESeq2"</span>,<span class="st">"styler"</span>,<span class="st">"utils"</span>,<span class="st">"IRanges"</span>,<span class="st">"BiocGenerics"</span>,<span class="st">"rtracklayer"</span>,<span class="st">"scuttle"</span>,<span class="st">"txdbmaker"</span>,<span class="st">"topGO"</span>,<span class="st">"drawProteins"</span>,<span class="st">"GenomicFeatures"</span>,<span class="st">"biomaRt"</span>,<span class="st">"AnnotationForge"</span>,<span class="st">"Biostrings"</span>,<span class="st">"GenomeInfoDb"</span>,<span class="st">"SingleCellExperiment"</span>,<span class="st">"SingleR"</span>,<span class="st">"NOISeq"</span>,<span class="st">"GenomicRanges"</span>,<span class="st">"BSgenome"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">biocpkgs</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>The <em>plasmoRUtils</em> package is available on CRAN and can be
installed as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"plasmoRUtils"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Once installed load the library as</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rohit-satyam.github.io/plasmoRUtils/">plasmoRUtils</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## To re-check if all the dependencies that are required by plasmoRUtils are installed</span></span>
<span><span class="fu"><a href="../reference/install_dependencies.html">install_dependencies</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Using <em>plasmoRUtils</em>, users can fetch data from VEuPathDB and
its 12 component sites databases (VEuPathDBs) and transform it into
formats compatible with other R packages in a straightforward manner.
Data tables (both preconfigured and user-configured) can be downloaded
from VEuPathDBs directly within R/RStudio, thanks to a variety of R
functions and the RESTful API provided by VEuPathDBs.</p>
<p>For databases that lack APIs, we developed database-specific
“searchX” functions (where X represents the database) that utilize the
rvest package for web crawling to retrieve data, which is then
transformed into tables that can be saved and shared. Additionally, we
created a function to enable programmatic access to the MPMP database
for the first time, allowing users to download and share data tables at
their convenience. The package also provides several other data sets
that we reanalyzed using the latest annotations from VEuPathDBs that can
be used by various functions.</p>
<p>Databases covered includes:</p>
<ol style="list-style-type: decimal">
<li><a href="https://www.hitpredict.org/" class="external-link">HitPredict</a></li>
<li><a href="https://bioinfo.icgeb.res.in/PtDB/" class="external-link">ApicoTFDB</a></li>
<li><a href="https://malaria.sbs.ntu.edu.sg/" class="external-link">Malaria.tools</a></li>
<li>Malaria Parasite Metabolic Pathways (<a href="https://mpmp.huji.ac.il/" class="external-link">MPMP</a>) database</li>
<li>Malaria Important Interacting Proteins (<a href="http://www.hpppi.iicb.res.in/pfnet/stage-map.html" class="external-link">MIIP</a>)</li>
<li><a href="https://phenoplasm.org/" class="external-link">Phenoplasm</a></li>
<li><a href="http://genome.lcqb.upmc.fr/plasmobase/" class="external-link">PlasmoBase</a></li>
<li><a href="https://www.uniprot.org/" class="external-link">Uniprot</a></li>
<li>
<a href="https://www.malariacellatlas.org/" class="external-link">Malaria Cell Atlas</a>,
etc. For exhaustive list, see subsections below.</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package and some other useful packages by using</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rohit-satyam.github.io/plasmoRUtils/">plasmoRUtils</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://had.co.nz/plyr" class="external-link">plyr</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="accessing-databases-with-plasmorutils-search-functions">Accessing databases with plasmoRUtils search functions<a class="anchor" aria-label="anchor" href="#accessing-databases-with-plasmorutils-search-functions"></a>
</h2>
<p><em>plasmoRUtils</em> package have several search function to fetch
information from databases. The functions are tabulated below:</p>
<table class="table">
<thead><tr class="header">
<th>Function</th>
<th>Database Access</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong><code><a href="../reference/searchApicoTFdb.html">searchApicoTFdb()</a></code></strong></td>
<td>ApicoTFdb</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/searchGSC.html">searchGSC()</a></code></strong></td>
<td>Google Scholar</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/searchHP.html">searchHP()</a></code></strong></td>
<td>Hit Predict</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/searchIpDb.html">searchIpDb()</a></code></strong></td>
<td>InParanoiDb</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/searchKipho.html">searchKipho()</a></code></strong></td>
<td>KiPho Database</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/searchMT.html">searchMT()</a></code></strong></td>
<td>Malaria Tools</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/searchMidb.html">searchMidb()</a></code></strong></td>
<td>Minor Intron Database</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/searchMiip.html">searchMiip()</a></code></strong></td>
<td>Malaria Important Interacting Proteins</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/searchPM.html">searchPM()</a></code></strong></td>
<td>PubMed</td>
</tr>
<tr class="even">
<td><strong><code><a href="../reference/searchPhPl.html">searchPhPl()</a></code></strong></td>
<td>PhenoPlasm</td>
</tr>
<tr class="odd">
<td><strong><code><a href="../reference/searchTedConsensus.html">searchTedConsensus()</a></code></strong></td>
<td>The Encyclopedia of Domains</td>
</tr>
</tbody>
</table>
<div class="section level3">
<h3 id="searchapidotfdb">
<strong><code>searchApidoTFdb()</code></strong><a class="anchor" aria-label="anchor" href="#searchapidotfdb"></a>
</h3>
<p>This function helps user fetch the all the transcription factors for
a particular apicomplexan of interest from ApicoTFDb<span class="citation">(<a href="#ref-Sardar2019">Sardar et al.
2019</a>)</span>. For ease of usage the organism names have been
abbreviated as follows in the Table below:</p>
<table class="table">
<thead><tr class="header">
<th>Category</th>
<th>Abbreviation</th>
<th>Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Plasmodium Species</strong></td>
<td>pb</td>
<td><em>Plasmodium berghii</em></td>
</tr>
<tr class="even">
<td></td>
<td>pv</td>
<td><em>Plasmodium vivax</em></td>
</tr>
<tr class="odd">
<td></td>
<td>pf</td>
<td><em>Plasmodium falciparum</em></td>
</tr>
<tr class="even">
<td></td>
<td>pk</td>
<td><em>Plasmodium knowlesi</em></td>
</tr>
<tr class="odd">
<td></td>
<td>py</td>
<td><em>Plasmodium yoelii</em></td>
</tr>
<tr class="even">
<td></td>
<td>pc</td>
<td><em>Plasmodium chabaudi</em></td>
</tr>
<tr class="odd">
<td><strong>Other Apicomplexan</strong></td>
<td>tg49</td>
<td><em>Toxoplasma Gondii ME49</em></td>
</tr>
<tr class="even">
<td></td>
<td>tg89</td>
<td><em>Toxoplasma Gondii P89</em></td>
</tr>
<tr class="odd">
<td></td>
<td>cp</td>
<td><em>Cryptosporidium parvum</em></td>
</tr>
<tr class="even">
<td></td>
<td>em</td>
<td><em>Eimeria maxima</em></td>
</tr>
<tr class="odd">
<td></td>
<td>bb</td>
<td><em>Babesia bovis</em></td>
</tr>
<tr class="even">
<td></td>
<td>et</td>
<td><em>Eimeria tenella</em></td>
</tr>
<tr class="odd">
<td></td>
<td>nu</td>
<td><em>Neospora caninum</em></td>
</tr>
<tr class="even">
<td></td>
<td>cy</td>
<td><em>Cyclospora cayetanensis</em></td>
</tr>
</tbody>
</table>
<p>Using the function is relatively easy and can be achieved as</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Searching all plasmodium TFs</span></span>
<span><span class="fu"><a href="../reference/searchApicoTFdb.html">searchApicoTFdb</a></span><span class="op">(</span>org<span class="op">=</span><span class="st">"pf"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     `Protein Length` `Product Description`              `TF- Family`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_1319600 1633             ACDC domain-containing protein, p… AP2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_0604100 1979             AP2 domain transcription factor    AP2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PF3D7_1222400 2558             AP2 domain transcription factor    AP2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PF3D7_1222600 2432             AP2 domain transcription factor A… AP2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PF3D7_1408200 1702             AP2 domain transcription factor A… AP2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PF3D7_1007700 1597             AP2 domain transcription factor A… AP2</span></span>
<span><span class="co">## Searching all cyclospora TFs</span></span>
<span><span class="fu"><a href="../reference/searchApicoTFdb.html">searchApicoTFdb</a></span><span class="op">(</span>org<span class="op">=</span><span class="st">"tg49"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   `Gene ID`     `Product Description`              `Protein Length` `TF- Family`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TGME49_200385 Myb family DNA-binding domain-con… 2258             Myb/SANT    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> TGME49_201220 zinc finger protein                603              BBOX        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> TGME49_201790 FHA domain-containing protein      556              FHA         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> TGME49_202690 DNA-directed RNA polymerase II RP… 250              General-TF  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> TGME49_202840 FHA domain-containing protein      1044             FHA         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> TGME49_202900 zinc finger (CCCH type) motif-con… 1298             Zn-Finger</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="searchgsc">
<strong><code>searchGSC()</code></strong><a class="anchor" aria-label="anchor" href="#searchgsc"></a>
</h3>
<p>Sometimes, it is difficult to keep track of the corpus while you are
working on your gene of interest and you might want to keep up with your
competing groups across the globe. <code><a href="../reference/searchGSC.html">searchGSC()</a></code> function can
help you collect all the necessary literature where your gene ID of
interest has been mentioned and return the results in form of a data
frame.</p>
<p>Since Google Scholar searches are not restricted to the Article
abstracts but extends till supplementary section, this function can be
very helpful to capture articles that mentions your gene ID of interest
and are otherwise missed by normal Google search. Besides, since most of
the pre-print literature is indexed at Google Scholar, you can also find
papers by your competing groups that are yet to be peer-reviewed.</p>
<blockquote>
<p>Note: We would like to warn users that this function is experimental
and have been seen to get your IP blocked temporarily for 24 hrs if used
more than 20 times. For large array of genes, we encourage users to use
more specialized APIs.</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Searching all plasmodium TFs</span></span>
<span><span class="fu"><a href="../reference/searchGSC.html">searchGSC</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0420300"</span>, <span class="st">"PF3D7_0621000"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 14 × 5</span></span></span>
<span><span class="co">#&gt;    GeneID        Title                                       Year  Url   Authors</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PF3D7_0420300 Plasmodium malariae and ovale genomes prov… 2017  http… C Newb…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PF3D7_0420300 Changes in genome organization of parasite… 2018  http… EM Bun…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PF3D7_0420300 Transcriptomics and proteomics reveal two … 2019  http… SE Lin…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PF3D7_0420300 The RNA structurome in the asexual blood s… 2021  http… DR Alv…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PF3D7_0420300 The Transcription Factor PfAP2-O Influence… 2021  http… EFG Cu…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PF3D7_0621000 The roles of plasmepsins IX and X in malar… 2021  http… AS Nas…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PF3D7_0420300 Investigation of Plasmodium falciparum mit… 2022  http… S Dass </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PF3D7_0621000 An Insight to Further Malaria Vaccine Deve… 2023  http… A Berry</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PF3D7_0621000 Coordination of apicoplast transcription i… 2023  http… Y Koba…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PF3D7_0420300 Systematic in vitro evolution in Plasmodiu… 2024  http… MR Lut…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PF3D7_0420300 A Plasmodium falciparum MORC protein compl… 2024  http… MK Sin…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PF3D7_0420300 Transcriptome analysis reveals a de novo D… 2025  http… A Okaf…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> PF3D7_0420300 Genome-wide gene expression profiles throu… 2025  http… G Zang…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> PF3D7_0621000 Advancing Functional Genomics in P. falcip… 2025  http… ST Win…</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="searchhp">
<strong><code>searchHP()</code></strong><a class="anchor" aria-label="anchor" href="#searchhp"></a>
</h3>
<p>This function enables you to search HitPredict<span class="citation">(<a href="#ref-L%C3%B3pez2015">López, Nakai, and Patil
2015</a>)</span> database and procure high-confidence Protein-Protein
interactions(PPI) for your organism of interest. All it requires is a
gene ID and taxon ID. HitPredict database provides PPI data in form of
Uniprot IDs which are not always ideal for apicomplexan biologists.
Therefore, we provide functionality to convert these Uniprot IDs back to
gene IDs by setting <code>uniprotToGID=TRUE</code> . Since the only
apicomplexan in HitPredict is
<em><code>Plasmodium falciparum</code></em> this gene ID mapping
conversion functionality is only limited for Plasmodium. It should be
turned off, when using it for non-apicomplexan organism as shown
below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Single gene query</span></span>
<span><span class="fu"><a href="../reference/searchHP.html">searchHP</a></span><span class="op">(</span><span class="st">"PF3D7_0418300"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Interactor Interaction       Name Experiments        Category Method.Score</span></span>
<span><span class="co">#&gt; 1 A0A5K1K7X4       46953 A0A5K1K7X4           1 High-throughput         0.35</span></span>
<span><span class="co">#&gt; 2     C0H4E0       82953     C0H4E0           1 High-throughput         0.39</span></span>
<span><span class="co">#&gt; 3     C0H4U4       83025     C0H4U4           1 High-throughput         0.39</span></span>
<span><span class="co">#&gt; 4     C0H586       83100     C0H586           1 High-throughput         0.39</span></span>
<span><span class="co">#&gt; 5     C0H5G3       83162     C0H5G3           1 High-throughput         0.39</span></span>
<span><span class="co">#&gt; 6     Q8I398     1211124     Q8I398           1 High-throughput         0.49</span></span>
<span><span class="co">#&gt;   Annotation.Score Interaction.Score Confidence       QueryID ensembl_gene_id</span></span>
<span><span class="co">#&gt; 1             0.16             0.238        Low PF3D7_0418300            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2             0.16             0.251        Low PF3D7_0418300   PF3D7_0515400</span></span>
<span><span class="co">#&gt; 3             0.16             0.251        Low PF3D7_0418300   PF3D7_0813300</span></span>
<span><span class="co">#&gt; 4             0.16             0.251        Low PF3D7_0418300   PF3D7_0933200</span></span>
<span><span class="co">#&gt; 5             0.16             0.251        Low PF3D7_0418300   PF3D7_1341300</span></span>
<span><span class="co">#&gt; 6             0.16             0.282       High PF3D7_0418300   PF3D7_0905100</span></span>
<span></span>
<span><span class="co">## To use it for other organism, turn off uniprotToGID and provide taxid of the organism</span></span>
<span><span class="va">test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchHP.html">searchHP</a></span><span class="op">(</span><span class="st">"BRCA1"</span>,taxid <span class="op">=</span> <span class="st">"9606"</span> , uniprotToGID <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Multiple gene query</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0418300"</span>,<span class="st">"PF3D7_1118500"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="../reference/searchHP.html">searchHP</a></span><span class="op">(</span><span class="va">x</span>,uniprotToGID <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/ldply.html" class="external-link">ldply</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Interaction Interactor       Name Experiments        Category Method.Score</span></span>
<span><span class="co">#&gt; 15       83162     C0H5G3     C0H5G3           1 High-throughput         0.39</span></span>
<span><span class="co">#&gt; 16       46953 A0A5K1K7X4 A0A5K1K7X4           1 High-throughput         0.35</span></span>
<span><span class="co">#&gt; 17     1211135     Q9U0N1     Q9U0N1           1 High-throughput         0.35</span></span>
<span><span class="co">#&gt; 18     1212791     Q8IJG6     Q8IJG6           1 High-throughput         0.49</span></span>
<span><span class="co">#&gt; 19       87015     C6KTD2       SET1           1 High-throughput         0.39</span></span>
<span><span class="co">#&gt; 20     1211131     Q8I1Q4     Q8I1Q4           1 High-throughput         0.49</span></span>
<span><span class="co">#&gt;    Annotation.Score Interaction.Score Confidence       QueryID</span></span>
<span><span class="co">#&gt; 15             0.16             0.251        Low PF3D7_0418300</span></span>
<span><span class="co">#&gt; 16             0.16             0.238        Low PF3D7_0418300</span></span>
<span><span class="co">#&gt; 17             0.16             0.238        Low PF3D7_0418300</span></span>
<span><span class="co">#&gt; 18             0.50             0.494       High PF3D7_1118500</span></span>
<span><span class="co">#&gt; 19             0.50             0.439       High PF3D7_1118500</span></span>
<span><span class="co">#&gt; 20             0.16             0.282       High PF3D7_1118500</span></span>
<span></span>
<span><span class="co">## You can now use toGeneid function which uses PlasmoDB release 68 annotation to</span></span>
<span><span class="co">## map the uniprot IDs back to the gene IDs</span></span>
<span><span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">Interactor</span>,from <span class="op">=</span> <span class="st">"uniprot"</span>,<span class="st">"ensembl"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">res</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UniProt ID(s)"</span> <span class="op">=</span> <span class="st">"Interactor"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 20 × 13</span></span></span>
<span><span class="co">#&gt;    `Gene ID`     `Product Description`     `Gene Name or Symbol` `UniProt ID(s)`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PF3D7_0113000 glutamic acid-rich prote… GARP                  Q9U0N1         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PF3D7_0418300 conserved Plasmodium pro… N/A                   Q8I1Q4         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PF3D7_0515400 conserved protein, unkno… N/A                   C0H4E0         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PF3D7_0526800 conserved Plasmodium pro… N/A                   Q8I3J7         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PF3D7_0532100 early transcribed membra… ETRAMP5               A0A5K1K7X4     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PF3D7_0629700 SET domain protein, puta… SET1                  C6KTD2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PF3D7_0802000 glutamate dehydrogenase,… GDH3                  Q8IAM0         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PF3D7_0813300 NPL domain-containing pr… N/A                   C0H4U4         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PF3D7_0825500 protein KRI1, putative    KRI1                  Q8IB88         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PF3D7_0905100 nucleoporin NUP221, puta… NUP221                Q8I398         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PF3D7_0933200 calcyclin-binding protei… N/A                   C0H586         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PF3D7_1023900 chromodomain-helicase-DN… CHD1                  Q8IJG6         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> PF3D7_1023900 chromodomain-helicase-DN… CHD1                  Q8IJG6         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> PF3D7_1112100 protein kinase, putative  N/A                   Q8IIP2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> PF3D7_1118500 nucleolar protein 56, pu… NOP56                 Q8III3         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> PF3D7_1228600 merozoite surface protei… MSP9                  Q8I5D2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> PF3D7_1302700 ATP-dependent RNA helica… N/A                   Q8IET8         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> PF3D7_1309400 HORMA domain protein, pu… N/A                   Q8IEM0         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> PF3D7_1341300 60S ribosomal protein L1… N/A                   C0H5G3         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> PF3D7_1468100 MORC family protein       MORC                  Q8IKF6         </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9 more variables: Interaction &lt;int&gt;, Name &lt;chr&gt;, Experiments &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Category &lt;chr&gt;, Method.Score &lt;dbl&gt;, Annotation.Score &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Interaction.Score &lt;dbl&gt;, Confidence &lt;chr&gt;, QueryID &lt;chr&gt;</span></span></span></code></pre></div>
<p>Another scenario where users might be interested in setting
<code>uniportToGID=FALSE</code> might be when they are querying
thousands of IDs. Since ID conversion is carried out using <em><a href="https://bioconductor.org/packages/3.19/biomaRt" class="external-link">biomaRt</a></em>,
it might be redundant to convert same Uniprot ID multiple times if it
has multiple interacting partners.</p>
<p>For convenience, we therefore provide another function
<code><a href="../reference/toGeneid.html">toGeneid()</a></code> which will quickly converts the Uniprot IDs back
to Ensembl IDs.</p>
</div>
<div class="section level3">
<h3 id="searchipdb">
<strong><code>searchIpDb()</code></strong><a class="anchor" aria-label="anchor" href="#searchipdb"></a>
</h3>
<p>This function enables you to search <a href="https://inparanoidb.sbc.su.se/" class="external-link">InParanoiDB 9</a> <span class="citation">(<a href="#ref-Persson2023">Persson and Sonnhammer
2023</a>)</span> database and procure high-confidence orthologs for your
organism of interest. The input required is a character vector of gene
IDs.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0807800"</span>, <span class="st">"PF3D7_1023900"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/searchIpDb.html">searchIpDb</a></span><span class="op">(</span><span class="va">gids</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; success Q8IAR6 </span></span>
<span><span class="co">#&gt; success Q8IJG6</span></span>
<span><span class="co">#&gt;   Group ID                 Species    Protein       Gene Name</span></span>
<span><span class="co">#&gt; 1      826       Perkinsus marinus     C5LD32 Pmar_PMAR029539</span></span>
<span><span class="co">#&gt; 2      826       Perkinsus marinus     C5L7W9 Pmar_PMAR009653</span></span>
<span><span class="co">#&gt; 3      826   Plasmodium falciparum     Q8IAR6   PF3D7_0807800</span></span>
<span><span class="co">#&gt; 4     1814   Plasmodium falciparum     Q8IAR6   PF3D7_0807800</span></span>
<span><span class="co">#&gt; 5     1814        Plasmodium vivax     A5KAC7      PVX_088150</span></span>
<span><span class="co">#&gt; 6      706 Cyclospora cayetanensis A0A1D3D0U4       cyc_02495</span></span>
<span><span class="co">#&gt;   Bitscore info_outline Inparalog Score info_outline Seed Score info_outline</span></span>
<span><span class="co">#&gt; 1                   126                        1.000                     1.0</span></span>
<span><span class="co">#&gt; 2                   126                        0.397                       -</span></span>
<span><span class="co">#&gt; 3                   126                        1.000                     1.0</span></span>
<span><span class="co">#&gt; 4                   515                        1.000                       1</span></span>
<span><span class="co">#&gt; 5                   515                        1.000                       1</span></span>
<span><span class="co">#&gt; 6                   179                        1.000                       1</span></span>
<span><span class="co">#&gt;                                                Description       queryid</span></span>
<span><span class="co">#&gt; 1   26S Proteasome Non-Atpase Regulatory Subunit, Putative PF3D7_0807800</span></span>
<span><span class="co">#&gt; 2   26S Proteasome Non-Atpase Regulatory Subunit, Putative PF3D7_0807800</span></span>
<span><span class="co">#&gt; 3        26S Proteasome Regulatory Subunit Rpn10, Putative PF3D7_0807800</span></span>
<span><span class="co">#&gt; 4        26S Proteasome Regulatory Subunit Rpn10, Putative PF3D7_0807800</span></span>
<span><span class="co">#&gt; 5 26S Proteasome Non-Atpase Regulatory Subunit 4, Putative PF3D7_0807800</span></span>
<span><span class="co">#&gt; 6               Ubiquitin Interaction Motif Family Protein PF3D7_0807800</span></span></code></pre></div>
<p>You might see some of the Uniprot ID failing such as
<code>Q2KNU4</code> and <code>Q2KNU5</code> and their respective URLs.
These Uniprot IDs are missing from the InParanoiDB 9 database.</p>
</div>
<div class="section level3">
<h3 id="searchkipho">
<strong><code>searchKipho()</code></strong><a class="anchor" aria-label="anchor" href="#searchkipho"></a>
</h3>
<p>This functions let you fetch the Malaria Parasite Kinome-Phosphatome
Resource (KiPho) database <span class="citation">(<a href="#ref-Pandey2017">Pandey, Kumar, and Gupta 2017</a>)</span> without
leaving R. The organism in KiPho includes (see below):</p>
<table class="table">
<thead><tr class="header">
<th>Abbreviation</th>
<th>Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pb</td>
<td><em>Plasmodium berghii</em></td>
</tr>
<tr class="even">
<td>pv</td>
<td><em>Plasmodium vivax</em></td>
</tr>
<tr class="odd">
<td>pf</td>
<td><em>Plasmodium falciparum</em></td>
</tr>
<tr class="even">
<td>pc</td>
<td><em>Plasmodium chabaudi</em></td>
</tr>
</tbody>
</table>
<p>Beside the organism, user needs to specify <code>type="kinase"</code>
to fetch the Kinome and <code>"type=phosphatase"</code> to fetch
Phosphatome.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/searchKipho.html">searchKipho</a></span><span class="op">(</span>org<span class="op">=</span><span class="st">"pf"</span>,type <span class="op">=</span> <span class="st">"kinase"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 148 × 7</span></span></span>
<span><span class="co">#&gt;    `Gene ID`     `Previous ID(s)`      `Product Description`    `Protein Length`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PF3D7_0102600 <span style="color: #949494;">"</span>PFA0130c MAL1P1.17<span style="color: #949494;">"</span>  serine/threonine protei…              630</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PF3D7_0103700 <span style="color: #949494;">"</span>PFA0185w MAL1P1.23<span style="color: #949494;">"</span>  L-seryl-tRNA(Sec) kinas…              535</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PF3D7_0107600 <span style="color: #949494;">"</span>PFA0380w\tMAL1P2.04<span style="color: #949494;">"</span> serine/threonine protei…             <span style="text-decoration: underline;">1</span>595</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PF3D7_0110600 <span style="color: #949494;">"</span>PFA0515w\tMAL1P2.32<span style="color: #949494;">"</span> phosphatidylinositol-4-…             <span style="text-decoration: underline;">1</span>710</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PF3D7_0110900 <span style="color: #949494;">"</span>PFA0530c\tMAL1P2.35<span style="color: #949494;">"</span> adenylate kinase-like p…              186</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PF3D7_0111500 <span style="color: #949494;">"</span>PFA0555c\tMAL1P2.40<span style="color: #949494;">"</span> UMP-CMP kinase, putative              371</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PF3D7_0203100 <span style="color: #949494;">"</span>PFB0150c\tPF02_0030<span style="color: #949494;">"</span> protein kinase, putative             <span style="text-decoration: underline;">2</span>485</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PF3D7_0211700 <span style="color: #949494;">"</span>PFB0520w\tPF02_0109<span style="color: #949494;">"</span> tyrosine kinase-like pr…             <span style="text-decoration: underline;">1</span>233</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PF3D7_0213400 <span style="color: #949494;">"</span>PFB0605w\tPF02_0125<span style="color: #949494;">"</span> protein kinase 7 (PK7)                343</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PF3D7_0214600 <span style="color: #949494;">"</span>PFB0665w\tPF02_0137<span style="color: #949494;">"</span> serine/threonine protei…             <span style="text-decoration: underline;">1</span>714</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 138 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: `Conserved Protein Domain Family(Accession No)` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Conserved Protein Domain Family(Name)` &lt;chr&gt;, `Ortholog Group` &lt;chr&gt;</span></span></span>
<span><span class="fu"><a href="../reference/searchKipho.html">searchKipho</a></span><span class="op">(</span>org<span class="op">=</span><span class="st">"pf"</span>,type <span class="op">=</span> <span class="st">"phosphatase"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 70 × 7</span></span></span>
<span><span class="co">#&gt;    `Gene ID`     `Previous ID(s)`      `Product Description`    `Protein Length`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PF3D7_0107200 <span style="color: #949494;">"</span>PFA0350w\tMAL1P1.64<span style="color: #949494;">"</span> carbon catabolite repre…              337</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PF3D7_0107800 <span style="color: #949494;">"</span>PFA0390w<span style="color: #949494;">"</span>            double-strand break rep…             <span style="text-decoration: underline;">1</span>233</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PF3D7_0303200 <span style="color: #949494;">"</span>PFC0150w<span style="color: #949494;">"</span>            HAD superfamily protein…             <span style="text-decoration: underline;">1</span>162</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PF3D7_0305600 <span style="color: #949494;">"</span>PFC0250c<span style="color: #949494;">"</span>            AP endonuclease (DNA-[a…              617</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PF3D7_0309000 <span style="color: #949494;">"</span>PFC0380w<span style="color: #949494;">"</span>            dual specificity protei…              575</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PF3D7_0310300 <span style="color: #949494;">"</span>PFC0430w<span style="color: #949494;">"</span>            phosphoglycerate mutase…             <span style="text-decoration: underline;">1</span>165</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PF3D7_0314400 <span style="color: #949494;">"</span>PFC0595c<span style="color: #949494;">"</span>            serine/threonine protei…              308</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PF3D7_0319200 <span style="color: #949494;">"</span>PFC0850c<span style="color: #949494;">"</span>            endonuclease/exonucleas…              906</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PF3D7_0322100 <span style="color: #949494;">"</span>PFC0980c<span style="color: #949494;">"</span>            RNA triphosphatase (Prt…              591</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PF3D7_0410300 <span style="color: #949494;">"</span>PFD0505c\tPFD0510c<span style="color: #949494;">"</span>  protein phosphatase PPM…              906</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 60 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: `Conserved Protein Domain Family(Accession_No)` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Conserved Protein Domain Family(Name)` &lt;chr&gt;, `Ortholog Group` &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="searchmt">
<strong><code>searchMT()</code></strong><a class="anchor" aria-label="anchor" href="#searchmt"></a>
</h3>
<p>This function enables you to find the Condition Specific and Tissue
Specific expression of gene of interest in two organisms: <em>Plasmodium
falciparum</em> and <em>Plasmodium berghi.</em></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PBANKA_0100600"</span>, <span class="st">"PBANKA_0102900"</span>, <span class="st">"PF3D7_0102900"</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchMT.html">searchMT</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="va">geneID</span><span class="op">)</span></span>
<span><span class="va">res</span></span>
<span></span>
<span><span class="co"># To get overview of stages your genes of interest are highly expressed in. Commented here as the html plot disrupts the HTML vignette rendering.</span></span>
<span><span class="co"># res %&gt;% easyPie()</span></span></code></pre></div>
<p>You can also feed the output of <code><a href="../reference/searchMT.html">searchMT()</a></code> to a
companion function to quickly get a sense of the stages in which your
genes of interests are highly expressed in. Another convenience function
for malaria.tools database is <code><a href="../reference/plotAllCondition.html">plotAllCondition()</a></code> function.
This let you create publication ready plots of TPM normalized expression
values across multiple stages of parasite using bulk-rnaseq data from
malaria.tools. These plots are similar to what you see in the database
itself.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># TPM plot (non-interactive)</span></span>
<span><span class="fu"><a href="../reference/plotAllCondition.html">plotAllCondition</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="st">"PBANKA_0100600"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Introduction_to_plasmoRUtils_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotAllCondition.html">plotAllCondition</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="st">"PBANKA_0100600"</span>,plotify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co">## interactive</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"data":[{"orientation":"v","width":[0.70000000000000018,0.69999999999999973,0.70000000000000007,0.70000000000000018],"base":[0,0,0,0],"x":[2,4,1,3],"y":[224.51766666666668,228.07049999999998,155.05500000000001,130.93100000000001],"text":["condition: 10 hpi, ab libitum: SRP059210<br />mean:  224.517667","condition: 10 hpi, diet restriction: SRP059210<br />mean:  228.070500","condition: 10 hpi, ab libitum, kin KO: SRP059210<br />mean:  155.055000","condition: 10 hpi, diet restriction, kin KO: SRP059210<br />mean:  130.931000"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(188,60,41,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"10","legendgroup":"10","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.69999999999999929,0.69999999999999929],"base":[0,0],"x":[6,5],"y":[6.9257299999999997,9.4709500000000002],"text":["condition: 16 hpi, WT: ERP004868<br />mean:    6.925730","condition: 16 hpi, CL: ERP004868<br />mean:    9.470950"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(0,114,181,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"16","legendgroup":"16","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.69999999999999929,0.69999999999999929],"base":[0,0],"x":[8,7],"y":[460.66033333333331,403.18299999999999],"text":["condition: Asexual: SRP099925<br />mean:  460.660333","condition: Asexual, PbSR-MG KO: SRP109709<br />mean:  403.183000"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(225,135,39,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Asexual","legendgroup":"Asexual","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.70000000000000284,0.70000000000000284,0.70000000000000284,0.69999999999999929,0.69999999999999929,0.69999999999999929,0.69999999999999929,0.69999999999999929,0.70000000000000107,0.69999999999999929,0.69999999999999929],"base":[0,0,0,0,0,0,0,0,0,0,0],"x":[17,19,18,12,10,13,14,15,16,11,9],"y":[81.092650000000006,112.122,179.88726666666665,530.976,151.57262500000002,113.68787500000001,149.86066666666667,176.55213333333333,201.19133333333332,485.38366666666667,27.065300000000001],"text":["condition: Gametocyte, WT: SRP073801<br />mean:   81.092650","condition: Gametocyte: SRP027529<br />mean:  112.122000","condition: Gametocyte: ERP004598<br />mean:  179.887267","condition: Gametocyte (male): SRP099925<br />mean:  530.976000","condition: Gametocyte (female): SRP099925<br />mean:  151.572625","condition: Gametocyte ap2-o KO: SRP073801<br />mean:  113.687875","condition: Gametocyte ap2-o2 KO: SRP073801<br />mean:  149.860667","condition: Gametocyte ap2-o3 KO: SRP073801<br />mean:  176.552133","condition: Gametocyte, ppm5 KO: ERP004598<br />mean:  201.191333","condition: Gametocyte (male), PbSR-MG KO: SRP109709<br />mean:  485.383667","condition: Gametocyte (female), PbSR-MG KO: SRP109709<br />mean:   27.065300"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(32,133,78,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Gametocyte","legendgroup":"Gametocyte","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284],"base":[0,0,0,0,0,0],"x":[25,24,20,21,22,23],"y":[100.18971666666667,3.33487,133.22649999999999,218.68774999999999,201.28733333333332,2.6294240000000002],"text":["condition: Ookinete, WT: SRP073801<br />mean:  100.189717","condition: Ookinete, WT: ERP004740<br />mean:    3.334870","condition: Ookinete ap2-o KO: SRP073801<br />mean:  133.226500","condition: Ookinete ap2-o2 KO: SRP073801<br />mean:  218.687750","condition: Ookinete ap2-o3 KO: SRP073801<br />mean:  201.287333","condition: Ookinete, mutant: ERP004740<br />mean:    2.629424"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(120,118,177,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Ookinete","legendgroup":"Ookinete","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284,0.70000000000000284],"base":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"x":[40,41,26,42,27,28,29,30,31,32,33,34,35,36,37,38,39],"y":[278.00299999999999,543.78749166666671,2477.52,1371.7592500000001,298.85351666666668,336.60766666666666,805.86933333333332,386.68023333333332,310.40750000000003,317.75266666666664,867.81899999999996,533.69816666666668,852.25999999999999,750.8266666666666,825.29383333333328,1612.1133333333332,1194.3885],"text":["condition: Schizont, predicted: ERP021229<br />mean:  278.003000","condition: Schizont, WT: SRP073801<br />mean:  543.787492","condition: Schizont, 22 hpi: SRP027529<br />mean: 2477.520000","condition: Schizont: ERP004598<br />mean: 1371.759250","condition: Schizont, ap2-g KO: SRP073801<br />mean:  298.853517","condition: Schizont, ap2-g2 KO: SRP073801<br />mean:  336.607667","condition: Schizont, ap2-l KO: SRP073801<br />mean:  805.869333","condition: Schizont, ap2-o KO: SRP073801<br />mean:  386.680233","condition: Schizont, ap2-o2 KO: SRP073801<br />mean:  310.407500","condition: Schizont, ap2-o3 KO: SRP073801<br />mean:  317.752667","condition: Schizont, ap2-o4 KO: SRP073801<br />mean:  867.819000","condition: Schizont, ap2-sp KO: SRP073801<br />mean:  533.698167","condition: Schizont, ap2-sp2 KO: SRP073801<br />mean:  852.260000","condition: Schizont, ap2-sp3 KO: SRP073801<br />mean:  750.826667","condition: Schizont, PBANKA_131970 KO: SRP073801<br />mean:  825.293833","condition: Schizont, ppm2 KO: ERP004598<br />mean: 1612.113333","condition: Schizont, ppm5 KO: ERP004598<br />mean: 1194.388500"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(111,153,173,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Schizont","legendgroup":"Schizont","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":0.70000000000000284,"base":0,"x":[43],"y":[317.93099999999998],"text":"condition: Trophozoite, 16 hpi: SRP027529<br />mean:  317.931000","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(255,220,145,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Trophozoite","legendgroup":"Trophozoite","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2,4,1,3],"y":[206.96700000000001,218.048,155.05500000000001,130.93100000000001],"text":["condition: 10 hpi, ab libitum: SRP059210<br />min:  206.967000","condition: 10 hpi, diet restriction: SRP059210<br />min:  218.048000","condition: 10 hpi, ab libitum, kin KO: SRP059210<br />min:  155.055000","condition: 10 hpi, diet restriction, kin KO: SRP059210<br />min:  130.931000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(188,60,41,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"10","legendgroup":"10","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6,5],"y":[5.1310200000000004,2.6703999999999999],"text":["condition: 16 hpi, WT: ERP004868<br />min:    5.131020","condition: 16 hpi, CL: ERP004868<br />min:    2.670400"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,181,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"16","legendgroup":"16","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[8,7],"y":[396.04199999999997,355.452],"text":["condition: Asexual: SRP099925<br />min:  396.042000","condition: Asexual, PbSR-MG KO: SRP109709<br />min:  355.452000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(225,135,39,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Asexual","legendgroup":"Asexual","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[17,19,18,12,10,13,14,15,16,11,9],"y":[49.604500000000002,112.122,61.708799999999997,491.12200000000001,26.485299999999999,96.850499999999997,107.39400000000001,79.417400000000001,115.861,464.988,19.726800000000001],"text":["condition: Gametocyte, WT: SRP073801<br />min:   49.604500","condition: Gametocyte: SRP027529<br />min:  112.122000","condition: Gametocyte: ERP004598<br />min:   61.708800","condition: Gametocyte (male): SRP099925<br />min:  491.122000","condition: Gametocyte (female): SRP099925<br />min:   26.485300","condition: Gametocyte ap2-o KO: SRP073801<br />min:   96.850500","condition: Gametocyte ap2-o2 KO: SRP073801<br />min:  107.394000","condition: Gametocyte ap2-o3 KO: SRP073801<br />min:   79.417400","condition: Gametocyte, ppm5 KO: ERP004598<br />min:  115.861000","condition: Gametocyte (male), PbSR-MG KO: SRP109709<br />min:  464.988000","condition: Gametocyte (female), PbSR-MG KO: SRP109709<br />min:   19.726800"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(32,133,78,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Gametocyte","legendgroup":"Gametocyte","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[25,24,20,21,22,23],"y":[52.692799999999998,1.3665099999999999,100.22499999999999,168.34700000000001,118.08199999999999,0.85884799999999994],"text":["condition: Ookinete, WT: SRP073801<br />min:   52.692800","condition: Ookinete, WT: ERP004740<br />min:    1.366510","condition: Ookinete ap2-o KO: SRP073801<br />min:  100.225000","condition: Ookinete ap2-o2 KO: SRP073801<br />min:  168.347000","condition: Ookinete ap2-o3 KO: SRP073801<br />min:  118.082000","condition: Ookinete, mutant: ERP004740<br />min:    0.858848"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(120,118,177,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Ookinete","legendgroup":"Ookinete","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[40,41,26,42,27,28,29,30,31,32,33,34,35,36,37,38,39],"y":[278.00299999999999,82.893900000000002,2477.52,996.35699999999997,98.079700000000003,105.753,741.60500000000002,61.130099999999999,167.607,117.51300000000001,684.63,493.779,742.98900000000003,618.774,605.92200000000003,1309.4200000000001,821.43399999999997],"text":["condition: Schizont, predicted: ERP021229<br />min:  278.003000","condition: Schizont, WT: SRP073801<br />min:   82.893900","condition: Schizont, 22 hpi: SRP027529<br />min: 2477.520000","condition: Schizont: ERP004598<br />min:  996.357000","condition: Schizont, ap2-g KO: SRP073801<br />min:   98.079700","condition: Schizont, ap2-g2 KO: SRP073801<br />min:  105.753000","condition: Schizont, ap2-l KO: SRP073801<br />min:  741.605000","condition: Schizont, ap2-o KO: SRP073801<br />min:   61.130100","condition: Schizont, ap2-o2 KO: SRP073801<br />min:  167.607000","condition: Schizont, ap2-o3 KO: SRP073801<br />min:  117.513000","condition: Schizont, ap2-o4 KO: SRP073801<br />min:  684.630000","condition: Schizont, ap2-sp KO: SRP073801<br />min:  493.779000","condition: Schizont, ap2-sp2 KO: SRP073801<br />min:  742.989000","condition: Schizont, ap2-sp3 KO: SRP073801<br />min:  618.774000","condition: Schizont, PBANKA_131970 KO: SRP073801<br />min:  605.922000","condition: Schizont, ppm2 KO: ERP004598<br />min: 1309.420000","condition: Schizont, ppm5 KO: ERP004598<br />min:  821.434000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(111,153,173,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Schizont","legendgroup":"Schizont","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[43],"y":[317.93099999999998],"text":"condition: Trophozoite, 16 hpi: SRP027529<br />min:  317.931000","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,220,145,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Trophozoite","legendgroup":"Trophozoite","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2,4,1,3],"y":[236.91200000000001,238.09299999999999,155.05500000000001,130.93100000000001],"text":["condition: 10 hpi, ab libitum: SRP059210<br />max:  236.91200","condition: 10 hpi, diet restriction: SRP059210<br />max:  238.09300","condition: 10 hpi, ab libitum, kin KO: SRP059210<br />max:  155.05500","condition: 10 hpi, diet restriction, kin KO: SRP059210<br />max:  130.93100"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(188,60,41,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"10","legendgroup":"10","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6,5],"y":[8.72044,16.2715],"text":["condition: 16 hpi, WT: ERP004868<br />max:    8.72044","condition: 16 hpi, CL: ERP004868<br />max:   16.27150"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,181,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"16","legendgroup":"16","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[8,7],"y":[551.80799999999999,442.661],"text":["condition: Asexual: SRP099925<br />max:  551.80800","condition: Asexual, PbSR-MG KO: SRP109709<br />max:  442.66100"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(225,135,39,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Asexual","legendgroup":"Asexual","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[17,19,18,12,10,13,14,15,16,11,9],"y":[106.816,112.122,295.51799999999997,570.83000000000004,479.55700000000002,119.42,192.036,261.30599999999998,255.499,496.01499999999999,37.720700000000001],"text":["condition: Gametocyte, WT: SRP073801<br />max:  106.81600","condition: Gametocyte: SRP027529<br />max:  112.12200","condition: Gametocyte: ERP004598<br />max:  295.51800","condition: Gametocyte (male): SRP099925<br />max:  570.83000","condition: Gametocyte (female): SRP099925<br />max:  479.55700","condition: Gametocyte ap2-o KO: SRP073801<br />max:  119.42000","condition: Gametocyte ap2-o2 KO: SRP073801<br />max:  192.03600","condition: Gametocyte ap2-o3 KO: SRP073801<br />max:  261.30600","condition: Gametocyte, ppm5 KO: ERP004598<br />max:  255.49900","condition: Gametocyte (male), PbSR-MG KO: SRP109709<br />max:  496.01500","condition: Gametocyte (female), PbSR-MG KO: SRP109709<br />max:   37.72070"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(32,133,78,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Gametocyte","legendgroup":"Gametocyte","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[25,24,20,21,22,23],"y":[123.65000000000001,5.3032300000000001,192.05500000000001,256.44499999999999,260.58100000000002,4.4000000000000004],"text":["condition: Ookinete, WT: SRP073801<br />max:  123.65000","condition: Ookinete, WT: ERP004740<br />max:    5.30323","condition: Ookinete ap2-o KO: SRP073801<br />max:  192.05500","condition: Ookinete ap2-o2 KO: SRP073801<br />max:  256.44500","condition: Ookinete ap2-o3 KO: SRP073801<br />max:  260.58100","condition: Ookinete, mutant: ERP004740<br />max:    4.40000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(120,118,177,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Ookinete","legendgroup":"Ookinete","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[40,41,26,42,27,28,29,30,31,32,33,34,35,36,37,38,39],"y":[278.00299999999999,1130.3299999999999,2477.52,1883.3499999999999,482.97899999999998,469.81799999999998,900.36500000000001,634.24099999999999,432.31400000000002,690.33100000000002,1018.54,600.30100000000004,1052.23,918.66099999999994,986.77099999999996,1996.8599999999999,1642.28],"text":["condition: Schizont, predicted: ERP021229<br />max:  278.00300","condition: Schizont, WT: SRP073801<br />max: 1130.33000","condition: Schizont, 22 hpi: SRP027529<br />max: 2477.52000","condition: Schizont: ERP004598<br />max: 1883.35000","condition: Schizont, ap2-g KO: SRP073801<br />max:  482.97900","condition: Schizont, ap2-g2 KO: SRP073801<br />max:  469.81800","condition: Schizont, ap2-l KO: SRP073801<br />max:  900.36500","condition: Schizont, ap2-o KO: SRP073801<br />max:  634.24100","condition: Schizont, ap2-o2 KO: SRP073801<br />max:  432.31400","condition: Schizont, ap2-o3 KO: SRP073801<br />max:  690.33100","condition: Schizont, ap2-o4 KO: SRP073801<br />max: 1018.54000","condition: Schizont, ap2-sp KO: SRP073801<br />max:  600.30100","condition: Schizont, ap2-sp2 KO: SRP073801<br />max: 1052.23000","condition: Schizont, ap2-sp3 KO: SRP073801<br />max:  918.66100","condition: Schizont, PBANKA_131970 KO: SRP073801<br />max:  986.77100","condition: Schizont, ppm2 KO: ERP004598<br />max: 1996.86000","condition: Schizont, ppm5 KO: ERP004598<br />max: 1642.28000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(111,153,173,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Schizont","legendgroup":"Schizont","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[43],"y":[317.93099999999998],"text":"condition: Trophozoite, 16 hpi: SRP027529<br />max:  317.93100","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,220,145,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Trophozoite","legendgroup":"Trophozoite","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":27.824636418824401,"r":7.3059360730593621,"b":266.80180536859615,"l":48.949771689497723},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,43.600000000000001],"tickmode":"array","ticktext":["10 hpi, ab libitum, kin KO: SRP059210","10 hpi, ab libitum: SRP059210","10 hpi, diet restriction, kin KO: SRP059210","10 hpi, diet restriction: SRP059210","16 hpi, CL: ERP004868","16 hpi, WT: ERP004868","Asexual, PbSR-MG KO: SRP109709","Asexual: SRP099925","Gametocyte (female), PbSR-MG KO: SRP109709","Gametocyte (female): SRP099925","Gametocyte (male), PbSR-MG KO: SRP109709","Gametocyte (male): SRP099925","Gametocyte ap2-o KO: SRP073801","Gametocyte ap2-o2 KO: SRP073801","Gametocyte ap2-o3 KO: SRP073801","Gametocyte, ppm5 KO: ERP004598","Gametocyte, WT: SRP073801","Gametocyte: ERP004598","Gametocyte: SRP027529","Ookinete ap2-o KO: SRP073801","Ookinete ap2-o2 KO: SRP073801","Ookinete ap2-o3 KO: SRP073801","Ookinete, mutant: ERP004740","Ookinete, WT: ERP004740","Ookinete, WT: SRP073801","Schizont, 22 hpi: SRP027529","Schizont, ap2-g KO: SRP073801","Schizont, ap2-g2 KO: SRP073801","Schizont, ap2-l KO: SRP073801","Schizont, ap2-o KO: SRP073801","Schizont, ap2-o2 KO: SRP073801","Schizont, ap2-o3 KO: SRP073801","Schizont, ap2-o4 KO: SRP073801","Schizont, ap2-sp KO: SRP073801","Schizont, ap2-sp2 KO: SRP073801","Schizont, ap2-sp3 KO: SRP073801","Schizont, PBANKA_131970 KO: SRP073801","Schizont, ppm2 KO: ERP004598","Schizont, ppm5 KO: ERP004598","Schizont, predicted: ERP021229","Schizont, WT: SRP073801","Schizont: ERP004598","Trophozoite, 16 hpi: SRP027529"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,14.999999999999998,16,17,18,19,20,21,22,23,24,25,26,27,28,28.999999999999996,29.999999999999996,31.000000000000004,32.000000000000007,33,34,35,36,37,38,39,40,41,42,43],"categoryorder":"array","categoryarray":["10 hpi, ab libitum, kin KO: SRP059210","10 hpi, ab libitum: SRP059210","10 hpi, diet restriction, kin KO: SRP059210","10 hpi, diet restriction: SRP059210","16 hpi, CL: ERP004868","16 hpi, WT: ERP004868","Asexual, PbSR-MG KO: SRP109709","Asexual: SRP099925","Gametocyte (female), PbSR-MG KO: SRP109709","Gametocyte (female): SRP099925","Gametocyte (male), PbSR-MG KO: SRP109709","Gametocyte (male): SRP099925","Gametocyte ap2-o KO: SRP073801","Gametocyte ap2-o2 KO: SRP073801","Gametocyte ap2-o3 KO: SRP073801","Gametocyte, ppm5 KO: ERP004598","Gametocyte, WT: SRP073801","Gametocyte: ERP004598","Gametocyte: SRP027529","Ookinete ap2-o KO: SRP073801","Ookinete ap2-o2 KO: SRP073801","Ookinete ap2-o3 KO: SRP073801","Ookinete, mutant: ERP004740","Ookinete, WT: ERP004740","Ookinete, WT: SRP073801","Schizont, 22 hpi: SRP027529","Schizont, ap2-g KO: SRP073801","Schizont, ap2-g2 KO: SRP073801","Schizont, ap2-l KO: SRP073801","Schizont, ap2-o KO: SRP073801","Schizont, ap2-o2 KO: SRP073801","Schizont, ap2-o3 KO: SRP073801","Schizont, ap2-o4 KO: SRP073801","Schizont, ap2-sp KO: SRP073801","Schizont, ap2-sp2 KO: SRP073801","Schizont, ap2-sp3 KO: SRP073801","Schizont, PBANKA_131970 KO: SRP073801","Schizont, ppm2 KO: ERP004598","Schizont, ppm5 KO: ERP004598","Schizont, predicted: ERP021229","Schizont, WT: SRP073801","Schizont: ERP004598","Trophozoite, 16 hpi: SRP027529"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-90,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-123.876,2601.3960000000002],"tickmode":"array","ticktext":["0","500","1000","1500","2000","2500"],"tickvals":[0,500,1000,1500,2000.0000000000002,2500],"categoryorder":"array","categoryarray":["0","500","1000","1500","2000","2500"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"TPM","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","layer":"below","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"31c83dff708c":{"x":{},"y":{},"fill":{},"type":"bar"},"31c87d3656b7":{"x":{},"y":{},"fill":{}},"31c8466852aa":{"x":{},"y":{},"fill":{}}},"cur_data":"31c83dff708c","visdat":{"31c83dff708c":["function (y) ","x"],"31c87d3656b7":["function (y) ","x"],"31c8466852aa":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## To get the data used for making above plot use returnData argument</span></span>
<span><span class="fu"><a href="../reference/plotAllCondition.html">plotAllCondition</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="st">"PBANKA_0100600"</span>,returnData <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                     condition     mean     min     max   group</span></span>
<span><span class="co">#&gt; 1                          Asexual: SRP099925 460.6603 396.042 551.808 Asexual</span></span>
<span><span class="co">#&gt; 2              Asexual, PbSR-MG KO: SRP109709 403.1830 355.452 442.661 Asexual</span></span>
<span><span class="co">#&gt; 3               10 hpi, ab libitum: SRP059210 224.5177 206.967 236.912      10</span></span>
<span><span class="co">#&gt; 4         10 hpi, diet restriction: SRP059210 228.0705 218.048 238.093      10</span></span>
<span><span class="co">#&gt; 5       10 hpi, ab libitum, kin KO: SRP059210 155.0550 155.055 155.055      10</span></span>
<span><span class="co">#&gt; 6 10 hpi, diet restriction, kin KO: SRP059210 130.9310 130.931 130.931      10</span></span></code></pre></div>
<p>Users can also plot stage specific average TPMs as well similar to
the plots rendered in malaria.tools using
<code><a href="../reference/plotStageSpecific.html">plotStageSpecific()</a></code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotStageSpecific.html">plotStageSpecific</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="st">"PBANKA_0100600"</span>,plotify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item" id="htmlwidget-e5c8c404fe174e4c81bd" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-e5c8c404fe174e4c81bd">{"x":{"data":[{"orientation":"v","width":0.70000000000000007,"base":0,"x":[1],"y":[317.93099999999998],"text":"condition: Trophozoite<br />mean:  317.9310<br />condition: Trophozoite","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(188,60,41,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Trophozoite","legendgroup":"Trophozoite","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":0.70000000000000018,"base":0,"x":[2],"y":[1167.7674354166668],"text":"condition: Schizont<br />mean: 1167.7674<br />condition: Schizont","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(0,114,181,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Schizont","legendgroup":"Schizont","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":0.70000000000000018,"base":0,"x":[3],"y":[530.976],"text":"condition: Gametocyte(male)<br />mean:  530.9760<br />condition: Gametocyte(male)","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(225,135,39,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Gametocyte(male)","legendgroup":"Gametocyte(male)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":0.69999999999999973,"base":0,"x":[4],"y":[151.57262500000002],"text":"condition: Gametocyte(female)<br />mean:  151.5726<br />condition: Gametocyte(female)","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(32,133,78,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Gametocyte(female)","legendgroup":"Gametocyte(female)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1],"y":[317.93099999999998],"text":"condition: Trophozoite<br />min: 317.9310<br />condition: Trophozoite","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(188,60,41,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Trophozoite","legendgroup":"Trophozoite","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2],"y":[278.00299999999999],"text":"condition: Schizont<br />min: 278.0030<br />condition: Schizont","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,181,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Schizont","legendgroup":"Schizont","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3],"y":[530.976],"text":"condition: Gametocyte(male)<br />min: 530.9760<br />condition: Gametocyte(male)","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(225,135,39,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Gametocyte(male)","legendgroup":"Gametocyte(male)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4],"y":[151.57262500000002],"text":"condition: Gametocyte(female)<br />min: 151.5726<br />condition: Gametocyte(female)","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(32,133,78,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Gametocyte(female)","legendgroup":"Gametocyte(female)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1],"y":[317.93099999999998],"text":"condition: Trophozoite<br />max:  317.9310<br />condition: Trophozoite","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(188,60,41,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Trophozoite","legendgroup":"Trophozoite","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2],"y":[2477.52],"text":"condition: Schizont<br />max: 2477.5200<br />condition: Schizont","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,114,181,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Schizont","legendgroup":"Schizont","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3],"y":[530.976],"text":"condition: Gametocyte(male)<br />max:  530.9760<br />condition: Gametocyte(male)","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(225,135,39,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Gametocyte(male)","legendgroup":"Gametocyte(male)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4],"y":[151.57262500000002],"text":"condition: Gametocyte(female)<br />max:  151.5726<br />condition: Gametocyte(female)","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(32,133,78,1)","opacity":0.5,"size":9.4488188976377963,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(99,99,99,1)"}},"hoveron":"points","name":"Gametocyte(female)","legendgroup":"Gametocyte(female)","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":27.824636418824401,"r":7.3059360730593621,"b":120.6830839074089,"l":48.949771689497723},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,4.5999999999999996],"tickmode":"array","ticktext":["Trophozoite","Schizont","Gametocyte(male)","Gametocyte(female)"],"tickvals":[1,2,3,4],"categoryorder":"array","categoryarray":["Trophozoite","Schizont","Gametocyte(male)","Gametocyte(female)"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-90,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-123.876,2601.3960000000002],"tickmode":"array","ticktext":["0","500","1000","1500","2000","2500"],"tickvals":[0,500,1000,1500,2000.0000000000002,2500],"categoryorder":"array","categoryarray":["0","500","1000","1500","2000","2500"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"TPM","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","layer":"below","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"31c847e93e12":{"x":{},"y":{},"fill":{},"type":"bar"},"31c816fe35f7":{"x":{},"y":{},"fill":{}},"31c88024069":{"x":{},"y":{},"fill":{}}},"cur_data":"31c847e93e12","visdat":{"31c847e93e12":["function (y) ","x"],"31c816fe35f7":["function (y) ","x"],"31c88024069":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="searchmidb">
<strong><code>searchMidb()</code></strong><a class="anchor" aria-label="anchor" href="#searchmidb"></a>
</h3>
<p>This function enables you to fetch minor-introns information from
MiDB database in bulk. By default, all intron classes are fetched
(major-like, major_hybrid, minor-like, minor_hybrid, non-canonical). For
more information on minor introns visit <a href="https://midb.pnb.uconn.edu/index.php" class="external-link">MiDB database</a>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Let's see what organisms are present in MiDB</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"midbSpecies"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchMidb.html">searchMidb</a></span><span class="op">(</span><span class="st">"Toxoplasma gondii ME49"</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 43</span></span></span>
<span><span class="co">#&gt;   gene_symbol ensembl_gene_id transcript_key intron_name intron_start intron_end</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> NULL        TGME49_200010   TGME49_200010… Toxoplasma…      2<span style="text-decoration: underline;">247</span>210    2<span style="text-decoration: underline;">247</span>553</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> NULL        TGME49_200290   TGME49_200290… Toxoplasma…      6<span style="text-decoration: underline;">776</span>668    6<span style="text-decoration: underline;">776</span>934</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> NULL        TGME49_200295   TGME49_200295… Toxoplasma…      6<span style="text-decoration: underline;">783</span>452    6<span style="text-decoration: underline;">783</span>939</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> NULL        TGME49_200295   TGME49_200295… Toxoplasma…      6<span style="text-decoration: underline;">782</span>107    6<span style="text-decoration: underline;">782</span>402</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> NULL        TGME49_200300   TGME49_200300… Toxoplasma…      6<span style="text-decoration: underline;">786</span>183    6<span style="text-decoration: underline;">786</span>931</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> NULL        TGME49_200320   TGME49_200320… Toxoplasma…      6<span style="text-decoration: underline;">796</span>988    6<span style="text-decoration: underline;">797</span>499</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 37 more variables: term_nt &lt;chr&gt;, `5ss_seq` &lt;chr&gt;, `3ss_seq` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   U2_BPS &lt;chr&gt;, U12_BPS &lt;chr&gt;, `5ss_class` &lt;dbl&gt;, `3ss_class` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   intron_class &lt;chr&gt;, flanking_aa &lt;chr&gt;, intron_aa_position &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   intron_phase &lt;chr&gt;, intron_rank &lt;dbl&gt;, major_5ss_score &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   major_5ss_LOD &lt;dbl&gt;, major_5ss_LOD_stdev &lt;dbl&gt;, major_5ss_match &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   major_5ss_match_stdev &lt;dbl&gt;, minor_5ss_score &lt;dbl&gt;, minor_5ss_LOD &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   minor_5ss_LOD_stdev &lt;dbl&gt;, minor_5ss_match &lt;dbl&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="searchmiip">
<strong><code>searchMiip()</code></strong><a class="anchor" aria-label="anchor" href="#searchmiip"></a>
</h3>
<p>This function enables you to fetch Protein-protein interaction pairs
of <em>Plasmodium falciparum</em> and the respective stage (sexual and
asexual) they interact from <a href="http://www.hpppi.iicb.res.in/pfnet/" class="external-link">MIIP database</a>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/searchMiip.html">searchMiip</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0807800"</span>,<span class="st">"PF3D7_1023900"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 5</span></span></span>
<span><span class="co">#&gt;   interactorA   descriptionA                      interactorB descriptionB stage</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_0807800 26S proteasome regulatory subuni… PF3D7_0710… conserved P… game…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_1023900 chromodomain-helicase-DNA-bindin… PF3D7_1014… protein KIC8 game…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PF3D7_1023900 chromodomain-helicase-DNA-bindin… PF3D7_1138… protein KIC5 ring </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PF3D7_1335100 merozoite surface protein 7       PF3D7_1023… chromodomai… schi…</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="searchpm">
<strong><code>searchPM()</code></strong><a class="anchor" aria-label="anchor" href="#searchpm"></a>
</h3>
<p>Aside from <code>searchGSC</code> you can also use
<code><a href="../reference/searchPM.html">searchPM()</a></code> to fetch literature information where your gene
IDs of interest have been mentioned. This will however limit the search
to title abstract and keywords. In the background, it makes use of
<code>easyPubMed()</code> functions such as <code>get_pubmed_ids</code>
and <code>articles_to_list</code> and then transforms the output in form
of a table that is easy explore</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/searchPM.html">searchPM</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0420300"</span>,<span class="st">"PF3D7_0621000"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; PubMed Query used for PF3D7_0420300 was: </span></span>
<span><span class="co">#&gt;  "Plasmodium falciparum"[All Fields] AND "PF3D7_0420300"[Title/Abstract:~0] AND 2010/01/01:2025/12/31[Date - Publication]</span></span>
<span><span class="co">#&gt;       pmid                       doi</span></span>
<span><span class="co">#&gt; 1 39412522       10.7554/eLife.92201</span></span>
<span><span class="co">#&gt; 2 30526479 10.1186/s12864-018-5257-x</span></span>
<span><span class="co">#&gt;                                                                                                                                      title</span></span>
<span><span class="co">#&gt; 1 A  Plasmodium falciparum  MORC protein complex modulates epigenetic control of gene expression through interaction with heterochromatin.</span></span>
<span><span class="co">#&gt; 2    Schizont transcriptome variation among clinical isolates and laboratory-adapted clones of the malaria parasite Plasmodium falciparum.</span></span>
<span><span class="co">#&gt;   year month day       jabbrv      journal        GeneID</span></span>
<span><span class="co">#&gt; 1 2024    10  16        Elife        eLife PF3D7_0420300</span></span>
<span><span class="co">#&gt; 2 2019    03  18 BMC Genomics BMC genomics PF3D7_0420300</span></span></code></pre></div>
<p>Gene IDs for which no results are available will be shown on the
screen. However, when a query is successful, the function also prints
the exact query that can be used by you for reproducibility purposes.
This behavior can be turned off if you have a lot of gene IDs using
<code>verbose=FALSE</code>.</p>
<pre><code>"Plasmodium falciparum"[All Fields] AND "PF3D7_0420300"[Title/Abstract:~0] AND 2010/01/01:2025/12/31[Date - Publication] </code></pre>
</div>
<div class="section level3">
<h3 id="searchphpl">
<strong><code>searchPhPl()</code></strong><a class="anchor" aria-label="anchor" href="#searchphpl"></a>
</h3>
<p>This convenience function allow users to fetch Disruptability and
Mutant Phenotypes tables for gene of interest from PhenoPlasm database.
<code>fetch=1</code> helps fetch the Disruptability and
<code>fetch=2</code> helps fetch the Mutant Phenotype table.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/searchPhPl.html">searchPhPl</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0420300"</span>,<span class="st">"PF3D7_0621000"</span>,<span class="st">"PF3D7_0523800"</span><span class="op">)</span>, org<span class="op">=</span><span class="st">"pf"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;             Species Disruptability                          Reference</span></span>
<span><span class="co">#&gt; 1 P. falciparum 3D7     Refractory USF piggyBac screen (Insert. mut.)</span></span>
<span><span class="co">#&gt; 2 P. falciparum 3D7     Refractory USF piggyBac screen (Insert. mut.)</span></span>
<span><span class="co">#&gt; 3 P. falciparum 3D7     Refractory       354041168 ko attempts failed</span></span>
<span><span class="co">#&gt;                                 Submitter      QueryGID</span></span>
<span><span class="co">#&gt; 1                     USF PiggyBac Screen PF3D7_0621000</span></span>
<span><span class="co">#&gt; 2                     USF PiggyBac Screen PF3D7_0523800</span></span>
<span><span class="co">#&gt; 3 Theo Sanderson, Francis Crick Institute PF3D7_0523800</span></span>
<span><span class="fu"><a href="../reference/searchPhPl.html">searchPhPl</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0420300"</span>,<span class="st">"PF3D7_0621000"</span>,<span class="st">"PF3D7_0523800"</span><span class="op">)</span>, org<span class="op">=</span><span class="st">"pf"</span>, fetch<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 6</span></span></span>
<span><span class="co">#&gt;   Species           Stage   Phenotype               Reference Submitter QueryGID</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P. falciparum 3D7 Asexual Difference from wild-t… <span style="color: #949494;">"</span>PMID 39… Paul Sig… PF3D7_0…</span></span></code></pre></div>
<p>Oftentime, you would like to get the summary table like the one
plotted in PhenoPlasm that combines both Disruptability and Mutant
Phenotype information. Rather than using screen grab to get the snapshot
of the table, one can now download the table from <a href="https://phenoplasm.org/advanced.php" class="external-link"><code>Advanced Search</code></a>
button by submitting the geneIDs of interest and can feed that file to
<code><a href="../reference/easyPhplplottbl.html">easyPhplplottbl()</a></code> function of <em>plasmoRUtils</em> to
render such table from the <code>phenotype.txt</code> files directly</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read the file</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"phenotype.txt"</span>, skip <span class="op">=</span> <span class="fl">2</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="op">-</span><span class="fl">4</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="co">#remove the empty cols: GeneLocalisation and OrthologLocalisation</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename_with</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"Sprozoite"</span>, <span class="st">"Sporozoite"</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span> <span class="co">#Correct the colnames</span></span>
<span></span>
<span><span class="fu"><a href="../reference/easyPhplplottbl.html">easyPhplplottbl</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Or you can pass the file path directly</span></span>
<span><span class="fu"><a href="../reference/easyPhplplottbl.html">easyPhplplottbl</a></span><span class="op">(</span><span class="st">"phenotype.txt"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Load sample data (subset of genes from phenotype.txt file above)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pf3d7PhplTable</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/easyPhplplottbl.html">easyPhplplottbl</a></span><span class="op">(</span><span class="va">pf3d7PhplTable</span><span class="op">)</span></span></code></pre></div>
<div id="mwlefoodfi" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#mwlefoodfi table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#mwlefoodfi thead, #mwlefoodfi tbody, #mwlefoodfi tfoot, #mwlefoodfi tr, #mwlefoodfi td, #mwlefoodfi th {
  border-style: none;
}

#mwlefoodfi p {
  margin: 0;
  padding: 0;
}

#mwlefoodfi .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#mwlefoodfi .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#mwlefoodfi .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#mwlefoodfi .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#mwlefoodfi .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mwlefoodfi .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mwlefoodfi .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mwlefoodfi .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#mwlefoodfi .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#mwlefoodfi .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#mwlefoodfi .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#mwlefoodfi .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#mwlefoodfi .gt_spanner_row {
  border-bottom-style: hidden;
}

#mwlefoodfi .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#mwlefoodfi .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#mwlefoodfi .gt_from_md > :first-child {
  margin-top: 0;
}

#mwlefoodfi .gt_from_md > :last-child {
  margin-bottom: 0;
}

#mwlefoodfi .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#mwlefoodfi .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#mwlefoodfi .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#mwlefoodfi .gt_row_group_first td {
  border-top-width: 2px;
}

#mwlefoodfi .gt_row_group_first th {
  border-top-width: 2px;
}

#mwlefoodfi .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mwlefoodfi .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#mwlefoodfi .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#mwlefoodfi .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mwlefoodfi .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mwlefoodfi .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#mwlefoodfi .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#mwlefoodfi .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#mwlefoodfi .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mwlefoodfi .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mwlefoodfi .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mwlefoodfi .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mwlefoodfi .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mwlefoodfi .gt_left {
  text-align: left;
}

#mwlefoodfi .gt_center {
  text-align: center;
}

#mwlefoodfi .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#mwlefoodfi .gt_font_normal {
  font-weight: normal;
}

#mwlefoodfi .gt_font_bold {
  font-weight: bold;
}

#mwlefoodfi .gt_font_italic {
  font-style: italic;
}

#mwlefoodfi .gt_super {
  font-size: 65%;
}

#mwlefoodfi .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#mwlefoodfi .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#mwlefoodfi .gt_indent_1 {
  text-indent: 5px;
}

#mwlefoodfi .gt_indent_2 {
  text-indent: 10px;
}

#mwlefoodfi .gt_indent_3 {
  text-indent: 15px;
}

#mwlefoodfi .gt_indent_4 {
  text-indent: 20px;
}

#mwlefoodfi .gt_indent_5 {
  text-indent: 25px;
}

#mwlefoodfi .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#mwlefoodfi div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Gene">Gene</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Asexual">Asexual</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Gametocyte">Gametocyte</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Liver">Liver</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Oocyte">Oocyte</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Ookinete">Ookinete</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Sporozoite">Sporozoite</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" style="font-weight: bold;" scope="col" id="Viability">Viability</th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td headers="Gene" class="gt_row gt_left">PF3D7_0105200</td>
<td headers="Asexual" class="gt_row gt_center"><span style="color:darkred;">❌</span></td>
<td headers="Gametocyte" class="gt_row gt_center"></td>
<td headers="Liver" class="gt_row gt_center"></td>
<td headers="Oocyte" class="gt_row gt_center"></td>
<td headers="Ookinete" class="gt_row gt_center"></td>
<td headers="Sporozoite" class="gt_row gt_center"></td>
<td headers="Viability" class="gt_row gt_center">
<span style="color:#00AA00 !important;">✔</span> <span style="color:darkred; opacity:0.4;">❌</span>
</td>
</tr>
<tr>
<td headers="Gene" class="gt_row gt_left">PF3D7_0105300</td>
<td headers="Asexual" class="gt_row gt_center"><span style="color:darkgreen; opacity:0.4;">✅</span></td>
<td headers="Gametocyte" class="gt_row gt_center"><span style="color:darkgreen; opacity:0.4;">✅</span></td>
<td headers="Liver" class="gt_row gt_center"><span style="color:darkred; opacity:0.4;">❗</span></td>
<td headers="Oocyte" class="gt_row gt_center"><span style="color:darkred; opacity:0.4;">❗</span></td>
<td headers="Ookinete" class="gt_row gt_center"><span style="color:darkgreen; opacity:0.4;">✅</span></td>
<td headers="Sporozoite" class="gt_row gt_center"><span style="color:darkred; opacity:0.4;">❗</span></td>
<td headers="Viability" class="gt_row gt_center">
<span style="color:darkred;">❌</span> <span style="color:darkgreen; opacity:0.4;">✔</span>
</td>
</tr>
<tr>
<td headers="Gene" class="gt_row gt_left">PF3D7_0105400</td>
<td headers="Asexual" class="gt_row gt_center"><span style="color:darkred; opacity:0.4;">❗</span></td>
<td headers="Gametocyte" class="gt_row gt_center"></td>
<td headers="Liver" class="gt_row gt_center"></td>
<td headers="Oocyte" class="gt_row gt_center"></td>
<td headers="Ookinete" class="gt_row gt_center"></td>
<td headers="Sporozoite" class="gt_row gt_center"></td>
<td headers="Viability" class="gt_row gt_center"><span style="color:#00AA00 !important;">✔</span></td>
</tr>
<tr>
<td headers="Gene" class="gt_row gt_left">PF3D7_0217500</td>
<td headers="Asexual" class="gt_row gt_center">
<span style="color:darkred;">⟴</span> <span style="color:darkred;">🟥</span> <span style="color:darkgreen; opacity:0.4;">✅</span>
</td>
<td headers="Gametocyte" class="gt_row gt_center"><span style="color:darkred;">❗</span></td>
<td headers="Liver" class="gt_row gt_center"></td>
<td headers="Oocyte" class="gt_row gt_center"><span style="color:darkred; opacity:0.4;">❗</span></td>
<td headers="Ookinete" class="gt_row gt_center"></td>
<td headers="Sporozoite" class="gt_row gt_center"></td>
<td headers="Viability" class="gt_row gt_center">
<span style="color:darkred;">❌</span> <span style="color:darkgreen; opacity:0.4;">✔</span> <span style="color:darkred; opacity:0.4;">❌</span>
</td>
</tr>
<tr>
<td headers="Gene" class="gt_row gt_left">PF3D7_1337800</td>
<td headers="Asexual" class="gt_row gt_center">
<span style="color:darkred;">❗</span> <span style="color:darkred;">🟥</span> <span style="color:darkred;">❗</span>
</td>
<td headers="Gametocyte" class="gt_row gt_center"></td>
<td headers="Liver" class="gt_row gt_center"></td>
<td headers="Oocyte" class="gt_row gt_center"></td>
<td headers="Ookinete" class="gt_row gt_center"></td>
<td headers="Sporozoite" class="gt_row gt_center"></td>
<td headers="Viability" class="gt_row gt_center">
<span style="color:darkred;">❌</span> <span style="color:darkred; opacity:0.4;">❌</span>
</td>
</tr>
</tbody>
</table>
</div>
<p>Windows users might face issues saving these plots as pdf directly in
which case, the tables can be saved as HTML files which can then be
converted to SVG or PDF formats using various online converters to
combine them with other plots.</p>
<blockquote>
<p><strong>Note:</strong> As per <a href="https://phenoplasm.org/csvsupport.php" class="external-link">Phenotype taxonomy</a> of
Phenoplasm, the database uses “D” for both
<code>Difference from wild-type</code> and <code>Egress defect</code>
which is confusing and difficult to resolve programmatically. An example
of this is <code>PF3D7_1337800</code> that have “D S D” in the “Gene
Asexual”. While we have requested the database maintainer to fix this,
please watch out for borderline cases like these.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="searchtedconsensus">
<strong><code>searchTedConsensus()</code></strong><a class="anchor" aria-label="anchor" href="#searchtedconsensus"></a>
</h3>
<p>This function helps users fetch the domain information from
<strong>The Encyclopedia of Domains</strong> database given set of
uniprot IDs. Usually these table contains a numeric CATH labels which
are difficult to comprehend and user has to click on them one by one to
find the domain name. We enable conversion of these CATH labels to
description using <code>returnCATHdesc=TRUE</code>. This will try to
scrap the labels for given CATH label from CATH database wherever
possible.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/searchTedConsensus.html">searchTedConsensus</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q7K6A1"</span>,<span class="st">"Q8IAP8"</span>,<span class="st">"C0H4D0"</span>,<span class="st">"C6KT90"</span>,<span class="st">"Q8IBJ7"</span><span class="op">)</span>, returnCATHdesc<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;                        ted_id uniprot_acc                       md5_domain</span></span>
<span><span class="co">#&gt; 1 AF-Q7K6A1-F1-model_v4_TED01      Q7K6A1 b99e920f0ded31aa96af0ef9be1338f4</span></span>
<span><span class="co">#&gt; 2 AF-C0H4D0-F1-model_v4_TED01      C0H4D0 cd912dcbbb5d070cbb254c0a88278fe4</span></span>
<span><span class="co">#&gt; 3 AF-C6KT90-F1-model_v4_TED02      C6KT90 70d20592d9f682bff23dc6188f318244</span></span>
<span><span class="co">#&gt; 4 AF-C6KT90-F1-model_v4_TED01      C6KT90 7cc174ebefe723733b6e63508fd23a9e</span></span>
<span><span class="co">#&gt; 5 AF-Q8IBJ7-F1-model_v4_TED01      Q8IBJ7 71697d50571d5fe2331a13ff16503478</span></span>
<span><span class="co">#&gt;   consensus_level chopping nres_domain num_segments   plddt</span></span>
<span><span class="co">#&gt; 1            high    6-376         371            1 97.1740</span></span>
<span><span class="co">#&gt; 2          medium   55-153          99            1 88.9028</span></span>
<span><span class="co">#&gt; 3          medium  322-382          61            1 45.3118</span></span>
<span><span class="co">#&gt; 4          medium  172-203          32            1 48.8553</span></span>
<span><span class="co">#&gt; 5          medium    54-88          35            1 87.3500</span></span>
<span><span class="co">#&gt;   num_helix_strand_turn num_helix num_strand num_helix_strand num_turn</span></span>
<span><span class="co">#&gt; 1                    60        16          8               24       35</span></span>
<span><span class="co">#&gt; 2                    15         5          4                9        6</span></span>
<span><span class="co">#&gt; 3                     3         3          0                3        0</span></span>
<span><span class="co">#&gt; 4                     2         1          0                1        1</span></span>
<span><span class="co">#&gt; 5                     5         0          3                3        2</span></span>
<span><span class="co">#&gt;   proteome_id   cath_label cath_assignment_level cath_assignment_method</span></span>
<span><span class="co">#&gt; 1       36329  3.40.800.20                     H               foldseek</span></span>
<span><span class="co">#&gt; 2       36329 3.30.70.2380                     H               foldseek</span></span>
<span><span class="co">#&gt; 3       36329     4.10.860                     T              foldclass</span></span>
<span><span class="co">#&gt; 4       36329       1.20.5                     T              foldclass</span></span>
<span><span class="co">#&gt; 5       36329            -                     -                      -</span></span>
<span><span class="co">#&gt;   packing_density norm_rg tax_common_name                 tax_scientific_name</span></span>
<span><span class="co">#&gt; 1          13.064   0.298                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 2          12.537   0.306                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 3           9.900   0.374                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 4           8.900   0.403                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 5           9.833   0.370                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt;                                                                                                                                                       tax_lineage</span></span>
<span><span class="co">#&gt; 1 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 2 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 3 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 4 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 5 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span></span>
<span><span class="fu"><a href="../reference/searchTedConsensus.html">searchTedConsensus</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q7K6A1"</span>,<span class="st">"Q8IAP8"</span>,<span class="st">"C0H4D0"</span>,<span class="st">"C6KT90"</span>,<span class="st">"Q8IBJ7"</span><span class="op">)</span>, returnCATHdesc<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;                        ted_id uniprot_acc                       md5_domain</span></span>
<span><span class="co">#&gt; 1 AF-Q7K6A1-F1-model_v4_TED01      Q7K6A1 b99e920f0ded31aa96af0ef9be1338f4</span></span>
<span><span class="co">#&gt; 2 AF-C0H4D0-F1-model_v4_TED01      C0H4D0 cd912dcbbb5d070cbb254c0a88278fe4</span></span>
<span><span class="co">#&gt; 3 AF-C6KT90-F1-model_v4_TED02      C6KT90 70d20592d9f682bff23dc6188f318244</span></span>
<span><span class="co">#&gt; 4 AF-C6KT90-F1-model_v4_TED01      C6KT90 7cc174ebefe723733b6e63508fd23a9e</span></span>
<span><span class="co">#&gt; 5 AF-Q8IBJ7-F1-model_v4_TED01      Q8IBJ7 71697d50571d5fe2331a13ff16503478</span></span>
<span><span class="co">#&gt;   consensus_level chopping nres_domain num_segments   plddt</span></span>
<span><span class="co">#&gt; 1            high    6-376         371            1 97.1740</span></span>
<span><span class="co">#&gt; 2          medium   55-153          99            1 88.9028</span></span>
<span><span class="co">#&gt; 3          medium  322-382          61            1 45.3118</span></span>
<span><span class="co">#&gt; 4          medium  172-203          32            1 48.8553</span></span>
<span><span class="co">#&gt; 5          medium    54-88          35            1 87.3500</span></span>
<span><span class="co">#&gt;   num_helix_strand_turn num_helix num_strand num_helix_strand num_turn</span></span>
<span><span class="co">#&gt; 1                    60        16          8               24       35</span></span>
<span><span class="co">#&gt; 2                    15         5          4                9        6</span></span>
<span><span class="co">#&gt; 3                     3         3          0                3        0</span></span>
<span><span class="co">#&gt; 4                     2         1          0                1        1</span></span>
<span><span class="co">#&gt; 5                     5         0          3                3        2</span></span>
<span><span class="co">#&gt;   proteome_id   cath_label cath_assignment_level cath_assignment_method</span></span>
<span><span class="co">#&gt; 1       36329  3.40.800.20                     H               foldseek</span></span>
<span><span class="co">#&gt; 2       36329 3.30.70.2380                     H               foldseek</span></span>
<span><span class="co">#&gt; 3       36329     4.10.860                     T              foldclass</span></span>
<span><span class="co">#&gt; 4       36329       1.20.5                     T              foldclass</span></span>
<span><span class="co">#&gt; 5       36329            -                     -                      -</span></span>
<span><span class="co">#&gt;   packing_density norm_rg tax_common_name                 tax_scientific_name</span></span>
<span><span class="co">#&gt; 1          13.064   0.298                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 2          12.537   0.306                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 3           9.900   0.374                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 4           8.900   0.403                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt; 5           9.833   0.370                 Plasmodium falciparum (isolate 3D7)</span></span>
<span><span class="co">#&gt;                                                                                                                                                       tax_lineage</span></span>
<span><span class="co">#&gt; 1 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 2 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 3 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 4 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt; 5 cellular organisms, Eukaryota, Sar, Alveolata, Apicomplexa, Aconoidasida, Haemosporida, Plasmodiidae, Plasmodium, Plasmodium (Laverania), Plasmodium falciparum</span></span>
<span><span class="co">#&gt;              cath_label_desc</span></span>
<span><span class="co">#&gt; 1 Histone deacetylase domain</span></span>
<span><span class="co">#&gt; 2                           </span></span>
<span><span class="co">#&gt; 3                           </span></span>
<span><span class="co">#&gt; 4                           </span></span>
<span><span class="co">#&gt; 5                       NULL</span></span></code></pre></div>
<p>In the example above, <code>C0H4D0</code> have CATH label <a href="https://www.cathdb.info/version/v4_4_0/superfamily/3.30.70.2380" class="external-link"><code>3.30.70.2380</code></a>.
But this superfamily doesn’t have a name. Besides, sometimes instead of
Superfamily CATH labels, TED might use CATH-Gene3D Hierarchy. No
description is returned in such cases.</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 26100)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_India.utf8  LC_CTYPE=English_India.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_India.utf8 LC_NUMERIC=C                  </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_India.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Riyadh</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] plyr_1.8.9         dplyr_1.1.4        plasmoRUtils_1.0.0 rlang_1.1.6       </span></span>
<span><span class="co">#&gt; [5] readr_2.1.5        janitor_2.2.1      BiocStyle_2.32.1  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] IRanges_2.38.1              dichromat_2.0-0.1          </span></span>
<span><span class="co">#&gt;   [3] vroom_1.6.5                 progress_1.2.3             </span></span>
<span><span class="co">#&gt;   [5] vsn_3.72.0                  nnet_7.3-19                </span></span>
<span><span class="co">#&gt;   [7] Biostrings_2.72.1           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;   [9] digest_0.6.37               png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [11] proxy_0.4-27                MSnbase_2.30.1             </span></span>
<span><span class="co">#&gt;  [13] echarts4r_0.4.5.9000        parallelly_1.45.0          </span></span>
<span><span class="co">#&gt;  [15] MASS_7.3-61                 pkgdown_2.1.3              </span></span>
<span><span class="co">#&gt;  [17] reshape2_1.4.4              httpuv_1.6.16              </span></span>
<span><span class="co">#&gt;  [19] foreach_1.5.2               BiocGenerics_0.50.0        </span></span>
<span><span class="co">#&gt;  [21] withr_3.0.2                 xfun_0.52                  </span></span>
<span><span class="co">#&gt;  [23] ggpubr_0.6.1                survival_3.8-3             </span></span>
<span><span class="co">#&gt;  [25] memoise_2.0.1               hexbin_1.28.5              </span></span>
<span><span class="co">#&gt;  [27] ggsci_3.2.0                 mixtools_2.0.0.1           </span></span>
<span><span class="co">#&gt;  [29] systemfonts_1.2.3           ragg_1.4.0                 </span></span>
<span><span class="co">#&gt;  [31] gtools_3.9.5                easyPubMed_2.13            </span></span>
<span><span class="co">#&gt;  [33] Formula_1.2-5               prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt;  [35] KEGGREST_1.44.1             promises_1.3.3             </span></span>
<span><span class="co">#&gt;  [37] httr_1.4.7                  rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [39] restfulr_0.0.16             globals_0.18.0             </span></span>
<span><span class="co">#&gt;  [41] ps_1.9.1                    rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;  [43] UCSC.utils_1.0.0            generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [45] processx_3.8.6              curl_6.4.0                 </span></span>
<span><span class="co">#&gt;  [47] ncdf4_1.24                  S4Vectors_0.42.1           </span></span>
<span><span class="co">#&gt;  [49] zlibbioc_1.50.0             ScaledMatrix_1.12.0        </span></span>
<span><span class="co">#&gt;  [51] randomForest_4.7-1.2        bio3d_2.4-5                </span></span>
<span><span class="co">#&gt;  [53] GenomeInfoDbData_1.2.12     SparseArray_1.4.8          </span></span>
<span><span class="co">#&gt;  [55] xtable_1.8-4                stringr_1.5.1              </span></span>
<span><span class="co">#&gt;  [57] desc_1.4.3                  doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [59] evaluate_1.0.4              S4Arrays_1.4.1             </span></span>
<span><span class="co">#&gt;  [61] BiocFileCache_2.12.0        preprocessCore_1.66.0      </span></span>
<span><span class="co">#&gt;  [63] hms_1.1.3                   GenomicRanges_1.56.2       </span></span>
<span><span class="co">#&gt;  [65] bookdown_0.43               irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [67] colorspace_2.1-1            filelock_1.0.3             </span></span>
<span><span class="co">#&gt;  [69] magrittr_2.0.3              snakecase_0.11.1           </span></span>
<span><span class="co">#&gt;  [71] later_1.4.2                 viridis_0.6.5              </span></span>
<span><span class="co">#&gt;  [73] lattice_0.22-6              MsCoreUtils_1.16.1         </span></span>
<span><span class="co">#&gt;  [75] future.apply_1.20.0         SparseM_1.84-2             </span></span>
<span><span class="co">#&gt;  [77] XML_3.99-0.18               scuttle_1.14.0             </span></span>
<span><span class="co">#&gt;  [79] matrixStats_1.5.0           class_7.3-22               </span></span>
<span><span class="co">#&gt;  [81] pillar_1.11.0               nlme_3.1-166               </span></span>
<span><span class="co">#&gt;  [83] iterators_1.0.14            compiler_4.4.1             </span></span>
<span><span class="co">#&gt;  [85] beachmat_2.20.0             stringi_1.8.7              </span></span>
<span><span class="co">#&gt;  [87] gower_1.0.2                 SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [89] dendextend_1.19.1           lubridate_1.9.4            </span></span>
<span><span class="co">#&gt;  [91] GenomicAlignments_1.40.0    drawProteins_1.24.0        </span></span>
<span><span class="co">#&gt;  [93] crayon_1.5.3                abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [95] BiocIO_1.14.0               bit_4.6.0                  </span></span>
<span><span class="co">#&gt;  [97] chromote_0.5.1              pcaMethods_1.96.0          </span></span>
<span><span class="co">#&gt;  [99] codetools_0.2-20            textshaping_1.0.1          </span></span>
<span><span class="co">#&gt; [101] recipes_1.3.1               BiocSingular_1.20.0        </span></span>
<span><span class="co">#&gt; [103] MLInterfaces_1.84.0         crosstalk_1.2.1            </span></span>
<span><span class="co">#&gt; [105] bslib_0.9.0                 e1071_1.7-16               </span></span>
<span><span class="co">#&gt; [107] plotly_4.11.0               LaplacesDemon_16.1.6       </span></span>
<span><span class="co">#&gt; [109] mime_0.13                   MultiAssayExperiment_1.30.3</span></span>
<span><span class="co">#&gt; [111] splines_4.4.1               Rcpp_1.0.14                </span></span>
<span><span class="co">#&gt; [113] dbplyr_2.5.0                sparseMatrixStats_1.16.0   </span></span>
<span><span class="co">#&gt; [115] knitr_1.50                  blob_1.2.4                 </span></span>
<span><span class="co">#&gt; [117] utf8_1.2.6                  clue_0.3-66                </span></span>
<span><span class="co">#&gt; [119] mzR_2.38.0                  AnnotationFilter_1.28.0    </span></span>
<span><span class="co">#&gt; [121] fs_1.6.6                    QFeatures_1.14.2           </span></span>
<span><span class="co">#&gt; [123] listenv_0.9.1               mzID_1.42.0                </span></span>
<span><span class="co">#&gt; [125] DelayedMatrixStats_1.26.0   ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt; [127] tibble_3.3.0                Matrix_1.7-1               </span></span>
<span><span class="co">#&gt; [129] statmod_1.5.0               tzdb_0.5.0                 </span></span>
<span><span class="co">#&gt; [131] lpSolve_5.6.23              pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [133] tools_4.4.1                 cachem_1.1.0               </span></span>
<span><span class="co">#&gt; [135] RSQLite_2.4.1               viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [137] rvest_1.0.4                 DBI_1.2.3                  </span></span>
<span><span class="co">#&gt; [139] impute_1.78.0               fastmap_1.2.0              </span></span>
<span><span class="co">#&gt; [141] rmarkdown_2.29              scales_1.4.0               </span></span>
<span><span class="co">#&gt; [143] grid_4.4.1                  gt_1.0.0                   </span></span>
<span><span class="co">#&gt; [145] Rsamtools_2.20.0            broom_1.0.8                </span></span>
<span><span class="co">#&gt; [147] sass_0.4.10                 coda_0.19-4.1              </span></span>
<span><span class="co">#&gt; [149] FNN_1.1.4.1                 BiocManager_1.30.26        </span></span>
<span><span class="co">#&gt; [151] graph_1.82.0                carData_3.0-5              </span></span>
<span><span class="co">#&gt; [153] selectr_0.4-2               SingleR_2.6.0              </span></span>
<span><span class="co">#&gt; [155] rpart_4.1.23                farver_2.1.2               </span></span>
<span><span class="co">#&gt; [157] yaml_2.3.10                 AnnotationForge_1.46.0     </span></span>
<span><span class="co">#&gt; [159] MatrixGenerics_1.16.0       rtracklayer_1.64.0         </span></span>
<span><span class="co">#&gt; [161] cli_3.6.5                   purrr_1.0.4                </span></span>
<span><span class="co">#&gt; [163] stats4_4.4.1                txdbmaker_1.0.1            </span></span>
<span><span class="co">#&gt; [165] lifecycle_1.0.4             caret_7.0-1                </span></span>
<span><span class="co">#&gt; [167] Biobase_2.64.0              mvtnorm_1.3-3              </span></span>
<span><span class="co">#&gt; [169] lava_1.8.1                  kernlab_0.9-33             </span></span>
<span><span class="co">#&gt; [171] backports_1.5.0             BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt; [173] annotate_1.82.0             timechange_0.3.0           </span></span>
<span><span class="co">#&gt; [175] gtable_0.3.6                rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [177] parallel_4.4.1              pROC_1.18.5                </span></span>
<span><span class="co">#&gt; [179] limma_3.60.6                jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt; [181] bitops_1.0-9                ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt; [183] bit64_4.6.0-1               pRoloc_1.44.1              </span></span>
<span><span class="co">#&gt; [185] jquerylib_0.1.4             segmented_2.1-4            </span></span>
<span><span class="co">#&gt; [187] timeDate_4041.110           lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [189] shiny_1.11.1                htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [191] affy_1.82.0                 GO.db_3.19.1               </span></span>
<span><span class="co">#&gt; [193] rappdirs_0.3.3              glue_1.8.0                 </span></span>
<span><span class="co">#&gt; [195] httr2_1.2.0                 XVector_0.44.0             </span></span>
<span><span class="co">#&gt; [197] RCurl_1.98-1.17             MALDIquant_1.22.3          </span></span>
<span><span class="co">#&gt; [199] mclust_6.1.1                gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [201] igraph_2.1.4                R6_2.6.1                   </span></span>
<span><span class="co">#&gt; [203] tidyr_1.3.1                 SingleCellExperiment_1.26.0</span></span>
<span><span class="co">#&gt; [205] labeling_0.4.3              GenomicFeatures_1.56.0     </span></span>
<span><span class="co">#&gt; [207] cluster_2.1.8               GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt; [209] ipred_0.9-15                DelayedArray_0.30.1        </span></span>
<span><span class="co">#&gt; [211] tidyselect_1.2.1            ProtGenerics_1.36.0        </span></span>
<span><span class="co">#&gt; [213] sampling_2.11               xml2_1.3.8                 </span></span>
<span><span class="co">#&gt; [215] car_3.1-3                   AnnotationDbi_1.66.0       </span></span>
<span><span class="co">#&gt; [217] future_1.58.0               ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">#&gt; [219] rsvd_1.0.5                  affyio_1.74.0              </span></span>
<span><span class="co">#&gt; [221] topGO_2.56.0                data.table_1.17.6          </span></span>
<span><span class="co">#&gt; [223] websocket_1.4.4             mgsub_1.7.3                </span></span>
<span><span class="co">#&gt; [225] htmlwidgets_1.6.4           RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt; [227] biomaRt_2.60.1              hardhat_1.4.1              </span></span>
<span><span class="co">#&gt; [229] prodlim_2025.04.28          PSMatch_1.8.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-López2015" class="csl-entry">
López, Yosvany, Kenta Nakai, and Ashwini Patil. 2015. <span>“HitPredict
Version 4: Comprehensive Reliability Scoring of Physical
Protein<span></span>protein Interactions from More Than 100
Species.”</span> <em>Database</em> 2015: bav117. <a href="https://doi.org/10.1093/database/bav117" class="external-link">https://doi.org/10.1093/database/bav117</a>.
</div>
<div id="ref-Pandey2017" class="csl-entry">
Pandey, Rajan, Pawan Kumar, and Dinesh Gupta. 2017. <span>“KiPho:
Malaria Parasite Kinome and Phosphatome Portal.”</span>
<em>Database</em> 2017 (January). <a href="https://doi.org/10.1093/database/bax063" class="external-link">https://doi.org/10.1093/database/bax063</a>.
</div>
<div id="ref-Persson2023" class="csl-entry">
Persson, Emma, and Erik L. L. Sonnhammer. 2023. <span>“InParanoiDB 9:
Ortholog Groups for Protein Domains and Full-Length Proteins.”</span>
<em>Journal of Molecular Biology</em> 435 (14): 168001. <a href="https://doi.org/10.1016/j.jmb.2023.168001" class="external-link">https://doi.org/10.1016/j.jmb.2023.168001</a>.
</div>
<div id="ref-Sardar2019" class="csl-entry">
Sardar, Rahila, Abhinav Kaushik, Rajan Pandey, Asif Mohmmed, Shakir Ali,
and Dinesh Gupta. 2019. <span>“ApicoTFdb: The Comprehensive Web
Repository of Apicomplexan Transcription Factors and
Transcription-Associated Co-Factors.”</span> <em>Database</em> 2019
(January). <a href="https://doi.org/10.1093/database/baz094" class="external-link">https://doi.org/10.1093/database/baz094</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Rohit Satyam, Alberto Maillo, David Gomez-Cabrero, Arnab Pain.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
