<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Other useful functions • plasmoRUtils</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Other useful functions">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">plasmoRUtils</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Introduction_to_plasmoRUtils.html">Introduction to plasmoRUtils</a></li>
    <li><a class="dropdown-item" href="../articles/Gene_ID_Conversion.html">Accessing component databases of VEuPathDB</a></li>
    <li><a class="dropdown-item" href="../articles/Miscellaneous_function.html">Other useful functions</a></li>
    <li><a class="dropdown-item" href="../articles/Need_for_reanalysis.html">RNASeq: Need for reanalysis?</a></li>
    <li><a class="dropdown-item" href="../articles/retrieving_orthologs.html">Retrieving Orthologs</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Other useful functions</h1>
                        <h4 data-toc-skip class="author">Rohit
Satyam</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Alberto
Maillo</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">David
Gomez-Cabrero</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Arnab
Pain</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="date">22 August, 2025</h4>
      

      <div class="d-none name"><code>Miscellaneous_function.Rmd</code></div>
    </address>
</address>
</address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This section covers the capability of plasmoRUtils to carry out
      other mundane bioinformatics tasks.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Other than search functions, the <em>plasmoRUtils</em> package offers
various other functions that are routinely needed when performing
bioinformatics analysis. Their usage is discussed below.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package and some other useful packages by using</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rohit-satyam.github.io/plasmoRUtils/">plasmoRUtils</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://had.co.nz/plyr" class="external-link">plyr</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="making-orgdb-and-txdb-objects">Making OrgDb and TxDb objects<a class="anchor" aria-label="anchor" href="#making-orgdb-and-txdb-objects"></a>
</h3>
<p>For non-model organism, using R packages such as <em><a href="https://bioconductor.org/packages/3.19/clusterProfiler" class="external-link">clusterProfiler</a></em>
for enrichment analysis requires org.db packages. Unfortunately, this
requires ability to understand and use <em><a href="https://bioconductor.org/packages/3.19/AnnotationDbi" class="external-link">AnnotationDbi</a></em>
package. To make development of such packages quickly, we wrote a
wrapper function that instantly creates a TAR file that can be used and
shared by the user in no time.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## List organism you are interested in making org.db package from VEuPathDB.</span></span>
<span><span class="co">## Use the links from PlasmoDB directly or provide locally saved GFF and GAF files.</span></span>
<span></span>
<span><span class="co">## Get taxonomy ID using listVeupathdb(c("primary_key","ncbi_tax_id"))</span></span>
<span></span>
<span><span class="fu"><a href="../reference/easymakeOrgdb.html">easymakeOrgdb</a></span><span class="op">(</span></span>
<span>  gff <span class="op">=</span></span>
<span>    <span class="st">"https://plasmodb.org/common/downloads/release-68/Pfalciparum3D7/gff/data/PlasmoDB-68_Pfalciparum3D7.gff"</span>,</span>
<span>  gaf <span class="op">=</span></span>
<span>    <span class="st">"https://plasmodb.org/common/downloads/release-68/Pfalciparum3D7/gaf/PlasmoDB-68_Pfalciparum3D7_Curated_GO.gaf.gz"</span>,</span>
<span>  out.dir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  taxid <span class="op">=</span> <span class="fl">36329</span>,</span>
<span>  genus <span class="op">=</span> <span class="st">"Plasmodium"</span>,</span>
<span>  sp <span class="op">=</span> <span class="st">"falciparum3D7"</span>,</span>
<span>  version <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  maintainer <span class="op">=</span> <span class="st">"John doe &lt;johndoe@gmail.com&gt;"</span> <span class="co">## Your name if you are maintaining it.</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Once the package is ready, one can use clusterProfiler as follows</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Pfalciparum3D7.eg.db</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene          <span class="op">=</span> <span class="va">genes</span>,</span>
<span>                universe      <span class="op">=</span> <span class="va">bkg_genes</span>,</span>
<span>                OrgDb         <span class="op">=</span> <span class="va">org.Pfalciparum3D7.eg.db</span>,</span>
<span>                keyType <span class="op">=</span> <span class="st">"GID"</span>,</span>
<span>                ont           <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>                minGSSize<span class="op">=</span><span class="fl">3</span>,</span>
<span>                pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                pvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                qvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        readable      <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The function above will create an org.db tar file which can be
installed and loaded to be used with <em><a href="https://bioconductor.org/packages/3.19/clusterProfiler" class="external-link">clusterProfiler</a></em>.</p>
<p>Similarly, if users want to make TxDb object in R to use other R
packages such as <em><a href="https://bioconductor.org/packages/3.19/gDNAx" class="external-link">gDNAx</a></em> used
to access the genomic DNA contamination, they can do so easily using
<code><a href="../reference/easyTxDbmaker.html">easyTxDbmaker()</a></code> function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">txdb</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/easyTxDbmaker.html">easyTxDbmaker</a></span><span class="op">(</span></span>
<span> gff<span class="op">=</span><span class="st">"https://toxodb.org/common/downloads/release-68/TgondiiME49/gff/data/ToxoDB-68_TgondiiME49.gff"</span>,</span>
<span> fasta<span class="op">=</span><span class="st">"https://toxodb.org/common/downloads/release-68/TgondiiME49/fasta/data/ToxoDB-68_TgondiiME49_Genome.fasta"</span>,</span>
<span> abbr<span class="op">=</span><span class="st">"TgondiiME49"</span>,</span>
<span> taxid<span class="op">=</span><span class="fl">508771</span>,org <span class="op">=</span> <span class="st">"Toxoplasma gondii ME49"</span>,</span>
<span> db <span class="op">=</span> <span class="st">"ToxoDB release 68"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="other-easy-functions">Other easy functions<a class="anchor" aria-label="anchor" href="#other-easy-functions"></a>
</h3>
<p>Users are also provided with easy functions which are wrapper
functions around routinely used to analyze bioinformatics data.</p>
<div class="section level4">
<h4 id="tpm-normalization-and-visualization">TPM normalization and visualization<a class="anchor" aria-label="anchor" href="#tpm-normalization-and-visualization"></a>
</h4>
<p>In this section we will see how we can calculate effective gene
lengths from a GFF file followed by TPM normalization and visualize the
resulting normalized matrix. These functions are provided since most of
the RNASeq data present on VEuPathDb is TPM normalized and therefore
make it easy to plot the database procured values alongside your own
samples.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># To calculate the effective length of the genes you can use the following function</span></span>
<span><span class="va">baseurl</span> <span class="op">&lt;-</span> <span class="st">"https://plasmodb.org/common/downloads/release-68/"</span></span>
<span><span class="fu"><a href="../reference/getEffLen.html">getEffLen</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">baseurl</span>,</span>
<span><span class="st">"Pfalciparum3D7/gff/data/PlasmoDB-68_Pfalciparum3D7.gff"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   GeneID        Length</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PF3D7_0100100   <span style="text-decoration: underline;">6</span>492</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PF3D7_0100200    996</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PF3D7_0100300   <span style="text-decoration: underline;">3</span>984</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PF3D7_0100400   <span style="text-decoration: underline;">1</span>119</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PF3D7_0100500    112</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PF3D7_0100600   <span style="text-decoration: underline;">1</span>080</span></span></code></pre></div>
<p>The above function returns a dataframe which can be fed to the
<code><a href="../reference/easyTPM.html">easyTPM()</a></code> function alongside the count matrix. The function
also adds the effective length as a column at the end so that people can
find it easy to convert the values to the raw counts. This is also
helpful since the exon-intron boundaries are always evolving as the new
data-sets become available.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="co">## Generating dummy effective length for quick demonstration</span></span>
<span><span class="va">gene_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>GeneID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0100100"</span>, <span class="st">"PF3D7_0100200"</span>, <span class="st">"PF3D7_0100300"</span><span class="op">)</span>, Length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6492</span>, <span class="fl">996</span>, <span class="fl">3984</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Making a dummy count matrix</span></span>
<span><span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span>, <span class="fl">40</span>, <span class="fl">50</span>, <span class="fl">60</span><span class="op">)</span>,</span>
<span>                       nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                       dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0100100"</span>, <span class="st">"PF3D7_0100200"</span>, <span class="st">"PF3D7_0100300"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample1"</span>, <span class="st">"sample2"</span><span class="op">)</span><span class="op">)</span></span>
<span>                       <span class="op">)</span></span>
<span></span>
<span><span class="co">## TPM normalization</span></span>
<span><span class="va">test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/easyTPM.html">easyTPM</a></span><span class="op">(</span><span class="va">count_matrix</span>, <span class="va">gene_info</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## TPM visualization as a bubble plot</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">test</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">test</span><span class="op">)</span><span class="op">]</span>,na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/easyExpPlot.html">easyExpPlot</a></span><span class="op">(</span><span class="va">df</span>,x<span class="op">=</span><span class="st">"variable"</span>,y<span class="op">=</span><span class="st">"GeneID"</span>,value<span class="op">=</span><span class="st">"value"</span>, type <span class="op">=</span> <span class="st">"bubble"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Samples"</span>,y<span class="op">=</span><span class="st">"Gene IDs"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Miscellaneous_function_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>You can also plot normalized expression data as line plot if you have
cluster of genes with similar pattern of expression. To demonstrate this
we will use a subset of genes from <span class="citation">(<a href="#ref-Subudhi2020">Subudhi et al. 2020</a>)</span> time series
dataset. We will perform TPM normalization, followed by log
transformation and eventually use the Z-scores for visualization.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"subudhi2020"</span><span class="op">)</span></span>
<span><span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="va">subudhi2020</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="va">baseurl</span> <span class="op">&lt;-</span> <span class="st">"https://plasmodb.org/common/downloads/release-68/"</span></span>
<span><span class="va">gene_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getEffLen.html">getEffLen</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">baseurl</span>,<span class="st">"Pfalciparum3D7/gff/data/PlasmoDB-68_Pfalciparum3D7.gff"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">normalised</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/easyTPM.html">easyTPM</a></span><span class="op">(</span><span class="va">count_matrix</span>, <span class="va">gene_info</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## Since normalized matrix have effective gene length column at the end</span></span>
<span><span class="va">logTransformed</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">normalised</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">normalised</span><span class="op">)</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Get z-scores</span></span>
<span><span class="va">z.mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">logTransformed</span><span class="op">)</span>, scale<span class="op">=</span><span class="cn">TRUE</span>, center<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Subsetting few genes of interest</span></span>
<span><span class="va">gois</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_1476300"</span>,<span class="st">"PF3D7_0220800"</span>,<span class="st">"PF3D7_0936200"</span>,<span class="st">"PF3D7_0402200"</span>,<span class="st">"PF3D7_1401600"</span>,<span class="st">"PF3D7_0831600"</span>,<span class="st">"PF3D7_0204100"</span>,<span class="st">"PF3D7_1458300"</span>,<span class="st">"PF3D7_0935800"</span>,<span class="st">"PF3D7_0929400"</span>,<span class="st">"PF3D7_1439000"</span>,<span class="st">"PF3D7_0905400"</span>,<span class="st">"PF3D7_1334600"</span>,<span class="st">"PF3D7_1121300"</span>,<span class="st">"PF3D7_0302500"</span>,<span class="st">"PF3D7_1232800"</span>,<span class="st">"PF3D7_0310400"</span>,<span class="st">"PF3D7_1001600"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Transforming data frame for visualisation</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">z.mat</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">.</span>,<span class="va">GeneID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">gois</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">.</span>,na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/easyExpPlot.html">easyExpPlot</a></span><span class="op">(</span><span class="va">df</span>,x<span class="op">=</span><span class="st">"variable"</span>,y<span class="op">=</span><span class="st">"value"</span>,value<span class="op">=</span><span class="st">"GeneID"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y<span class="op">=</span><span class="st">"Z-scores"</span>,x<span class="op">=</span><span class="st">"Samples"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Miscellaneous_function_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p><code><a href="../reference/easyExpPlot.html">easyExpPlot()</a></code> can be used to plot expression values
obtained from any kind of applied normalization and transformation.</p>
</div>
<div class="section level4">
<h4 id="support-for-noiseq-readdata-function-for-non-model-organism">Support for NOISeq <code>readData</code> function for non-model
organism<a class="anchor" aria-label="anchor" href="#support-for-noiseq-readdata-function-for-non-model-organism"></a>
</h4>
<p>When using <em><a href="https://bioconductor.org/packages/3.19/NOISeq" class="external-link">NOISeq</a></em> for
QC, users are often required to provide various information alongside
count matrices. These information includes biotype, chromosome, gc
content, length of the features etc. These arguments are listed as
optional but are required by <em><a href="https://bioconductor.org/packages/3.19/NOISeq" class="external-link">NOISeq</a></em>
package for making QC plots. To enable users to get these information
from the same reference FASTA file and GFF file used by user for
alignment, we wrote <code><a href="../reference/easyNOISeqAnnot.html">easyNOISeqAnnot()</a></code> function that enable
users to quickly get this information from GTF/GFF file and FASTA file
and return a data frame. These can be provided as a URL or as locally
stored files.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/easyNOISeqAnnot.html">easyNOISeqAnnot</a></span><span class="op">(</span></span>
<span> gff<span class="op">=</span><span class="st">"https://toxodb.org/common/downloads/release-68/EpraecoxHoughton/gff/data/ToxoDB-68_EpraecoxHoughton.gff"</span>,</span>
<span> fasta <span class="op">=</span> <span class="st">"https://toxodb.org/common/downloads/release-68/EpraecoxHoughton/fasta/data/ToxoDB-68_EpraecoxHoughton_Genome.fasta"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene_info</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 gene_id   gene_biotype        gc</span></span>
<span><span class="co">#&gt; EPH_0000010 EPH_0000010 protein_coding 0.5329587</span></span>
<span><span class="co">#&gt; EPH_0000020 EPH_0000020 protein_coding 0.4712575</span></span>
<span><span class="co">#&gt; EPH_0000030 EPH_0000030 protein_coding 0.5618822</span></span>
<span><span class="co">#&gt; EPH_0000040 EPH_0000040 protein_coding 0.5624426</span></span>
<span><span class="co">#&gt; EPH_0000050 EPH_0000050 protein_coding 0.4910354</span></span>
<span><span class="co">#&gt; EPH_0000060 EPH_0000060 protein_coding 0.5811796</span></span>
<span><span class="co">#&gt;                                                                    desc length</span></span>
<span><span class="co">#&gt; EPH_0000010 Helicase associated domain HA2 containing protein , related   6584</span></span>
<span><span class="co">#&gt; EPH_0000020                      DNA polymerase alpha subunit, putative   5010</span></span>
<span><span class="co">#&gt; EPH_0000030                                        hypothetical protein   2529</span></span>
<span><span class="co">#&gt; EPH_0000040                             hypothetical protein, conserved   6534</span></span>
<span><span class="co">#&gt; EPH_0000050                             hypothetical protein, conserved   4462</span></span>
<span><span class="co">#&gt; EPH_0000060                             hypothetical protein, conserved   1509</span></span>
<span><span class="co">#&gt;             starts  ends      chr</span></span>
<span><span class="co">#&gt; EPH_0000010   2602  9185 HG688746</span></span>
<span><span class="co">#&gt; EPH_0000020  50573 55582 HG688746</span></span>
<span><span class="co">#&gt; EPH_0000030    101  2629 HG695046</span></span>
<span><span class="co">#&gt; EPH_0000040  19822 26355 HG697798</span></span>
<span><span class="co">#&gt; EPH_0000050  30217 34678 HG697798</span></span>
<span><span class="co">#&gt; EPH_0000060  36161 37669 HG697798</span></span></code></pre></div>
<p>You can now pass the data frame contents to
<code><a href="https://rdrr.io/pkg/NOISeq/man/readData.html" class="external-link">NOISeq::readData()</a></code> function as follows.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_NOISEQ</span> <span class="op">&lt;-</span> <span class="fu">NOISeq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/NOISeq/man/readData.html" class="external-link">readData</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">counts</span>,</span>
<span>                        length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">gene_info</span><span class="op">$</span><span class="va">length</span>, <span class="va">gene_info</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span>,</span>
<span>                        gc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">gene_info</span><span class="op">$</span><span class="va">gc</span>, <span class="va">gene_info</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span>,</span>
<span>                        biotype<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">gene_info</span><span class="op">$</span><span class="va">gene_biotype</span>, <span class="va">gene_info</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span>,</span>
<span>                        chromosome <span class="op">=</span> <span class="va">gene_info</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">"starts"</span>,<span class="st">"ends"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                        factors <span class="op">=</span> <span class="va">meta</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="check-synchronization-of-bulk-rnaseq-samples-using-easylabeltransfer">Check synchronization of Bulk RNASeq samples using
<code>easyLabelTransfer()</code><a class="anchor" aria-label="anchor" href="#check-synchronization-of-bulk-rnaseq-samples-using-easylabeltransfer"></a>
</h4>
<p><code><a href="../reference/easyLabelTransfer.html">easyLabelTransfer()</a></code> is a wrapper function written around
<em><a href="https://bioconductor.org/packages/3.19/SingleR" class="external-link">SingleR</a></em>
<code><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html" class="external-link">SingleR()</a></code> function to quickly use count matrices from Bulk/
Single cell Reference data sets to check synchronization (time point and
stage) of the Parasite Bulk RNASeq samples and can be used for QC
purposes.</p>
<p>Here, for demonstration purpose, we will use single cell RNASeq data
from Malaria Cell Atlas and use <span class="citation">(<a href="#ref-Subudhi2020">Subudhi et al. 2020</a>)</span> Bulk time-series
dataset to transfer labels and see the distribution of the cells across
different time points.</p>
<p>Other data sets provided with the package includes:</p>
<ul>
<li>Time course Microarray dataset from <span class="citation">(<a href="#ref-painter2018genome">Painter et al. 2018</a>)</span>:
<code>data("painter2018")</code>
</li>
<li>Time course Bulk RNASeq (Single-end) from <span class="citation">(<a href="#ref-toenhake2018chromatin">Toenhake et al. 2018</a>)</span>:
<code>data("toenhake2018")</code>
</li>
<li>Time course Bulk RNASeq (Paired-end) from <span class="citation">(<a href="#ref-Kucharski2020">Kucharski et al. 2020</a>)</span>:
<code>data("kucharski2020")</code>
</li>
<li>Reanalyzed scRNAseq of <em>T. gondii</em> ME49 data from <span class="citation">(<a href="#ref-lou2024single">Lou et al.
2024</a>)</span>: <code>data("gubbels2024")</code>
</li>
<li>Reanalyzed scRNAseq of <em>T. gondii</em> ME49 data from <span class="citation">(<a href="#ref-xue2020single">Xue et al.
2020</a>)</span>: <code>data("boothroyd2020")</code>
</li>
<li>Reanalyzed Bulk RNASeq (Paired-end) from <span class="citation">(<a href="#ref-Wichers2019">Wichers et al. 2019</a>)</span>:
<code>data("wichers2019")</code>
</li>
<li>Reanalyzed scRNAseq of *T. gondii* data from <span class="citation">(<a href="#ref-lou2024single">Lou et al.
2024</a>)</span>: <code>data("gubbels2024")</code>
</li>
<li>Reanalyzed scRNAseq of *T. gondii* data from <span class="citation">(<a href="#ref-xue2020single">Xue et al.
2020</a>)</span>: <code>data("boothroyd2020")</code>
</li>
<li>Reanalyzed Bulk RNASeq (Paired -End) from <span class="citation">(<a href="#ref-Wichers2019">Wichers et al. 2019</a>)</span>:
<code>data("wichers2019")</code>
</li>
</ul>
<p>Next, Malaria cell atlas (MCA) is a database comprising of single
cell data sets and can be accessed using the following functions.</p>
<ol style="list-style-type: decimal">
<li>
<code><a href="../reference/listMCA.html">listMCA()</a></code> function enables you to filter and list
scRNASeq data sets for which data download links are available. User can
use this function to view a table to metadata about the dataset or
download these datasets recursively rather than manually downloading
single dataset at a time.</li>
<li>
<code><a href="../reference/easyMCA.html">easyMCA()</a></code> enables users to import the count matrix and
metadata directly in the R environment directly. Written mainly for
pipeline development.</li>
</ol>
<p>Now, lets use the function enlisted above alongside
<code><a href="../reference/easyLabelTransfer.html">easyLabelTransfer()</a></code> as shown below.</p>
<blockquote>
<p>Note: The output obtained from <code><a href="../reference/easyLabelTransfer.html">easyLabelTransfer()</a></code>
function is a DFrame object and can be fed to other SingleR functions
for visual purposes.</p>
</blockquote>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/SingleR" class="external-link">SingleR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12458</span><span class="op">)</span></span>
<span><span class="co">## Fetching the URL</span></span>
<span><span class="va">mcalist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/listMCA.html">listMCA</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"subudhi2020"</span><span class="op">)</span> <span class="co">## reference dataset</span></span>
<span><span class="va">subudhi2020</span> <span class="op">&lt;-</span> <span class="va">subudhi2020</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">42</span><span class="op">]</span> <span class="co">## Skipping samples after 43 Hpi since they highly correlate with 43hpi itself.</span></span>
<span></span>
<span><span class="co">## Using this reference set</span></span>
<span><span class="va">url</span> <span class="op">&lt;-</span> <span class="st">"https://www.malariacellatlas.org/downloads/pf-ch10x-set4-biorxiv.zip"</span></span>
<span></span>
<span><span class="va">raw_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/easyMCA.html">easyMCA</a></span><span class="op">(</span><span class="va">url</span>,type <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">raw_counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"-"</span>,<span class="st">"_"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">raw_counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/easyMCA.html">easyMCA</a></span><span class="op">(</span><span class="va">url</span>,type<span class="op">=</span><span class="st">"data"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Retaining only Asexual stage cells and Lab isolates.</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">meta</span>, <span class="va">meta</span><span class="op">$</span><span class="va">STAGE_LR</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ring"</span>,<span class="st">"trophozoite"</span>,<span class="st">"schizont"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">DAY</span> <span class="op">!=</span> <span class="st">"Field"</span><span class="op">)</span></span>
<span><span class="va">raw_counts</span> <span class="op">&lt;-</span> <span class="va">raw_counts</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Filtering away field isolates and sexual stage cells</span></span>
<span></span>
<span><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/easyLabelTransfer.html">easyLabelTransfer</a></span><span class="op">(</span>queryCounts <span class="op">=</span> <span class="va">raw_counts</span>,</span>
<span>                            refCounts <span class="op">=</span> <span class="va">subudhi2020</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">counts</span>, referenceMeta <span class="op">=</span> <span class="va">subudhi2020</span><span class="op">@</span><span class="va">colData</span>, labelCol <span class="op">=</span> <span class="st">"timetag"</span>, isrefBulk <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">labels</span><span class="op">$</span><span class="va">pruned.labels</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 11hpi 13hpi 15hpi 17hpi 19hpi 21hpi 23hpi 25hpi 27hpi 29hpi 31hpi 33hpi 35hpi </span></span>
<span><span class="co">#&gt;   762   615   396   698   155  2126    65  3027    18  3855    14  2882     3 </span></span>
<span><span class="co">#&gt; 37hpi 39hpi  3hpi 41hpi 43hpi  5hpi  7hpi  9hpi </span></span>
<span><span class="co">#&gt;   375  2066  2361  1579  5727  1199   241   336</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">labels</span><span class="op">$</span><span class="va">labels</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">.</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">.</span>, levels <span class="op">=</span> <span class="va">subudhi2020</span><span class="op">$</span><span class="va">timetag</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggbarplot.html" class="external-link">ggbarplot</a></span><span class="op">(</span><span class="va">df</span>,x<span class="op">=</span><span class="st">"."</span>,y<span class="op">=</span><span class="st">"Freq"</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"#2595be"</span>, fill <span class="op">=</span> <span class="st">"#2595be"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">100</span>,colour <span class="op">=</span> <span class="st">"red"</span>, linetype<span class="op">=</span><span class="fl">2</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Time-points"</span>,y<span class="op">=</span><span class="st">"No. of Cells"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Miscellaneous_function_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/plotScoreHeatmap.html" class="external-link">plotScoreHeatmap</a></span><span class="op">(</span><span class="va">labels</span>,show_colnames <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                       show.pruned <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                       cluster_cols <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                       cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                       rows.order<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">subudhi2020</span><span class="op">$</span><span class="va">timetag</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Miscellaneous_function_files/figure-html/unnamed-chunk-10-2.png" width="700"></p>
<p>You can see that cells from some time points are underrepresented in
our Malaria Cell Atlas.</p>
</div>
<div class="section level4">
<h4 id="go-enrichment-using-easytopgo">GO Enrichment using <code>easytopGO()</code><a class="anchor" aria-label="anchor" href="#go-enrichment-using-easytopgo"></a>
</h4>
<p>VEuPathDB constituent database allow users to perform GO Enrichment
analysis. However, there is no option to set the background genes and by
default all the genes in the are used. This is not an ideal approach.
Moreover, since Gene Ontologies are curated using different pipelines by
different enrichment providers, the results might differ just based on
source of Ontologies. Similarly the Ontology data might not be available
in other databases such as BioMart and you would like to use custom GAF
file.</p>
<p>Users of parasite domain might wish to use GAF files provided by
VEuPathDB constituent database and of specific genome assembly as per
their hypothesis. To enable users to do that, GO enrichment analysis can
be performed easily and quickly using our <code><a href="../reference/easytopGO.html">easytopGO()</a></code>
wrapper function. This function requires users to provide a named
numeric vector, where names are Ensembl Gene IDs and numeric vector
could be adjusted p-values. Users can also supply genes to be used as
background using <code>bkggset</code> argument. Finally users can use
<code>gaf</code> argument to supply .gaf file obtained from VEuPathDBs
constituent database of interest.</p>
<p>For well known organisms, Biomart Ontologies can also be used. In
such case user has to provide <code>mart</code> argument and figure out
key for emsembl gene IDs . However, many times the parasite assemblies
used as reference by Ensembl might not match with parasite assembly used
by VEuPathDB. In such cases there might be mismatches between gene IDs
(see one such issue <a href="https://github.com/Huber-group-EMBL/biomaRt/issues/110" class="external-link">here</a>).</p>
<p>You can also change <code>category</code> argument to specify which
GO category you want to test enrichment for. Since for most of the
parasites, sub-cellular localization and Molecular function information
is sparingly available, “BP” has been set as the default.</p>
<p>Finally, you can also use ORA results obtained from VEuPathDBs as
input to <code><a href="../reference/easytopGO.html">easytopGO()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Making a numeric vector of padjusted values</span></span>
<span><span class="va">gois</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PF3D7_0102200"</span>,<span class="st">"PF3D7_0207400"</span>,<span class="st">"PF3D7_0207500"</span>,<span class="st">"PF3D7_0207600"</span>,<span class="st">"PF3D7_0207700"</span>,<span class="st">"PF3D7_0207800"</span>,<span class="st">"PF3D7_0207900"</span>,<span class="st">"PF3D7_0208000"</span>,<span class="st">"PF3D7_0404700"</span>,<span class="st">"PF3D7_0501500"</span>,<span class="st">"PF3D7_0502400"</span>,<span class="st">"PF3D7_0618000"</span>,<span class="st">"PF3D7_0731800"</span>,<span class="st">"PF3D7_0930300"</span>,<span class="st">"PF3D7_1116000"</span>,<span class="st">"PF3D7_1247800"</span>,<span class="st">"PF3D7_1334700"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">p_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.767929e-49</span>,<span class="fl">3.886063e-148</span>,<span class="fl">6.459285e-269</span>,<span class="fl">0.000000e+00</span>,<span class="fl">6.842121e-132</span>,</span>
<span><span class="fl">5.282318e-178</span>,<span class="fl">1.140221e-224</span>,<span class="fl">1.625665e-100</span>,<span class="fl">6.177129e-08</span>,<span class="fl">1.484155e-03</span>,<span class="fl">2.789648e-10</span>,<span class="fl">2.300720e-12</span>,<span class="fl">7.450697e-30</span>,<span class="fl">3.417972e-89</span>,<span class="fl">4.846689e-73</span>,<span class="fl">1.207966e-11</span>,<span class="fl">4.173777e-51</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">p_values</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gois</span></span>
<span></span>
<span><span class="co">## Using all genes captured as a background</span></span>
<span></span>
<span><span class="va">background.gset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">subudhi2020</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></span>
<span><span class="va">baseurl</span> <span class="op">&lt;-</span> <span class="st">"https://plasmodb.org/common/downloads/Current_Release/"</span></span>
<span><span class="va">url</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">baseurl</span>,<span class="st">"Pfalciparum3D7/gaf/PlasmoDB-68_Pfalciparum3D7_GO.gaf.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Performing ORA</span></span>
<span><span class="va">gores</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/easytopGO.html">easytopGO</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="va">p_values</span>,useGAF <span class="op">=</span> <span class="cn">TRUE</span>,useBiomart <span class="op">=</span> <span class="cn">FALSE</span>,gaf<span class="op">=</span><span class="va">url</span>,</span>
<span>bkggset <span class="op">=</span> <span class="va">background.gset</span>, category <span class="op">=</span> <span class="st">"BP"</span>, stats <span class="op">=</span> <span class="st">"ks"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Plotting the results. </span></span>
<span><span class="va">goplt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/easyGOPlot.html">easyGOPlot</a></span><span class="op">(</span><span class="va">gores</span>, title <span class="op">=</span> <span class="st">"GO Biological processes"</span>, limit <span class="op">=</span> <span class="fl">20</span>, sortby <span class="op">=</span> <span class="st">"ks"</span><span class="op">)</span></span>
<span><span class="va">goplt</span></span></code></pre></div>
<p><img src="Miscellaneous_function_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Making circular plot</span></span>
<span><span class="va">goplt</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_polar.html" class="external-link">coord_radial</a></span><span class="op">(</span>inner.radius <span class="op">=</span> <span class="fl">0.3</span>, rotate.angle <span class="op">=</span> <span class="cn">TRUE</span>,r_axis_inside <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span><span class="op">+</span><span class="fu">ggeasy</span><span class="fu">::</span><span class="fu"><a href="https://jonocarroll.github.io/ggeasy/reference/easy_change_legend.html" class="external-link">easy_legend_at</a></span><span class="op">(</span><span class="st">"bottom"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Miscellaneous_function_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<p>The <code><a href="../reference/easyGOPlot.html">easyGOPlot()</a></code> is adapted and modified version of what
has been provided by <a href="https://www.biostars.org/p/471549/" class="external-link">Kevin
Blighe</a> on Biostars. You can use <code>limit</code> argument to limit
number of terms to be plotted.</p>
</div>
<div class="section level4">
<h4 id="screening-the-signal-peptides-from-pdbs">Screening the Signal Peptides from PDBs<a class="anchor" aria-label="anchor" href="#screening-the-signal-peptides-from-pdbs"></a>
</h4>
<p>We also offer a convenient function, <code><a href="../reference/easyAF2Signal.html">easyAF2Signal()</a></code>, to
diagnose potential false-positive signal peptides using SignalP
predictions from the VEuPathDB database and AlphaFold2 structures, based
on an observational study conducted by <span class="citation">(<a href="#ref-Sanaboyana2024">Sanaboyana and Elcock 2024</a>)</span>. The
authors found that true N-terminal signal peptides (~24-24 amino acids)
are typically disengaged from the protein body and lack atomic contacts,
and AlphaFold2 attempts to model them in a similar way.</p>
<p><code><a href="../reference/easyAF2Signal.html">easyAF2Signal()</a></code> is an R equivalent with slight
modifications to the Fortran code provided by the authors, which reports
additional information, such as the number of residues remaining after
pLDDT filtering.</p>
<p>We did this modification because, if all the signal peptide residues
have low pLDDT scores and are filtered out, no residues will remain to
calculate contacts with the protein body, resulting in zero contacts.
This could give a false impression that the first 25 amino acids are
signal peptides, but it cannot be confirmed, as the zero-contact
observed is due to no residues remaining after filtering, not because
the signal peptide was disengaged from the protein body.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/easyAF2Signal.html">easyAF2Signal</a></span><span class="op">(</span><span class="st">"https://alphafold.ebi.ac.uk/files/AF-Q9TY95-F1-model_v4.pdb"</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    Name length_signalpeptide length_protein</span></span>
<span><span class="co">#&gt; 1 AF-Q9TY95-F1-model_v4                   25            997</span></span>
<span><span class="co">#&gt;   postbfacFiltered_signalP_res medianSignalP_bfac clevagesite_bfac</span></span>
<span><span class="co">#&gt; 1                            0              35.36            32.71</span></span>
<span><span class="co">#&gt;   postbfacFiltered_rest_res medianRest_bfac res_res_conts atm_atm_conts</span></span>
<span><span class="co">#&gt; 1                       308           81.67             0             0</span></span></code></pre></div>
<p>In the example above, we see that signal peptide has low mean pLDDT
score and therefore there are no residues left post filtering of bad
quality residues leading to zero residue-residue counts
<code>res_res_conts</code>. Thus we can’t say for sure if the PDB in
hand have signal peptide or not.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 26100)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_India.utf8  LC_CTYPE=English_India.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_India.utf8 LC_NUMERIC=C                  </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_India.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Riyadh</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] topGO_2.56.0                SparseM_1.84-2             </span></span>
<span><span class="co">#&gt;  [3] GO.db_3.19.1                AnnotationDbi_1.66.0       </span></span>
<span><span class="co">#&gt;  [5] graph_1.82.0                SingleR_2.6.0              </span></span>
<span><span class="co">#&gt;  [7] SummarizedExperiment_1.34.0 Biobase_2.64.0             </span></span>
<span><span class="co">#&gt;  [9] GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt; [11] IRanges_2.38.1              S4Vectors_0.42.1           </span></span>
<span><span class="co">#&gt; [13] BiocGenerics_0.50.0         MatrixGenerics_1.16.0      </span></span>
<span><span class="co">#&gt; [15] matrixStats_1.5.0           ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt; [17] tibble_3.3.0                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [19] dplyr_1.1.4                 plasmoRUtils_1.1.0         </span></span>
<span><span class="co">#&gt; [21] rlang_1.1.6                 readr_2.1.5                </span></span>
<span><span class="co">#&gt; [23] janitor_2.2.1               BiocStyle_2.32.1           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] dichromat_2.0-0.1           vroom_1.6.5                </span></span>
<span><span class="co">#&gt;   [3] progress_1.2.3              vsn_3.72.0                 </span></span>
<span><span class="co">#&gt;   [5] nnet_7.3-19                 Biostrings_2.72.1          </span></span>
<span><span class="co">#&gt;   [7] vctrs_0.6.5                 digest_0.6.37              </span></span>
<span><span class="co">#&gt;   [9] png_0.1-8                   proxy_0.4-27               </span></span>
<span><span class="co">#&gt;  [11] MSnbase_2.30.1              parallelly_1.45.1          </span></span>
<span><span class="co">#&gt;  [13] MASS_7.3-61                 pkgdown_2.1.3              </span></span>
<span><span class="co">#&gt;  [15] reshape2_1.4.4              foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [17] withr_3.0.2                 xfun_0.52                  </span></span>
<span><span class="co">#&gt;  [19] ggpubr_0.6.1                survival_3.8-3             </span></span>
<span><span class="co">#&gt;  [21] memoise_2.0.1               hexbin_1.28.5              </span></span>
<span><span class="co">#&gt;  [23] ggsci_3.2.0                 mixtools_2.0.0.1           </span></span>
<span><span class="co">#&gt;  [25] systemfonts_1.2.3           ragg_1.4.0                 </span></span>
<span><span class="co">#&gt;  [27] gtools_3.9.5                easyPubMed_2.13            </span></span>
<span><span class="co">#&gt;  [29] Formula_1.2-5               prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt;  [31] KEGGREST_1.44.1             promises_1.3.3             </span></span>
<span><span class="co">#&gt;  [33] httr_1.4.7                  rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [35] restfulr_0.0.16             globals_0.18.0             </span></span>
<span><span class="co">#&gt;  [37] ps_1.9.1                    rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;  [39] UCSC.utils_1.0.0            generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [41] processx_3.8.6              curl_6.4.0                 </span></span>
<span><span class="co">#&gt;  [43] ncdf4_1.24                  zlibbioc_1.50.0            </span></span>
<span><span class="co">#&gt;  [45] ScaledMatrix_1.12.0         randomForest_4.7-1.2       </span></span>
<span><span class="co">#&gt;  [47] bio3d_2.4-5                 GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt;  [49] SparseArray_1.4.8           xtable_1.8-4               </span></span>
<span><span class="co">#&gt;  [51] stringr_1.5.1               desc_1.4.3                 </span></span>
<span><span class="co">#&gt;  [53] doParallel_1.0.17           evaluate_1.0.4             </span></span>
<span><span class="co">#&gt;  [55] S4Arrays_1.4.1              BiocFileCache_2.12.0       </span></span>
<span><span class="co">#&gt;  [57] preprocessCore_1.66.0       hms_1.1.3                  </span></span>
<span><span class="co">#&gt;  [59] bookdown_0.43               irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [61] colorspace_2.1-1            filelock_1.0.3             </span></span>
<span><span class="co">#&gt;  [63] magrittr_2.0.3              snakecase_0.11.1           </span></span>
<span><span class="co">#&gt;  [65] later_1.4.2                 viridis_0.6.5              </span></span>
<span><span class="co">#&gt;  [67] lattice_0.22-6              MsCoreUtils_1.16.1         </span></span>
<span><span class="co">#&gt;  [69] future.apply_1.20.0         XML_3.99-0.18              </span></span>
<span><span class="co">#&gt;  [71] scuttle_1.14.0              class_7.3-22               </span></span>
<span><span class="co">#&gt;  [73] pillar_1.11.0               nlme_3.1-166               </span></span>
<span><span class="co">#&gt;  [75] iterators_1.0.14            compiler_4.4.1             </span></span>
<span><span class="co">#&gt;  [77] beachmat_2.20.0             stringi_1.8.7              </span></span>
<span><span class="co">#&gt;  [79] gower_1.0.2                 dendextend_1.19.1          </span></span>
<span><span class="co">#&gt;  [81] lubridate_1.9.4             GenomicAlignments_1.40.0   </span></span>
<span><span class="co">#&gt;  [83] drawProteins_1.24.0         crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [85] abind_1.4-8                 BiocIO_1.14.0              </span></span>
<span><span class="co">#&gt;  [87] bit_4.6.0                   chromote_0.5.1             </span></span>
<span><span class="co">#&gt;  [89] pcaMethods_1.96.0           codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [91] textshaping_1.0.1           recipes_1.3.1              </span></span>
<span><span class="co">#&gt;  [93] BiocSingular_1.20.0         MLInterfaces_1.84.0        </span></span>
<span><span class="co">#&gt;  [95] bslib_0.9.0                 e1071_1.7-16               </span></span>
<span><span class="co">#&gt;  [97] plotly_4.11.0               LaplacesDemon_16.1.6       </span></span>
<span><span class="co">#&gt;  [99] MultiAssayExperiment_1.30.3 splines_4.4.1              </span></span>
<span><span class="co">#&gt; [101] Rcpp_1.1.0                  dbplyr_2.5.0               </span></span>
<span><span class="co">#&gt; [103] sparseMatrixStats_1.16.0    knitr_1.50                 </span></span>
<span><span class="co">#&gt; [105] blob_1.2.4                  utf8_1.2.6                 </span></span>
<span><span class="co">#&gt; [107] clue_0.3-66                 mzR_2.38.0                 </span></span>
<span><span class="co">#&gt; [109] AnnotationFilter_1.28.0     fs_1.6.6                   </span></span>
<span><span class="co">#&gt; [111] QFeatures_1.14.2            listenv_0.9.1              </span></span>
<span><span class="co">#&gt; [113] mzID_1.42.0                 DelayedMatrixStats_1.26.0  </span></span>
<span><span class="co">#&gt; [115] ggsignif_0.6.4              Matrix_1.7-1               </span></span>
<span><span class="co">#&gt; [117] statmod_1.5.0               tzdb_0.5.0                 </span></span>
<span><span class="co">#&gt; [119] lpSolve_5.6.23              pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [121] pheatmap_1.0.13             tools_4.4.1                </span></span>
<span><span class="co">#&gt; [123] cachem_1.1.0                RSQLite_2.4.1              </span></span>
<span><span class="co">#&gt; [125] viridisLite_0.4.2           rvest_1.0.4                </span></span>
<span><span class="co">#&gt; [127] DBI_1.2.3                   impute_1.78.0              </span></span>
<span><span class="co">#&gt; [129] fastmap_1.2.0               rmarkdown_2.29             </span></span>
<span><span class="co">#&gt; [131] scales_1.4.0                grid_4.4.1                 </span></span>
<span><span class="co">#&gt; [133] gt_1.0.0                    Rsamtools_2.20.0           </span></span>
<span><span class="co">#&gt; [135] broom_1.0.8                 sass_0.4.10                </span></span>
<span><span class="co">#&gt; [137] coda_0.19-4.1               FNN_1.1.4.1                </span></span>
<span><span class="co">#&gt; [139] BiocManager_1.30.26         carData_3.0-5              </span></span>
<span><span class="co">#&gt; [141] rpart_4.1.23                farver_2.1.2               </span></span>
<span><span class="co">#&gt; [143] yaml_2.3.10                 AnnotationForge_1.46.0     </span></span>
<span><span class="co">#&gt; [145] rtracklayer_1.64.0          cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [147] purrr_1.1.0                 txdbmaker_1.0.1            </span></span>
<span><span class="co">#&gt; [149] lifecycle_1.0.4             caret_7.0-1                </span></span>
<span><span class="co">#&gt; [151] mvtnorm_1.3-3               lava_1.8.1                 </span></span>
<span><span class="co">#&gt; [153] kernlab_0.9-33              backports_1.5.0            </span></span>
<span><span class="co">#&gt; [155] BiocParallel_1.38.0         annotate_1.82.0            </span></span>
<span><span class="co">#&gt; [157] timechange_0.3.0            gtable_0.3.6               </span></span>
<span><span class="co">#&gt; [159] rjson_0.2.23                parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [161] pROC_1.18.5                 limma_3.60.6               </span></span>
<span><span class="co">#&gt; [163] jsonlite_2.0.0              bitops_1.0-9               </span></span>
<span><span class="co">#&gt; [165] bit64_4.6.0-1               pRoloc_1.44.1              </span></span>
<span><span class="co">#&gt; [167] jquerylib_0.1.4             segmented_2.1-4            </span></span>
<span><span class="co">#&gt; [169] timeDate_4041.110           lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [171] htmltools_0.5.8.1           affy_1.82.0                </span></span>
<span><span class="co">#&gt; [173] rappdirs_0.3.3              glue_1.8.0                 </span></span>
<span><span class="co">#&gt; [175] polyglotr_1.7.0             httr2_1.2.0                </span></span>
<span><span class="co">#&gt; [177] XVector_0.44.0              RCurl_1.98-1.17            </span></span>
<span><span class="co">#&gt; [179] MALDIquant_1.22.3           mclust_6.1.1               </span></span>
<span><span class="co">#&gt; [181] BSgenome_1.72.0             gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [183] igraph_2.1.4                R6_2.6.1                   </span></span>
<span><span class="co">#&gt; [185] tidyr_1.3.1                 SingleCellExperiment_1.26.0</span></span>
<span><span class="co">#&gt; [187] labeling_0.4.3              GenomicFeatures_1.56.0     </span></span>
<span><span class="co">#&gt; [189] cluster_2.1.8               ipred_0.9-15               </span></span>
<span><span class="co">#&gt; [191] DelayedArray_0.30.1         tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt; [193] ProtGenerics_1.36.0         sampling_2.11              </span></span>
<span><span class="co">#&gt; [195] xml2_1.3.8                  car_3.1-3                  </span></span>
<span><span class="co">#&gt; [197] future_1.67.0               ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">#&gt; [199] rsvd_1.0.5                  affyio_1.74.0              </span></span>
<span><span class="co">#&gt; [201] data.table_1.17.8           ggeasy_0.1.6               </span></span>
<span><span class="co">#&gt; [203] websocket_1.4.4             mgsub_1.7.3                </span></span>
<span><span class="co">#&gt; [205] htmlwidgets_1.6.4           RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt; [207] biomaRt_2.60.1              hardhat_1.4.1              </span></span>
<span><span class="co">#&gt; [209] prodlim_2025.04.28          PSMatch_1.8.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Kucharski2020" class="csl-entry">
Kucharski, Michal, Jaishree Tripathi, Sourav Nayak, Lei Zhu, Grennady
Wirjanata, Rob W. van der Pluijm, Mehul Dhorda, Arjen Dondorp, and
Zbynek Bozdech. 2020. <span>“A Comprehensive RNA Handling and
Transcriptomics Guide for High-Throughput Processing of Plasmodium
Blood-Stage Samples.”</span> <em>Malaria Journal</em> 19 (1). <a href="https://doi.org/10.1186/s12936-020-03436-w" class="external-link">https://doi.org/10.1186/s12936-020-03436-w</a>.
</div>
<div id="ref-lou2024single" class="csl-entry">
Lou, Jingjing, Yasaman Rezvani, Argenis Arriojas, Yihan Wu, Nachiket
Shankar, David Degras, Caroline D Keroack, Manoj T Duraisingh, Kourosh
Zarringhalam, and Marc-Jan Gubbels. 2024. <span>“Single Cell Expression
and Chromatin Accessibility of the Toxoplasma Gondii Lytic Cycle
Identifies AP2XII-8 as an Essential Ribosome Regulon Driver.”</span>
<em>Nature Communications</em> 15 (1): 7419.
</div>
<div id="ref-painter2018genome" class="csl-entry">
Painter, Heather J, Neo Christopher Chung, Aswathy Sebastian, Istvan
Albert, John D Storey, and Manuel Llinás. 2018. <span>“Genome-Wide
Real-Time in Vivo Transcriptional Dynamics During Plasmodium Falciparum
Blood-Stage Development.”</span> <em>Nature Communications</em> 9 (1):
2656.
</div>
<div id="ref-Sanaboyana2024" class="csl-entry">
Sanaboyana, Venkata R., and Adrian H. Elcock. 2024. <span>“Improving
Signal and Transit Peptide Predictions Using AlphaFold2-Predicted
Protein Structures.”</span> <em>Journal of Molecular Biology</em> 436
(2): 168393. <a href="https://doi.org/10.1016/j.jmb.2023.168393" class="external-link">https://doi.org/10.1016/j.jmb.2023.168393</a>.
</div>
<div id="ref-Subudhi2020" class="csl-entry">
Subudhi, Amit K., Aidan J. O’Donnell, Abhinay Ramaprasad, Hussein M.
Abkallo, Abhinav Kaushik, Hifzur R. Ansari, Alyaa M. Abdel-Haleem, et
al. 2020. <span>“Malaria Parasites Regulate Intra-Erythrocytic
Development Duration via Serpentine Receptor 10 to Coordinate with Host
Rhythms.”</span> <em>Nature Communications</em> 11 (1). <a href="https://doi.org/10.1038/s41467-020-16593-y" class="external-link">https://doi.org/10.1038/s41467-020-16593-y</a>.
</div>
<div id="ref-toenhake2018chromatin" class="csl-entry">
Toenhake, Christa Geeke, Sabine Anne-Kristin Fraschka, Mahalingam
Shanmugiah Vijayabaskar, David Robert Westhead, Simon Jan van Heeringen,
and Richárd Bártfai. 2018. <span>“Chromatin Accessibility-Based
Characterization of the Gene Regulatory Network Underlying Plasmodium
Falciparum Blood-Stage Development.”</span> <em>Cell Host &amp;
Microbe</em> 23 (4): 557–69.
</div>
<div id="ref-Wichers2019" class="csl-entry">
Wichers, J. Stephan, Judith A. M. Scholz, Jan Strauss, Susanne Witt,
Andrés Lill, Laura-Isabell Ehnold, Niklas Neupert, et al. 2019.
<span>“Dissecting the Gene Expression, Localization, Membrane Topology,
and Function of the Plasmodium Falciparum STEVOR Protein Family.”</span>
Edited by John C. Boothroyd. <em>mBio</em> 10 (4). <a href="https://doi.org/10.1128/mbio.01500-19" class="external-link">https://doi.org/10.1128/mbio.01500-19</a>.
</div>
<div id="ref-xue2020single" class="csl-entry">
Xue, Yuan, Terence C Theisen, Suchita Rastogi, Abel Ferrel, Stephen R
Quake, and John C Boothroyd. 2020. <span>“A Single-Parasite
Transcriptional Atlas of Toxoplasma Gondii Reveals Novel Control of
Antigen Expression.”</span> <em>Elife</em> 9: e54129.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Rohit Satyam, Alberto Maillo, David Gomez-Cabrero, Arnab Pain.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
