<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>RNASeq: Need for reanalysis? • plasmoRUtils</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="RNASeq: Need for reanalysis?">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">plasmoRUtils</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Introduction_to_plasmoRUtils.html">Introduction to plasmoRUtils</a></li>
    <li><a class="dropdown-item" href="../articles/Gene_ID_Conversion.html">Accessing component databases of VEuPathDB</a></li>
    <li><a class="dropdown-item" href="../articles/Miscellaneous_function.html">Other useful functions</a></li>
    <li><a class="dropdown-item" href="../articles/Need_for_reanalysis.html">RNASeq: Need for reanalysis?</a></li>
    <li><a class="dropdown-item" href="../articles/retrieving_orthologs.html">Retrieving Orthologs</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/rohit-satyam/plasmoRUtils"><span class="fa fab fa-github"></span> </a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>RNASeq: Need for reanalysis?</h1>
                        <h4 data-toc-skip class="author">Rohit
Satyam</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Alberto
Maillo</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">David
Gomez-Cabrero</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="author">Arnab
Pain</h4>
            <address class="author_afil">
      King Abdullah University of Science &amp; Technology, Saudi
Arabia<br><h4 data-toc-skip class="date">22 August, 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/Rohit-Satyam/plasmoRUtils/blob/HEAD/vignettes/Need_for_reanalysis.Rmd" class="external-link"><code>vignettes/Need_for_reanalysis.Rmd</code></a></small>
      <div class="d-none name"><code>Need_for_reanalysis.Rmd</code></div>
    </address>
</address>
</address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This section examines how annotations evolve across successive
      VEuPathDB releases. As exon–intron boundaries are continually
      refined with incoming sequencing data, tracking shifts in
      effective gene length enables users to decide whether existing
      published datasets warrant reanalysis based on the magnitude of
      those changes.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In the bioinformatics community, it is common to reuse published
RNA-seq count or normalized count data without reprocessing, often
overlooking important technical differences such as the version of
genome annotation. Assembly updates typically include revisions to:</p>
<ul>
<li>
<strong>GFF files</strong>, where gene models and exon‐intron
structures are updated, UTRs redefined etc.</li>
<li>
<strong>GAF files</strong>, where GO terms are revised based on
newly published evidence</li>
</ul>
<p>For RNA-seq, these updates can substantially affect downstream
analyses (e.g., over-representation or pathway enrichment). These
changes are often documented in the <strong>NEWS</strong> sections of
VEuPathDB.</p>
<p>For instance, in <em>Plasmodium</em> and <em>Toxoplasma</em>, two key
assembly updates were:</p>
<ul>
<li>
<a href="https://plasmodb.org/plasmo/app/static-content/PlasmoDB/news.html#PlasmoDB51Released" class="external-link">Release
51 (16 Mar 2021)</a> for <strong>PlasmoDB</strong>
</li>
<li>
<a href="https://toxodb.org/toxo/app/static-content/ToxoDB/news.html#ToxoDB66Released" class="external-link">Release
66 (28 Nov 2023)</a> for <strong>ToxoDB</strong>, which included
extensive gene‐model revisions</li>
</ul>
<p>From an RNA-seq perspective, here we will focus on two factors that
are known to affect biological outcomes and mappability. These include
changes in:</p>
<ol style="list-style-type: decimal">
<li><strong>Number of Genes</strong></li>
<li>
<strong>Exon‐intron boundaries:</strong> Measured here as
<strong>effective gene length</strong> (the sum of exon lengths)</li>
</ol>
<p>In the example below, we will use <em>Plasmodium</em> and
<em>Toxoplasma</em> releases and the <em>plasmoRUtils</em> function
<code><a href="../reference/getEffLen.html">getEffLen()</a></code> to track how effective gene lengths hift across
versions.</p>
</div>
<div class="section level2">
<h2 id="assessing-changes-in-gene-number">Assessing Changes in Gene Number<a class="anchor" aria-label="anchor" href="#assessing-changes-in-gene-number"></a>
</h2>
<p>We will compare six releases
(<code>24, 30, 40, 50, 60, and 68</code>) for both <em>Plasmodium</em>
and <em>Toxoplasma</em>. From the NEWS section, we know that in
<em>Plasmodium</em>, several loci initially annotated as rRNAs, lncRNAs,
or coding genes were later removed, and the mitochondrial genome of
PF3D7 was revised in <a href="https://pubmed.ncbi.nlm.nih.gov/31080894/" class="external-link">2019</a>. Similarly,
<em>Toxoplasma</em> TGME49 gene models underwent extensive updates in
release 66.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rohit-satyam.github.io/plasmoRUtils/">plasmoRUtils</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GeneStructureTools</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">releases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">24</span>,<span class="fl">30</span>,<span class="fl">40</span>,<span class="fl">50</span>,<span class="fl">60</span>,<span class="fl">68</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Getching GFFs from PlasmoDB and calculating effective lengths</span></span>
<span><span class="va">pf3d7.gffs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">releases</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://plasmodb.org/common/downloads/release-"</span>,<span class="va">x</span>,<span class="st">"/Pfalciparum3D7/gff/data/PlasmoDB-"</span>,<span class="va">x</span>,<span class="st">"_Pfalciparum3D7.gff"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/getEffLen.html">getEffLen</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">releases</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## Getching GFFs from ToxoDB and calculating effective lengths</span></span>
<span><span class="va">tgme49.gffs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">releases</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://toxodb.org/common/downloads/release-"</span>,<span class="va">x</span>,<span class="st">"/TgondiiME49/gff/data/ToxoDB-"</span>,<span class="va">x</span>,<span class="st">"_TgondiiME49.gff"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/getEffLen.html">getEffLen</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">releases</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Let's plot changes in number of genes </span></span>
<span><span class="va">pf3d7.ngenes</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">pf3d7.gffs</span>, <span class="va">nrow</span><span class="op">)</span>  <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">releases</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/ldply.html" class="external-link">ldply</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tgme49.ngenes</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">tgme49.gffs</span>, <span class="va">nrow</span><span class="op">)</span>  <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">releases</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/ldply.html" class="external-link">ldply</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggline.html" class="external-link">ggline</a></span><span class="op">(</span><span class="va">pf3d7.ngenes</span>,x<span class="op">=</span><span class="st">".id"</span>,y<span class="op">=</span><span class="st">"V1"</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"#2595be"</span>, fill <span class="op">=</span> <span class="st">"#2595be"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Releases"</span>,y<span class="op">=</span><span class="st">"No. of Genes"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Need_for_reanalysis_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggline.html" class="external-link">ggline</a></span><span class="op">(</span><span class="va">tgme49.ngenes</span>,x<span class="op">=</span><span class="st">".id"</span>,y<span class="op">=</span><span class="st">"V1"</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"#2595be"</span>, fill <span class="op">=</span> <span class="st">"#2595be"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Releases"</span>,y<span class="op">=</span><span class="st">"No. of Genes"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Need_for_reanalysis_files/figure-html/unnamed-chunk-2-2.png" width="700"></p>
<p>For <em>Plasmodium</em>, the current release has 5720 genes. 57 genes
have been removed since Release 24. Similarly 142 genes have been
removed from <em>Toxoplasma</em> as compared to previous releases.</p>
</div>
<div class="section level2">
<h2 id="assessing-the-difference-in-effective-gene-lengths">Assessing the difference in Effective gene lengths<a class="anchor" aria-label="anchor" href="#assessing-the-difference-in-effective-gene-lengths"></a>
</h2>
<p>To get intuition about counts of which genes will be affected mostly,
we will calculate coefficient of variation where:</p>
<p><code>CV = (Standard Deviation / Mean) × 100</code></p>
<p>If the effective gene length of the gene didn’t change across
releases, the CV will be equal to zero. Similarly if the change in
effective length is major for a gene, the CV value will be higher. The
CV also gives us intuition about which genes underwent drastic changes;
eg: a 100 bp change will matter more for a 200 bp gene than for a 5000
bp gene.</p>
<p>We will calculate CV using following steps:</p>
<ol style="list-style-type: decimal">
<li>Since gene IDs might vary across releases, we will first convert old
Gene IDs to new gene IDs using <code><a href="../reference/toGeneid.html">toGeneid()</a></code> and retain IDs
which are common across all releases.</li>
<li>In old annotations like release 24, some genes coding for multiple
transcripts are present as separate gene entries each coding for 1
transcripts often denoted by versions before them such as
<code>PF3D7_0618900.1,F3D7_0618900.2</code>. This issue has been fixed
in latest releases. To handle this we will remove the versions from gene
IDs and take the average effective length for such genes.</li>
<li>We will than calculate the CV and filter all the genes with
CV&gt;0.</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">###  Step1</span></span>
<span><span class="co"># Remove version suffix from GeneIDs</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">pf3d7.gffs</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/ldply.html" class="external-link">ldply</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>GeneID <span class="op">=</span> <span class="fu">GeneStructureTools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneStructureTools/man/removeVersion.html" class="external-link">removeVersion</a></span><span class="op">(</span><span class="va">GeneID</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get annotation table and identify outdated IDs</span></span>
<span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTable.html">getTable</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"Plasmodium falciparum 3D7"</span>, db <span class="op">=</span> <span class="st">"plasmodb"</span><span class="op">)</span></span>
<span><span class="va">old</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">GeneID</span><span class="op">)</span>, <span class="va">annot</span><span class="op">$</span><span class="va">`Gene ID`</span><span class="op">)</span> <span class="co">## 152 old IDs</span></span>
<span></span>
<span><span class="co"># Map outdated to Ensembl IDs</span></span>
<span><span class="va">new_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span><span class="va">old</span>, from <span class="op">=</span> <span class="st">"old"</span>, to <span class="op">=</span> <span class="st">"ensembl"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`Gene ID`</span>, <span class="va">`Previous ID(s)`</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="co">## only 62 mapped</span></span>
<span></span>
<span><span class="co"># Replace outdated IDs in df. These are mostly apicoplast genes</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">GeneID</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">new_ids</span><span class="op">$</span><span class="va">`Gene ID`</span>,<span class="va">df</span><span class="op">$</span><span class="va">GeneID</span> <span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">new_ids</span><span class="op">$</span><span class="va">`Gene ID`</span></span>
<span></span>
<span><span class="co"># Filter for valid GeneIDs</span></span>
<span><span class="va">subdf</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">GeneID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">annot</span><span class="op">$</span><span class="va">`Gene ID`</span><span class="op">)</span></span>
<span></span>
<span><span class="co">###  Step 2 </span></span>
<span><span class="co"># Aggregate length per gene across releases</span></span>
<span><span class="va">df_agg</span> <span class="op">&lt;-</span> <span class="va">subdf</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">.id</span>, <span class="va">GeneID</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>mean_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Length</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">GeneID</span>, values_from <span class="op">=</span> <span class="va">mean_length</span>, values_fill <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">".id"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">###  Step 3</span></span>
<span><span class="co"># Coefficient of variation across releases</span></span>
<span><span class="va">cv</span> <span class="op">&lt;-</span> <span class="va">cv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">df_agg</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">x</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check for how many genes the effective length has changed</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cv</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span> <span class="co">#4535 genes have their effective length changed</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;  1185  4535</span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/gghistogram.html" class="external-link">gghistogram</a></span><span class="op">(</span><span class="va">cv</span>,color <span class="op">=</span> <span class="st">"grey"</span>, fill <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Need_for_reanalysis_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Let's filter top 20 genes whose effective length has changed drastically and see what they are</span></span>
<span></span>
<span><span class="va">annot</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`Gene ID`</span>, <span class="va">`Product Description`</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">`Gene ID`</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Gene ID"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cv <span class="op">=</span> <span class="va">cv</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co">## We see a lot of hypothetical genes</span></span>
<span><span class="co">#&gt;                                                Product Description       cv</span></span>
<span><span class="co">#&gt; PF3D7_1312500       conserved Plasmodium protein, unknown function 124.1221</span></span>
<span><span class="co">#&gt; PF3D7_1105000                                           histone H4 123.2382</span></span>
<span><span class="co">#&gt; PF3D7_1102800              early transcribed membrane protein 11.2 122.8477</span></span>
<span><span class="co">#&gt; PF3D7_1252300       conserved Plasmodium protein, unknown function 120.9319</span></span>
<span><span class="co">#&gt; PF3D7_1133000       conserved Plasmodium protein, unknown function 119.7104</span></span>
<span><span class="co">#&gt; PF3D7_0926200       conserved Plasmodium protein, unknown function 118.7044</span></span>
<span><span class="co">#&gt; PF3D7_0627000       conserved Plasmodium protein, unknown function 117.7699</span></span>
<span><span class="co">#&gt; PF3D7_0221800                                 hypothetical protein 117.3651</span></span>
<span><span class="co">#&gt; PF3D7_0624500      anaphase-promoting complex subunit 11, putative 117.0429</span></span>
<span><span class="co">#&gt; PF3D7_0202800       conserved Plasmodium protein, unknown function 116.1717</span></span>
<span><span class="co">#&gt; PF3D7_0936000                              ring-exported protein 2 114.3509</span></span>
<span><span class="co">#&gt; PF3D7_1003500                 40S ribosomal protein S20e, putative 113.8707</span></span>
<span><span class="co">#&gt; PF3D7_1325700       conserved Plasmodium protein, unknown function 113.8358</span></span>
<span><span class="co">#&gt; PF3D7_0817800     respiratory chain complex 3 associated protein 3 113.8013</span></span>
<span><span class="co">#&gt; PF3D7_0323000 translation machinery-associated protein 7, putative 113.3105</span></span>
<span><span class="co">#&gt; PF3D7_0530000       conserved Plasmodium protein, unknown function 112.0993</span></span>
<span><span class="co">#&gt; PF3D7_1202900                       high mobility group protein B1 112.0186</span></span>
<span><span class="co">#&gt; PF3D7_1355800               splicing factor 3B subunit 5, putative 111.9045</span></span>
<span><span class="co">#&gt; PF3D7_1102700              early transcribed membrane protein 11.1 111.8313</span></span>
<span><span class="co">#&gt; PF3D7_0705700                  40S ribosomal protein S29, putative 111.1493</span></span>
<span></span>
<span><span class="co"># plot top 2 variable genes</span></span>
<span><span class="va">plts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">gid</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">GeneID</span> <span class="op">==</span> <span class="va">gid</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggline.html" class="external-link">ggline</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">".id"</span>, y <span class="op">=</span> <span class="st">"Length"</span>,color <span class="op">=</span> <span class="st">"#2595be"</span>, fill <span class="op">=</span> <span class="st">"#2595be"</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="va">gid</span>, x <span class="op">=</span> <span class="st">"Release"</span>, y <span class="op">=</span> <span class="st">"Effective Length"</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://strengejacke.github.io/sjPlot/" class="external-link">sjPlot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://strengejacke.github.io/sjPlot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">plts</span><span class="op">)</span></span></code></pre></div>
<p><img src="Need_for_reanalysis_files/figure-html/unnamed-chunk-3-2.png" width="700"></p>
<p>From the exploration above, we see a 79% of gene models have been
revised since release 50 and therefore inform us that any published
RNASeq dataset using annotations older than Release 50 must be
reanalysed since the read counts are destined to change for these
genes.</p>
<blockquote>
<p>If you are working on annotation of hypothetical genes, you should be
cautious about which assembly you use since your gene might appear to be
down regulated as per old annotation but this might change with change
in the current annotation.</p>
</blockquote>
<p>Let’s recycle the code above for <em>Toxoplasma</em>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GeneStructureTools</span><span class="op">)</span></span>
<span></span>
<span><span class="co">###  Step1</span></span>
<span><span class="co"># Remove version suffix from GeneIDs</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">tgme49.gffs</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/ldply.html" class="external-link">ldply</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>GeneID <span class="op">=</span> <span class="fu">GeneStructureTools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneStructureTools/man/removeVersion.html" class="external-link">removeVersion</a></span><span class="op">(</span><span class="va">GeneID</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get annotation table and identify outdated IDs</span></span>
<span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTable.html">getTable</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"Toxoplasma gondii ME49"</span>, db <span class="op">=</span> <span class="st">"toxodb"</span><span class="op">)</span></span>
<span><span class="va">old</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">GeneID</span><span class="op">)</span>, <span class="va">annot</span><span class="op">$</span><span class="va">`Gene ID`</span><span class="op">)</span> <span class="co">## 583 old IDs</span></span>
<span></span>
<span><span class="co"># Map outdated to Ensembl IDs</span></span>
<span><span class="va">new_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/toGeneid.html">toGeneid</a></span><span class="op">(</span><span class="va">old</span>, from <span class="op">=</span> <span class="st">"old"</span>, to <span class="op">=</span> <span class="st">"ensembl"</span>, org <span class="op">=</span> <span class="st">"Toxoplasma gondii ME49"</span>, db <span class="op">=</span> <span class="st">"toxodb"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`Gene ID`</span>, <span class="va">`Previous ID(s)`</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="co">## only 332 mapped</span></span>
<span></span>
<span><span class="co"># Replace outdated IDs in df. These are mostly apicoplast genes</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">GeneID</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">new_ids</span><span class="op">$</span><span class="va">`Gene ID`</span>,<span class="va">df</span><span class="op">$</span><span class="va">GeneID</span> <span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">new_ids</span><span class="op">$</span><span class="va">`Gene ID`</span></span>
<span></span>
<span><span class="co"># Filter for valid GeneIDs</span></span>
<span><span class="va">subdf</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">GeneID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">annot</span><span class="op">$</span><span class="va">`Gene ID`</span><span class="op">)</span></span>
<span></span>
<span><span class="co">###  Step 2 </span></span>
<span><span class="co"># Aggregate length per gene across releases</span></span>
<span><span class="va">df_agg</span> <span class="op">&lt;-</span> <span class="va">subdf</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">.id</span>, <span class="va">GeneID</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>mean_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Length</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">GeneID</span>, values_from <span class="op">=</span> <span class="va">mean_length</span>, values_fill <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">".id"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">###  Step 3</span></span>
<span><span class="co"># Coefficient of variation across releases</span></span>
<span><span class="va">cv</span> <span class="op">&lt;-</span> <span class="va">cv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">df_agg</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">x</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check for how many genes the effective length has changed</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cv</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span> <span class="co">#3390 genes have their effective length changed</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;  4947  3390</span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/gghistogram.html" class="external-link">gghistogram</a></span><span class="op">(</span><span class="va">cv</span>,color <span class="op">=</span> <span class="st">"grey"</span>, fill <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Need_for_reanalysis_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Let's filter top 20 genes whose effective length has changed drastically and see what they are</span></span>
<span></span>
<span><span class="va">annot</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`Gene ID`</span>, <span class="va">`Product Description`</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">`Gene ID`</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Gene ID"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cv <span class="op">=</span> <span class="va">cv</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co">## We see a lot of hypothetical genes</span></span>
<span><span class="co">#&gt;                                                          Product Description</span></span>
<span><span class="co">#&gt; TGME49_313740               zinc finger (CCCH type) motif-containing protein</span></span>
<span><span class="co">#&gt; TGME49_265100                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_288850                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_245440                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_223845                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_313890                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_231998                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_291160                 dynein light polypeptide 4, axonemal, putative</span></span>
<span><span class="co">#&gt; TGME49_311820                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_230135                                     Kinesin-associated protein</span></span>
<span><span class="co">#&gt; TGME49_209945                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_254140                          DNA-directed RNA polymerase II RPABC4</span></span>
<span><span class="co">#&gt; TGME49_244890                                               cyclin, putative</span></span>
<span><span class="co">#&gt; TGME49_310177                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_240790                                   dynein light chain, putative</span></span>
<span><span class="co">#&gt; TGME49_260825                                           hypothetical protein</span></span>
<span><span class="co">#&gt; TGME49_308830                           dual specificity protein phosphatase</span></span>
<span><span class="co">#&gt; TGME49_261470           U6 snRNA-associated Sm family protein LSm6, putative</span></span>
<span><span class="co">#&gt; TGME49_275750                    small nuclear ribonucleoprotein E, putative</span></span>
<span><span class="co">#&gt; TGME49_283470 Kazal-type serine protease inhibitor domain-containing protein</span></span>
<span><span class="co">#&gt;                     cv</span></span>
<span><span class="co">#&gt; TGME49_313740 178.6757</span></span>
<span><span class="co">#&gt; TGME49_265100 171.6388</span></span>
<span><span class="co">#&gt; TGME49_288850 161.6959</span></span>
<span><span class="co">#&gt; TGME49_245440 157.8947</span></span>
<span><span class="co">#&gt; TGME49_223845 152.1074</span></span>
<span><span class="co">#&gt; TGME49_313890 151.9148</span></span>
<span><span class="co">#&gt; TGME49_231998 146.3761</span></span>
<span><span class="co">#&gt; TGME49_291160 143.9601</span></span>
<span><span class="co">#&gt; TGME49_311820 134.4396</span></span>
<span><span class="co">#&gt; TGME49_230135 134.2367</span></span>
<span><span class="co">#&gt; TGME49_209945 129.9972</span></span>
<span><span class="co">#&gt; TGME49_254140 126.7883</span></span>
<span><span class="co">#&gt; TGME49_244890 125.7379</span></span>
<span><span class="co">#&gt; TGME49_310177 125.1441</span></span>
<span><span class="co">#&gt; TGME49_240790 121.8464</span></span>
<span><span class="co">#&gt; TGME49_260825 121.0684</span></span>
<span><span class="co">#&gt; TGME49_308830 120.6382</span></span>
<span><span class="co">#&gt; TGME49_261470 120.3902</span></span>
<span><span class="co">#&gt; TGME49_275750 119.9072</span></span>
<span><span class="co">#&gt; TGME49_283470 119.0964</span></span>
<span></span>
<span><span class="co"># plot top 2 variable genes</span></span>
<span><span class="va">plts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cv</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">gid</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">GeneID</span> <span class="op">==</span> <span class="va">gid</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggline.html" class="external-link">ggline</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">".id"</span>, y <span class="op">=</span> <span class="st">"Length"</span>,color <span class="op">=</span> <span class="st">"#2595be"</span>, fill <span class="op">=</span> <span class="st">"#2595be"</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="va">gid</span>, x <span class="op">=</span> <span class="st">"Release"</span>, y <span class="op">=</span> <span class="st">"Effective Length"</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://strengejacke.github.io/sjPlot/" class="external-link">sjPlot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://strengejacke.github.io/sjPlot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">plts</span><span class="op">)</span></span></code></pre></div>
<p><img src="Need_for_reanalysis_files/figure-html/unnamed-chunk-4-2.png" width="700"></p>
<p>In this case 40% of the genes show change in effective length as
compared to past releases and this change happened after release 60
(Release 66 to be precise but for brevity of vignette, we chose to use
60 and 68).</p>
</div>
<div class="section level2">
<h2 id="discussion">Discussion<a class="anchor" aria-label="anchor" href="#discussion"></a>
</h2>
<p>The scope of reannotation directly influences whether legacy datasets
remain reliable or must be reprocessed; such technical choices
inevitably affect downstream biological interpretations. We found that
every major bulk transcriptomic study for <em>Plasmodium</em> in
<code>PlasmoDB</code> relies on annotations predating release 51 and can
benefit from reanalysis. Therefore, any analysis—be it correlation,
co‐expression, or single‐cell mapping—should use reanalyzed data based
on up‐to‐date GFFs. Given VEuPathDB’s constrained resources, the
reanalysis responsibility falls to the end users to regenerate and share
these reprocessed datasets whenever feasible.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session">Session<a class="anchor" aria-label="anchor" href="#session"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 26100)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_India.utf8  LC_CTYPE=English_India.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_India.utf8 LC_NUMERIC=C                  </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_India.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Riyadh</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] sjPlot_2.9.0              GeneStructureTools_1.24.0</span></span>
<span><span class="co">#&gt;  [3] ggpubr_0.6.1              ggplot2_3.5.2            </span></span>
<span><span class="co">#&gt;  [5] tibble_3.3.0              tidyr_1.3.1              </span></span>
<span><span class="co">#&gt;  [7] dplyr_1.1.4               plasmoRUtils_1.1.0       </span></span>
<span><span class="co">#&gt;  [9] rlang_1.1.6               readr_2.1.5              </span></span>
<span><span class="co">#&gt; [11] janitor_2.2.1             BiocStyle_2.32.1         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] IRanges_2.38.1                     dichromat_2.0-0.1                 </span></span>
<span><span class="co">#&gt;   [3] vroom_1.6.5                        progress_1.2.3                    </span></span>
<span><span class="co">#&gt;   [5] vsn_3.72.0                         nnet_7.3-19                       </span></span>
<span><span class="co">#&gt;   [7] Biostrings_2.72.1                  vctrs_0.6.5                       </span></span>
<span><span class="co">#&gt;   [9] digest_0.6.37                      png_0.1-8                         </span></span>
<span><span class="co">#&gt;  [11] proxy_0.4-27                       MSnbase_2.30.1                    </span></span>
<span><span class="co">#&gt;  [13] deldir_2.0-4                       parallelly_1.45.1                 </span></span>
<span><span class="co">#&gt;  [15] MASS_7.3-61                        pkgdown_2.1.3                     </span></span>
<span><span class="co">#&gt;  [17] reshape2_1.4.4                     foreach_1.5.2                     </span></span>
<span><span class="co">#&gt;  [19] BiocGenerics_0.50.0                withr_3.0.2                       </span></span>
<span><span class="co">#&gt;  [21] xfun_0.52                          survival_3.8-3                    </span></span>
<span><span class="co">#&gt;  [23] memoise_2.0.1                      hexbin_1.28.5                     </span></span>
<span><span class="co">#&gt;  [25] ggsci_3.2.0                        mixtools_2.0.0.1                  </span></span>
<span><span class="co">#&gt;  [27] systemfonts_1.2.3                  ragg_1.4.0                        </span></span>
<span><span class="co">#&gt;  [29] gtools_3.9.5                       easyPubMed_2.13                   </span></span>
<span><span class="co">#&gt;  [31] Formula_1.2-5                      prettyunits_1.2.0                 </span></span>
<span><span class="co">#&gt;  [33] KEGGREST_1.44.1                    promises_1.3.3                    </span></span>
<span><span class="co">#&gt;  [35] httr_1.4.7                         rstatix_0.7.2                     </span></span>
<span><span class="co">#&gt;  [37] restfulr_0.0.16                    globals_0.18.0                    </span></span>
<span><span class="co">#&gt;  [39] ps_1.9.1                           rstudioapi_0.17.1                 </span></span>
<span><span class="co">#&gt;  [41] UCSC.utils_1.0.0                   generics_0.1.4                    </span></span>
<span><span class="co">#&gt;  [43] base64enc_0.1-3                    processx_3.8.6                    </span></span>
<span><span class="co">#&gt;  [45] curl_6.4.0                         ncdf4_1.24                        </span></span>
<span><span class="co">#&gt;  [47] S4Vectors_0.42.1                   zlibbioc_1.50.0                   </span></span>
<span><span class="co">#&gt;  [49] ScaledMatrix_1.12.0                randomForest_4.7-1.2              </span></span>
<span><span class="co">#&gt;  [51] bio3d_2.4-5                        GenomeInfoDbData_1.2.12           </span></span>
<span><span class="co">#&gt;  [53] SparseArray_1.4.8                  xtable_1.8-4                      </span></span>
<span><span class="co">#&gt;  [55] stringr_1.5.1                      desc_1.4.3                        </span></span>
<span><span class="co">#&gt;  [57] doParallel_1.0.17                  evaluate_1.0.4                    </span></span>
<span><span class="co">#&gt;  [59] S4Arrays_1.4.1                     BiocFileCache_2.12.0              </span></span>
<span><span class="co">#&gt;  [61] preprocessCore_1.66.0              hms_1.1.3                         </span></span>
<span><span class="co">#&gt;  [63] GenomicRanges_1.56.2               bookdown_0.43                     </span></span>
<span><span class="co">#&gt;  [65] irlba_2.3.5.1                      colorspace_2.1-1                  </span></span>
<span><span class="co">#&gt;  [67] filelock_1.0.3                     magrittr_2.0.3                    </span></span>
<span><span class="co">#&gt;  [69] snakecase_0.11.1                   later_1.4.2                       </span></span>
<span><span class="co">#&gt;  [71] viridis_0.6.5                      lattice_0.22-6                    </span></span>
<span><span class="co">#&gt;  [73] MsCoreUtils_1.16.1                 future.apply_1.20.0               </span></span>
<span><span class="co">#&gt;  [75] SparseM_1.84-2                     XML_3.99-0.18                     </span></span>
<span><span class="co">#&gt;  [77] scuttle_1.14.0                     matrixStats_1.5.0                 </span></span>
<span><span class="co">#&gt;  [79] Hmisc_5.2-3                        class_7.3-22                      </span></span>
<span><span class="co">#&gt;  [81] pillar_1.11.0                      nlme_3.1-166                      </span></span>
<span><span class="co">#&gt;  [83] iterators_1.0.14                   compiler_4.4.1                    </span></span>
<span><span class="co">#&gt;  [85] beachmat_2.20.0                    stringi_1.8.7                     </span></span>
<span><span class="co">#&gt;  [87] gower_1.0.2                        SummarizedExperiment_1.34.0       </span></span>
<span><span class="co">#&gt;  [89] dendextend_1.19.1                  lubridate_1.9.4                   </span></span>
<span><span class="co">#&gt;  [91] GenomicAlignments_1.40.0           drawProteins_1.24.0               </span></span>
<span><span class="co">#&gt;  [93] plyr_1.8.9                         crayon_1.5.3                      </span></span>
<span><span class="co">#&gt;  [95] abind_1.4-8                        BiocIO_1.14.0                     </span></span>
<span><span class="co">#&gt;  [97] bit_4.6.0                          chromote_0.5.1                    </span></span>
<span><span class="co">#&gt;  [99] pcaMethods_1.96.0                  codetools_0.2-20                  </span></span>
<span><span class="co">#&gt; [101] textshaping_1.0.1                  recipes_1.3.1                     </span></span>
<span><span class="co">#&gt; [103] BiocSingular_1.20.0                MLInterfaces_1.84.0               </span></span>
<span><span class="co">#&gt; [105] bslib_0.9.0                        e1071_1.7-16                      </span></span>
<span><span class="co">#&gt; [107] biovizBase_1.52.0                  plotly_4.11.0                     </span></span>
<span><span class="co">#&gt; [109] LaplacesDemon_16.1.6               MultiAssayExperiment_1.30.3       </span></span>
<span><span class="co">#&gt; [111] splines_4.4.1                      Rcpp_1.1.0                        </span></span>
<span><span class="co">#&gt; [113] dbplyr_2.5.0                       sparseMatrixStats_1.16.0          </span></span>
<span><span class="co">#&gt; [115] interp_1.1-6                       knitr_1.50                        </span></span>
<span><span class="co">#&gt; [117] blob_1.2.4                         clue_0.3-66                       </span></span>
<span><span class="co">#&gt; [119] mzR_2.38.0                         AnnotationFilter_1.28.0           </span></span>
<span><span class="co">#&gt; [121] fs_1.6.6                           checkmate_2.3.2                   </span></span>
<span><span class="co">#&gt; [123] QFeatures_1.14.2                   listenv_0.9.1                     </span></span>
<span><span class="co">#&gt; [125] mzID_1.42.0                        DelayedMatrixStats_1.26.0         </span></span>
<span><span class="co">#&gt; [127] Gviz_1.48.0                        ggsignif_0.6.4                    </span></span>
<span><span class="co">#&gt; [129] Matrix_1.7-1                       statmod_1.5.0                     </span></span>
<span><span class="co">#&gt; [131] tzdb_0.5.0                         lpSolve_5.6.23                    </span></span>
<span><span class="co">#&gt; [133] pkgconfig_2.0.3                    tools_4.4.1                       </span></span>
<span><span class="co">#&gt; [135] cachem_1.1.0                       BSgenome.Mmusculus.UCSC.mm10_1.4.3</span></span>
<span><span class="co">#&gt; [137] RSQLite_2.4.1                      viridisLite_0.4.2                 </span></span>
<span><span class="co">#&gt; [139] rvest_1.0.4                        DBI_1.2.3                         </span></span>
<span><span class="co">#&gt; [141] impute_1.78.0                      fastmap_1.2.0                     </span></span>
<span><span class="co">#&gt; [143] rmarkdown_2.29                     scales_1.4.0                      </span></span>
<span><span class="co">#&gt; [145] grid_4.4.1                         gt_1.0.0                          </span></span>
<span><span class="co">#&gt; [147] Rsamtools_2.20.0                   broom_1.0.8                       </span></span>
<span><span class="co">#&gt; [149] sass_0.4.10                        coda_0.19-4.1                     </span></span>
<span><span class="co">#&gt; [151] FNN_1.1.4.1                        BiocManager_1.30.26               </span></span>
<span><span class="co">#&gt; [153] VariantAnnotation_1.50.0           graph_1.82.0                      </span></span>
<span><span class="co">#&gt; [155] carData_3.0-5                      SingleR_2.6.0                     </span></span>
<span><span class="co">#&gt; [157] rpart_4.1.23                       farver_2.1.2                      </span></span>
<span><span class="co">#&gt; [159] yaml_2.3.10                        AnnotationForge_1.46.0            </span></span>
<span><span class="co">#&gt; [161] latticeExtra_0.6-30                foreign_0.8-87                    </span></span>
<span><span class="co">#&gt; [163] MatrixGenerics_1.16.0              rtracklayer_1.64.0                </span></span>
<span><span class="co">#&gt; [165] cli_3.6.5                          purrr_1.1.0                       </span></span>
<span><span class="co">#&gt; [167] stats4_4.4.1                       txdbmaker_1.0.1                   </span></span>
<span><span class="co">#&gt; [169] lifecycle_1.0.4                    caret_7.0-1                       </span></span>
<span><span class="co">#&gt; [171] Biobase_2.64.0                     mvtnorm_1.3-3                     </span></span>
<span><span class="co">#&gt; [173] lava_1.8.1                         kernlab_0.9-33                    </span></span>
<span><span class="co">#&gt; [175] backports_1.5.0                    BiocParallel_1.38.0               </span></span>
<span><span class="co">#&gt; [177] annotate_1.82.0                    timechange_0.3.0                  </span></span>
<span><span class="co">#&gt; [179] gtable_0.3.6                       rjson_0.2.23                      </span></span>
<span><span class="co">#&gt; [181] parallel_4.4.1                     pROC_1.18.5                       </span></span>
<span><span class="co">#&gt; [183] limma_3.60.6                       jsonlite_2.0.0                    </span></span>
<span><span class="co">#&gt; [185] bitops_1.0-9                       bit64_4.6.0-1                     </span></span>
<span><span class="co">#&gt; [187] pRoloc_1.44.1                      jquerylib_0.1.4                   </span></span>
<span><span class="co">#&gt; [189] segmented_2.1-4                    timeDate_4041.110                 </span></span>
<span><span class="co">#&gt; [191] lazyeval_0.2.2                     htmltools_0.5.8.1                 </span></span>
<span><span class="co">#&gt; [193] affy_1.82.0                        GO.db_3.19.1                      </span></span>
<span><span class="co">#&gt; [195] rappdirs_0.3.3                     ensembldb_2.28.1                  </span></span>
<span><span class="co">#&gt; [197] glue_1.8.0                         polyglotr_1.7.0                   </span></span>
<span><span class="co">#&gt; [199] httr2_1.2.0                        XVector_0.44.0                    </span></span>
<span><span class="co">#&gt; [201] RCurl_1.98-1.17                    MALDIquant_1.22.3                 </span></span>
<span><span class="co">#&gt; [203] mclust_6.1.1                       BSgenome_1.72.0                   </span></span>
<span><span class="co">#&gt; [205] jpeg_0.1-11                        gridExtra_2.3                     </span></span>
<span><span class="co">#&gt; [207] igraph_2.1.4                       R6_2.6.1                          </span></span>
<span><span class="co">#&gt; [209] SingleCellExperiment_1.26.0        labeling_0.4.3                    </span></span>
<span><span class="co">#&gt; [211] GenomicFeatures_1.56.0             cluster_2.1.8                     </span></span>
<span><span class="co">#&gt; [213] stringdist_0.9.15                  GenomeInfoDb_1.40.1               </span></span>
<span><span class="co">#&gt; [215] ipred_0.9-15                       DelayedArray_0.30.1               </span></span>
<span><span class="co">#&gt; [217] tidyselect_1.2.1                   ProtGenerics_1.36.0               </span></span>
<span><span class="co">#&gt; [219] htmlTable_2.4.3                    sampling_2.11                     </span></span>
<span><span class="co">#&gt; [221] xml2_1.3.8                         car_3.1-3                         </span></span>
<span><span class="co">#&gt; [223] AnnotationDbi_1.66.0               future_1.67.0                     </span></span>
<span><span class="co">#&gt; [225] ModelMetrics_1.2.2.2               rsvd_1.0.5                        </span></span>
<span><span class="co">#&gt; [227] affyio_1.74.0                      topGO_2.56.0                      </span></span>
<span><span class="co">#&gt; [229] data.table_1.17.8                  websocket_1.4.4                   </span></span>
<span><span class="co">#&gt; [231] mgsub_1.7.3                        htmlwidgets_1.6.4                 </span></span>
<span><span class="co">#&gt; [233] RColorBrewer_1.1-3                 biomaRt_2.60.1                    </span></span>
<span><span class="co">#&gt; [235] hardhat_1.4.1                      prodlim_2025.04.28                </span></span>
<span><span class="co">#&gt; [237] PSMatch_1.8.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Rohit Satyam, Alberto Maillo, David Gomez-Cabrero, Arnab Pain.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
